# Introduction

In his Nobel Prize speech in 1973 [@Anfinsen1973] Anfinsen postulated one of the basic principles in molecular biology, which is known as *Anfinsen's dogma*: a protein's native structure is uniquely determined by its amino acid sequence. 
With certain exceptions (e.g. [IDP](#abbrev) [@Wright1999]), this dogma has proven to hold true for the majority of proteins.

Ever since, it is regarded as the biggest challenge in structural bioinformatics [@Samish2015], to realiably predict a protein's structure given only its amino acid sequence. 
*De-novo* protein structure prediction methods use physical or knowledge based energy potentials to find a protein conformation that minimizes the protein's energy landscape. 
However, these methods are limited by the complexity of the conformational space and the accuracy of the energy potentials. 
Considering a protein with 150 amino acids, that has approximately 450 degrees of freedom, 
Regarding the rotational and translational degrees of freedom of the protein chain, the complexity scales with XXX [@Anfinsen1973].

Far more successfull are template-based modelling approaches. 
Given the observation that structure is more conserved than sequence in a protein family [@Lesk1980], the structure of a target protein can be inferred from a homologue protein [@Sander1991]. 
The degree of structural conservation is linked to the level of pairwise sequence identity [@Chothia1986]. 
Therefore, the accuracy of a model crucially depends on the sequence identity between target and template and determines the applicability of the model [@Marti-Renom2000]. 
By definition, homology derived models are unable to capture new folds [@Dorn2014] and their main limitation lies in the availability of suitable templates.  

```{r seq-str-gap, echo = FALSE, out.width = '100%', fig.cap = 'Yearly growth of number of solved structures in the PDB[@Berman2000] and protein sequences in the Uniprot[@TheUniProtConsortium2013].'}
knitr::include_url("img/pdb_growth.html", height = "750px")
```

Unfortunately, the number of solved protein structures increases only slowly, as experimental methods are both time consuming and expensive [@Dorn2014].
The [PDB](#abbrev)[@Berman2000] is the main repository for marcomolecular structures and currently (Jul 2017) holds about 120 000 atomic models of proteins. 
The primary technique for determining protein structures is X-ray crystallography, accounting for roughly 90% of entries in the [PDB](#abbrev). 
About 9% of protein structures have been solved using [NMR](#abbrev) and less than 1% using [EM](#abbrev) (see FIG 1).

All three experimental techniques have advantages and limitations with respect to certain modelling aspects.
X-ray crystallography requires the protein to form crystals, which is an arduous and sometimes impossible task. 
Furthermore, crystal packing forces the protein into a unnatural and rigid environment preventing the observation of conformational flexibility. 
$\ac{NMR}$ studies the protein in an physiological environment in solution and enables the study of protein dynamics as ensembles of protein structures can be observed. However, $\ac{NMR}$ is limited to look at small proteins. 
Recently, $\ac{EM}$ has undergone a "resolution revolution" [@Egelman2016] and macromolecular structures have been solved with resolutions up to 2A[citation]. 
The limit of $\ac{cryo-EM}$ lies in the size of proteins.

Compared to the tedious task of revealing atomic resolution of a protein tertiary structure, it has become very easy to decipher the primary sequence of proteins. 
With the latest sequencing technologies [examples], it takes only hours to sequence millions of basepaires at low costs [example numbers] and the number of sequenced genomes has risen tremendously.
The UniProtKB [@TheUniProtConsortium2013], the leading resource for protein sequences, contains more than 80 million sequence entries (24 July 2017). 
 
Consequently, the gap between the number of protein structures and sequences is still growing and even new developments as single protein structure determination [@CITE] are not expected to close this gap near in time.
[Figure sequence structure gap]

Protein structure determines protein function. Therefore, structural insights are of uttermost importance. They are essential for a detailed understanding of chemical reactions, regulatory processes and transport mechanisms.
They are fundamental for the design of drugs and antibiotics. Moreover structural abnormalities can lead to misfolding and aggregation potentially causing diseases so studying them is pathologically relevant.

The aformentioned trends illustrate the need of computational methods and motivate research to solve *Ansinsens Dogma* to reliably predict protein structures from sequence alone. 

## Protein Structure

- Primary: Amino Acid Sewuence
- Secondary: Helices, sheets, coils, repeats,..
- tertiary: interaction of secondary structure elementws
- quartary: interaction of domains


### Amino Acid Interactions {#amino-acid-interactions}

The Venn diagram in figure \@ref(fig:amino-acid-physico-chemical-properties) displays a typical classification of amino acids with respect to their physico-chemical properties. 


The aromatic amino acids tryptophan (W), tyrosine (Y), phenylalanine (F), and histidine (H) contain an aromatic ring system. 
Generally, aromatic ring systems are planar, and electons are shared over the whole ring structure.
Interactions between aromatic residues have very constrained geometries regarding the angle between the centroid of their rings. 
The $\pi$-electron systems favour T-shaped or offset stacked conformations [@Waters2002].
Preferred distances between aromatic residues have been observed between 4.5\AA and 7\AA of their ring centroids [@Burley1985].

Cysteine (C) residues can form disulphide bonds, which are the only covalent bonds between two amino acid side chains.
They comprise the strongest side chain interactions in protein structures and their length varies between 3.5\AA to 4\AA.
Disulphide bonds also have a well defined geometry: there are five dihedral angles in a disulphide bond resulting in 20 different possible configurations. 
Only one configuration is favoured so that the dihedral angle between the carbon and sulfur atoms is close to 90 degrees [@Thornton1981].
They play a very important role in stabilizing protein structures. 
The number of disulfide bonds is negatively correlated with protein length: smaller proteins have more disulfide bonds helping to stabilize the structure in absence of strong hydrophobic packing in the core.
It has also been found that disulfide bonds are more frequently observed in proteins of hyperthermophilic bacteria, being positively selected for increased stability [@Bastolla2005].

Salt bridges are based on electrostatic interactions between positively charged residues (arginine (R) and lysine (K)) and negatively charged residues (aspartic acid (D) and glutamic acid (E)). 
The strength of electrostatic interactions, as described by Coulomb's law, decreases with distance between the point charges at the functional groups. 
It has been found to be maximal at 4\AA with respect to the functional groups of the both residues [@Donald2011].

Hydrogen bonds can be formed between a donor residue which possesses an hydrogen atom attached to a strongly electronegative atom and an acceptor residue which possesses an electronegative atom with a lone electron pair. 
They are electrostatic interactions as well and thus their strength depends on distance as well. 
Hydrogen bonds are formed at distances of 2.4\AA to 3.5\AA between the non-hydrogen atoms (Berg JM, Tymoczko JL, 2002).

Salt bridges as well as hydrogen bonds have strong geometric preferences (Kumar and Nussinov, 1999). 
The geometry of a hydrogen bond depends on the angle between the HB donor, the hydrogen atom and the HB acceptor (Torshin et al., 2002).

Cationâ€“$\pi$ interactions are formed between positively charged or partially charged amino acids with amino groups (K,R,Q,E) and aromatic residues (W,Y,F,H). 
The preferential distance of the amino group to the $\pi$-electron system has been determined between 3.4\AA and 6\AA [@Burley1986] [@Crowley2005]
Their role in stabilizing protein structures is still under debate [@Slutsky2004].

Proline residues are conformationally restricted, with the alpha-amino group of the backbone directly attached to the side chain. 
The sterical rigidity of the proline side chain restricts the backbone angle and thus affects secondary structure formation.
Proline is known as a helix-breaker. 
Whereas other aromatic side chains are defined by their negatively charged $\pi$ faces, the face of proline side chains is partially positively charged.
Thus, aromatic and proline residues can interact favorably with each other.
Once due to the hydrophobic nature of the residues and also due to the interaction between the negatively charged aroamtic $\pi$ face and the polarized C-H bonds in proline, called a CH/$\pi$  interaction.

Petersen et al. (2012) found clear secondary structure elements preferences for each amino acid pair. 
For example, residue pairs containing Alanine and Leucine are predominantly found in buried $\alpha$-helices, whereas pairs containing Isoleucine and Valine preferentially are located in $\beta$-sheet environments. 
Of course, solvent accessibility represents an important criterion for residue interactions. 
Hydrophobic residues are rather buried in the structure, whereas polar and charged residues are found more frequently on the protein surface and interact with water molecules.

```{r amino-acid-physico-chemical-properties, echo = FALSE, out.width = '50%',  fig.cap = 'Physico-chemical properties of amino acids. The 20 naturally occuring amino acids are grouped with respect to ten physico-chemical properties. Adapted from Figure 1a in [@Livingstone1993].'}
knitr::include_graphics("img/amino_acid_physico_chemical_properties_venn_diagramm.png")
```


## Structure Prediction

Despite the knowledge of Anfinsen's postulate, we are not able to reliably predict the structure of a protein from its sequence alone. Generally it is assumed that a protein folds into a unique, well-defined native structure that is near the global free energy minimum (\autoref{fig:folding_funnel}). Levinthal's paradox [@Levinthal1969] describes the complexity of the folding process towards this minimum. It stresses the problem that it is not possible for a protein to exhaustively search the conformational space to get to its native fold. Due to the "combinatorial explosion" of possible conformations, an exhaustive search would take unreasonably long. Hence, it is not a feasible approach for structure prediction to scan all possible conformations. Different approaches have been developed over time to overcome or elude this problem.


### Template-based methods

Homology modeling is by far the most successful approach to structure prediction. 
The basic concept of this strategy relates to the fact that structure is more conserved than sequence [@Lesk1980]. 
After detecting a homologous protein of known structure, that has sufficient sequence similarity, it can be used as a template to model the structure of the target protein.

The degree of structural conservation is linked to the level of pariwise sequence identity [@Chothia1986]. 
Homology Modelling is assumed to yield reliably accurate models when query and target protein share more than 30\% sequence similarity, depending on the sequence length (*safe homology zone*) [@Sander1991]. 
Below a threshold of ~20-35% pairwise sequence identity (*twighlight-zone*) the number of false positives regarding structural similarity explodes and structural inference becomes less reliable and more than 95% of structures are dissmilar [@Rost1999]. 
Advances in remote homology detection and alignment generation have improved the quality of models, even beyond the once postulated limit of the *twighlight-zone* [@Yan2013].
Integration of multiple templates has also proved to increase model quality [@Meier2015]

After the identification of a suitable template, there are different strategies that can be followed to obtain a model for the target protein. 
The the backbone of the model is generated by simply copying the coordinates of the target backbone atoms onto the model. 
Non-aligned residues due to gaps in the alignment have to be modelled \textit{de-novo}, meaning from scratch.
This can be done by a knowledge-based search for suitable fragments in the PDB or by true energy-based \textit{de-novo} modelling. 
When the backbone is generated, the side chains are modelled, usually by searching rotamer libraries for energetically favoured residue conformations. 
Finally, the model is energetically optimized in an iterative procedure. 
Force fields are applied to correct the backbone and side chain conformations [@Gu2009}. 
Several automated pipelines for homology modelling are well-established (Modeller [@Eswar2007}, 3D-Jigsaw [@Bates2001}, SwissModel [@Arnold2006}) which allow more or less manual intervention in the modelling process. 

        
Fold Recognition describes the inverse folding problem \citep{Bowie1993}: instead of finding the compatible structure for a given sequence, one tries to find sequences that fit onto a given structure. Whether the query sequence fits a structure from the database is not determined by sequence similarities but rather energetic or environment specific measures. Thus, fold recognition methods are able to recognize structural similarity even in the absence of sequence similarity. The rationale basis for this strategy is the assumption that the fold space is limited. It has been found that seemingly unrelated proteins often adopt similar folds. This might be due to divergent evolution (proteins are related, but homology cannot be detected at the corresponding sequence level) or convergent evolution (functional requirements lead to similar folds for unrelated proteins) \citep{Gu2009}. Early approaches include profile based methods. Here, the structural information of the protein is encoded into profiles, which subsequently are aligned to the sequences \citep{Bowie1991,Fischer1996,Ouzounis1993}. Advanced techniques are known as  "threading" techniques, describing the process of threading a sequence through a structure and determining the optimal fit via energy functions. \citep{Jones1992,Jones1998,Lemer1995}


### Template-free structure prediction

Ab initio or de-novo modeling techniques implement Anfinsenâ€™s Dogma most closely in mimicking the folding process based only on physico-chemical principles.
Energy functions (physical or knowledge-based) are used to describe the folding landscape and are minimized to arrive at the global energy minimum corresponding to the native conformation. 
Since the native conformation can be found near the global energy minimum of the folding landscape, energy functions (physical or knowledge-based) have been developed to describe this landscape.
With respect to the idea of a folding funnel, the energy function is minimized to mimic the folding process that automatically leads to the global minimum.
 Again, there exist numerous webservers that combine energy minimization,  threading techniques and fragment-based approaches, e.g. Rosetta \citep{Simons1999}, Tasser \citep{Zhang2004}, Touchstone II \citep{Zhang2003}.

Drawbacks of these methods are the time requirements due to the computational complexity of energy functions as well as their inaccuracy. 
    
    

Minimize a physical or knowledge-based energy function for the protein. This has huge complexity due to large conformational space that needs to be sampled. 

### contact assisted denovo predictions

Structure Reconstruction from true contacts maps works well. Even a small number of contacts is sufficient to reconstruct the fold of the protein. Distance maps work even better. 

What is the optimal distance cutoff to define a contact? 
Duarte et al 2010: between 8 and 12A
Dyrka et al 2016
Konopka et al 2014
Sathyapriya et al 2009

Many studies that successfuly predict structures denovo with the help of predicted contact.

Vice versa, because contacts at large primary distances are rare, they are most informative for protein structure prediction: Izarzugaza J, Gran Ëœa O, Tress M, Valencia A, Clarke N (2007) Assessment of intramolecular contact predictions for CASP7







## Contact Prediction

### Correlated mutations

contact prediction methods aim to identify correlated mutations from an  alignment of homologue protein sequences. 
main assumption is that two interacting amino acid residues are coevolving: mutation of one of the two residues can be compensated by mutation of the other residue

### Local methods
MI and correlation measures suffer from transitivity of correlations


### Global methods

In 2010, Burger and Nijmwegen  developed a Bayesian network model [@Burger2010] that is able to disentangle direct from indirect correlations and introducing informative priors to improve precision. independently of weight in 2009

### Meta-predictors

- combining different approaches
 - jones et al: overlap between methods but also many unique predictions
- machine learning methods incorporate sequence-derived features:
  - secondary structure predictions
  - solvent accessibilty
  - contact potentials
  - msa properties
  - pssms 
  - physico-chemcial properties of amino acids

However, Meta-predictors will improve if basic methods improve. Ultra-deep learning paper identifies coevolution features as crucial feature. 

### Benchmarking methods {#intro-cp-evaluation}

- threshold for defining a contact: 
usually distance between $C_\beta$ atoms ($C_\alpha$ for Glycin) < 8 angstrom. 

- PPV: TP/(FP+TP)
fraction of correct predictions among all predictions

#### Sequence Separation

Residue pairs too close in sequence (e.g $|i-j| < 6$) are filtered out for benchmarking contact prediction methods.
Commonly, sequence separation bins are applied to distuinguish short ($6 \le |i-j| < 12$), medium ($12 \le |i-j| < 24$) and long range ($|i-j| \ge 24$) contacts.
Especially long range contacts are of importance for structure prediction as they are informative and able to constrain the overal fold of a protein [@CITE].

In contrast, residue pairs close in sequence ($|i-j| < 6$) are trivial to predict and often reflect local geometrical constraints from secondary structure elements.
Figure \@ref(fig:Cb-distribution) shows the distribution of $\Cb$ distances for various minimal sequence separation thresholds. 
Without filtering local residue pairs (sequence separation 1), there are several additional peaks in the distribution around $5.5\AA$, $7.4\AA$ and $10.6\AA$ that can be attributed to local interactions in e.g. helices (see Figure \@ref(fig:peaks-Cb-distribution)).
 
 
(ref:caption-Cb-distribution) Distribution of residue pair $\Cb$ distances over all proteins in the dataset (see Methods \@ref(dataset)) at different minimal sequence separation thresholds: blue = $|i-j| > 1$ (all residue pairs), orange = $|i-j| > 6$, green = $|i-j| > 12$, red = $|i-j| > 24$.

```{r Cb-distribution, echo = FALSE, out.width = '100%', out.height = '100%', fig.cap = '(ref:caption-Cb-distribution)'}
knitr::include_url("img/dataset_statistics/Cb_distribution_all_data43579541_log.html", height="500px")
```

(ref:caption-peaks-Cb-distribution) $\Cb$ distances between neighboring residues in $\alpha$-helices. Left: Direct neighbors in $\alpha$-helices have $\Cb$ distances around $5.4\AA$ due to the geometrical constraints from $\alpha$-helical architecture. Right: Residues separated by two positions ($|i-j| = 2$) are less geometrically restricted to $\Cb$ distances between $7\AA$ and $7.5\AA$.

```{r peaks-Cb-distribution, echo = FALSE, out.width = '50%', fig.show = 'hold', fig.cap = '(ref:caption-peaks-Cb-distribution)'}
knitr::include_graphics(c("img/dataset_statistics/cb_distribution_peak_5-6.png","img/dataset_statistics/cb_distribution_peak_7.png"))
```

#### Contact Definition

Methods are validated with respect to precision of predictions (see section \@ref(intro-cp-evaluation)), where a correctly predicted contact is commonly defined as a residue pair whose $\Cb$ atoms are within 8 $\AA$ [@CASP].
However, whether two residues truly interact in a protein structure depends only marginally on the distance between their atoms.
More importantly, interactions between side-chains depend on their physico-chemical properties, on their orientation and on their environment [@Bettsa] (see section \@ref(amino-acid-interactions).

Other distance thresholds or definitions for contacts (e.g minimal atomic distances or distance between functional groups) have been studied as well.
In fact, Duarte and colleagues found that using a $\Cb$ distance threshold between 9$\AA$ and 11$\AA$ yields optimal results when predicting the 3D structure from the respective contacts [@Duarte2010].

Anishchenko and colleagues [@Anishchenko2017] analysed false positive predictions with respect to a minimal atom distance threshold $< 5 \AA$, as they found that this cutoff optimally defines direct physical interactions of residue pairs.

However, keeping in mind that predicted contacts shall be used for structure prediction protocols, it is more convenient to apply a simple $\CB$ cutoff as this threshold can be easily implemented into protocols (e.g Modeller).

Amino acid interactions strongly depend physico-chemical properties, distance, geometry and environmental preferences.
Therefore, one has to keep in mind that a contact definition based merely on the distance between the atoms of two residues is a poor gold standard. 
It cannot capture the true interaction preferences of amino acids which are based on their physico-chemical properties and which vary within the vast number of alternative environments within proteins.
  
  


  

### Pitfalls

(ref:caption-pfam) Distribution of PFAM family sizes. Less than half of the families in PFAM (7990 compared to 8489 families) do not have an annotated structure. The median family size in number of sequences for families with and without annotated structures is 185 and 827 respectively. Data taken from PFAM 31.0 (March 2017, 16712 entries).

```{r pfam, echo = FALSE, out.width='100%', fig.cap = '(ref:caption-pfam)'}
knitr::include_url("img/pfam_pdb_notitle.html", height = "650px")
```


Coevolution of residues can be mediated by intermediate molecules (e.g metal ions) and will not always imply spatial proximity in structure [@Morcos2011]. 

Transitivity can lead to correlation signals. 

Phylogenetic bias can also lead to correlations. 

Some of these distant dependencies have been suggested to be caused by homooligomeric interactions [14,22][@Burger2010][@Morcos2011].

Alternative conformations may lead to false positive predictions [@Morcos2011].


Sampling bias needs to be taken into account. See sequence reweighting strategies in methods section

That is, very low entropy columns have on average almost twice as many contacts as high entropy columns[@Burger2010]....reiterates the well-known dependence between surface accessibility and conservation...Obviously, since this position shows no variation whatsoever it cannot display any signs of statistical dependency with any other column, even though it may contact many other residues. This is a basic limitation of using statistical dependency for contact prediction that cannot be avoided

- theoretical upper boundary on precision: simply not all contacts within 8A interact with each other and therefore will never be detected using co-variance methods --> influences sensitivity.
- strongly conserved residues can also not show co-variance signals --> influences sensitivity



#### Correlation vs Causation 

One important shortcoming of covariance approaches arises from the fact that chains of amino acid interactions are very common in protein structures and lead to direct as well as indirect correlation signals [@Lapedes1999] [@Burger2010].  
Many erronous couplings occur due to the correlation versus causation phenomenon. 
Considering three protein residues. 
Residue A interacts with residue B and B interacts with residue C. 
Residue A and C do not physically interact, but there will still be a correlation between A and C.
These transitive effects of covariation need to be disentangled to obtain the true coupling signal.
Traditional covariance methods are unable to distinguish direct and indirect correlations.






## Maximum entropy models for modelling protein families {#maxent}

The principle of maximum entropy was postulated by Jaynes in 1957 [@Jaynes1957a] [@Jaynes1957b], stating that the probability distribution which best represents observed data is the one that is in agreement with measured constraints and has the largest entropy. It is a distribution that makes minimal assumptions and therefore is the least biased estimate of a distribution given the measured constraints.

Applied to the problem of modelling protein families, one would seek a probability distribution over an alignment $\mathbf{X}$ comprising $N$ protein sequences $\seq^{(n)} = (x_i^{(n)},..., x_L^{(n)})$ of length $L$. 
Every position $x_i^{(n)}$ in the [MSA](#abbrev) can take one of 21 states representing the 20 naturally occuring amino acids and a gap ('-') state.
The distribution should reproduce the empirical single $\mathcal{f}(x_i \eq a)$ and pairwise $\mathcal{f}(x_i \eq a, x_j \eq b)$ amino acid frequencies of the alignment:

\begin{equation}
 p(x_i\eq a) = \mathcal{f}(x_i\eq a) = \frac{1}{N}\sum_{n=1}^N I(x_i^{(n)} \eq a) \\
 p(x_i\eq a, x_j \eq b) = \mathcal{f}(x_i\eq a, x_j\eq b) = \frac{1}{N} \sum_{n=1}^N I(x_i^{(n)} \eq a, x_j^{(n)} \eq b)
 (\#eq:emp-freq)
\end{equation}

Introducing Lagrange multipliers to maximize the entropy $S= -\sum_{\seq} p(\seq) \log p(\seq)$ of the distribution subject to the constraints given in eq. \@ref(eq:emp-freq) results in the formulation of an exponential model known as *Potts model* or its generalized form a *Markov Random Field*:

\begin{equation}
p(\seq | \v,\w) = \frac{1}{Z} \exp \left( \sum_{i=1}^L v_i(x_i) \sum_{1 \leq i < j \leq L}^L w_{ij}(x_i, x_j) \right)
(\#eq:max-ent-model)
\end{equation}


$Z$ is a normalization constant also known as *partition function* that ensures the total probabilty adds up to one by summing over all possible assignments to $\seq$. 

\begin{equation}
  Z = \sum_{\seq' \in [1,...,20]^L} p(\seq' | \v, \w)
  (\#eq:partition-fct-likelihood)
\end{equation}


### Properties of model parameters

The Langrange multipliers $\v$ and $\w$, introduced to satisfy the constraints in eq. \@ref(eq:emp-freq), specify the maximum entropy model and need to be tuned.
The single potentials $v_i(a)$ correspond to the single amino acid frequencies whereas the pairwise potentials $w_{ij}(a,b)$, also called couplings, reflect the tendency of an amino acid a at position i to co-occur with an amino acid b at position j in the alignment.


Maximum entropy models belong to the family of exponential models and thus have a unique global minimum. 


Markov Random Field belongs to the class of undirected graphical models. 
Therefore they can be represented as a graph with nodes corresponding to positions in the alignment and edges describing the dependency structure between nodes. 


Can disentangle direct and indirect correlations.

Infer parameters of a maximum entropy model, more specifically a Potts model (statistical physics) aka markov random fiels (computer science).
Likelihood function is convex, but Maximum Likelihood inference of model is infeasible: 
Likelihood function needs to be reevaluated at each iteration during optimization but partition function term sums over 20^L sequences.



### Infering model parameters {#infering-max-ent-models}

Typically, one would obtain parameter estimates by maximizing the log-likelihood function of the parameters over observed sequences in the alignment $\mathbf{X}$: 

\begin{equation}
    \mathcal{L}(\v, \w | \mathbf{X}) = \sum_{n=1}^N \log p(\seq^{(n)}) = \sum_{n=1}^N \left[ \sum_{i=1}^L v_i(x_i^{(n)}) \sum_{1 \leq i < j \leq L}^L w_{ij}(x_i^{(n)}, x_j^{(n)}) - \log Z \right]
\end{equation}

However, optimizing the log-likelihood requires computing the partition function that sums $20^L$ terms. 
Naturally occurig protein domains comprise hundreds of residues. 
Because of this exponential complexity in protein length L it is computationally intractable to evaluate the log-likelihood function at every iteration of an optimization procedure. 

Several approximate solutions have been developed to evade the infeasible computation of the partition function.
In 1999 Lapedes et al. [@Lapedes1999] were the first to apply maximum entropy models to the problem of predicting residue pairs in spatial proximity and thus entangling transitive effects.
They used an iterative Monte Carlo procedure to obtain estimates of the partition function. 

In 2009 Weight et al proposed a message-passing algorithm to approximate the partition function [@Weigt2009].
Eventhough their approach is computationally very expensive and only applicable to small proteins, they obtained remarkable results for the two-component signaling system in bacteria.

Balakrishnan and collegues [@Balakrishnan2011] were the first to apply L1-regularized pseudo-likelihood optimization in 2011. 
Pseudo-likelihood optimizes a different objective and does not aim at approximating the partition function Z.
This approach is described in more detail in \@ref(Pseudo-Likelihood) as it represents the most successfull approach, eventhough it was not taken notice of at that time.

Also in 2011, another approximate solution to the partition fucntion was proposed that uses mean-field expansion [@Morcos2011].
This method is termed mfDCA.
This study resulted in the first 3D models of yet unsolved protein structures being computed with the help of predicted contacts [@CITE]
This method allowed high-throughput predictions as the mean-field approach boils down to inverting the covariance matrix which allows for shorter running times.

A related approach is sparse inverse covariance estimation (PSICOV) method by Jones and colleggues.[@Jones2012] 
They use L1-regularization to invert the correlation matrix and enforce sparsity which is known as grapgical Lasso [@Friedman2008].
Both procedures, mfDCA and PSICOV, assume that model distribution is a multivariate Gaussian. 
it has been shown by Banerjee et al. (2008) that this dual optimization solution also applies to binary data (as is the case in this application). 
In order to represent the [MSA](#abbrev) as continuous distributed, each position is encoded as a 20-dimensional binary vector. 

So far using pseudo-likelihood maximization to infer the model directly from the input alignment has proven to be the most accurate approach. 
There exist several implementations of pseudo-likelihood maximization that vary only in implemenation details, perform similarly and thus are equally popular in the community: CCmpred [@Seemayer2014, ], plmDCA[@Ekeberg2014], GREMLIN [@Kamisetty2013].

### Pseudo-Likelihood {#pseudo-likelihood}

Instead of optimizing the likelihood, Besag suggested in 1975 to rather optimize a function that he termed *pseudo-likelihood* that replaces the joint probability with the product over conditionals [@Besag1975]:

\begin{equation}
  p(\seq | \v,\w) =   \prod_{i=1}^L p(x_i | \seq_{/xi}\v,\w) =  \prod_{i=1}^L \frac{1}{Z_i} \exp \left(  v_i(x_i) \sum_{1 \leq i < j \leq L}^L w_{ij}(x_i, x_j) \right)
\end{equation}

Here, the normalization term $Z_i$ sums only over all assignments to one position $i$ in sequence:

\begin{equation}
  Z_i = \sum_{a=1}^{20} \exp \left( v_i(a) \sum_{1 \leq i < j \leq L}^L w_{ij}(a, x_j) \right)
\end{equation}


The learning objective must include a regularization, as it is overparametrized. 
The indeterminacy can be fixed by including a regularization prior that also prevents overfitting.
Commonly [@Seemayer2014] [@Ekeberg2014], an L2-regularization is used which corresponds to a zero centered Gaussian prior.


\begin{align}
  R(\v, \w)  &= \mathcal{N}(\v | \vec{0}, \lambda_v I^{-1})  + \mathcal{N}(\w | \vec{0}, \lambda_w I^{-1}) \\
             &= \lambda_v ||\v||_2^2 + \lambda_w ||\w||_2^2 
\end{align}


The pseudo-likelihood function retains the concavity of the likelihood.[@Stein2015a]
Moreover, it has been shown that *pseudo-likelihood* is a consistent estimator in the limit of infinite data for pairwise [MRF](#abbrev) [@Besag1975] [@Gidas1988]. 
That is, as the number of sequences in the alignment increases, pseudo-likelihood estimates converge towards the true full likelihood parameters.


the proba- bility of them-th observation, xm, is approximated by the product of the conditional probabilities[@Stein2015a]

Recall from above that our pseudo-likelihood uses the full representation and fixes the gauge by the regularization terms Rl2. Our procedure is therefore first to infer the
interaction parameters using using the pseudo-likelihood and the regularization, and then change to the zero-sum gauge [@Ekeberg2013]


On one hand, this improvement might not be surpris-ing: it is known that, for very large data sets, pseudo- likelihood maximization becomes asymptotically equiva- lent to full maximum-likelihood inference, whereas mean- field inference remains intrinsically approximate, and this may result in an improved PLM performance. On the other hand, the above advantage holds if and only if the following two conditions are fulfilled: data a drawn independently from a probability distribution, and this probability distribution is the the Boltzmann distribution of a Potts model. None of these two con- ditions actually hold for real protein sequences [@Ekeberg2013]


Disregarding the improvements, we find that overall
the predicted contact pairs for plmDCA and mfDCA are highly overlapping, illustrating the robustness of DCA results with respect to the algorithmic implementation[@Ekeberg2013]


### Computing contact map from coupling matrix {#post-processing-heuristics}

- direct information
- frobenius norm

average product correction (also for MI)

(benchmark plot for localmethods + ccmpred)



