<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>PhD thesis: residue-residue contact prediction</title>
  <meta name="description" content="This is my PhD thesis on residue-residue contact prediction.">
  <meta name="generator" content="bookdown 0.3.9 and GitBook 2.6.7">

  <meta property="og:title" content="PhD thesis: residue-residue contact prediction" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is my PhD thesis on residue-residue contact prediction." />
  <meta name="github-repo" content="susannvorberg/phd_thesis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="PhD thesis: residue-residue contact prediction" />
  
  <meta name="twitter:description" content="This is my PhD thesis on residue-residue contact prediction." />
  

<meta name="author" content="Susann Vorberg">


<meta name="date" content="2017-10-26">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="interpretation-of-coupling-matrices.html">
<link rel="next" href="training-of-the-random-forest-contact-prior.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />








<script type="text/x-mathjax-config">
 MathJax.Hub.Config({
  tex2jax: {inlineMath: [["$","$"],["\\(","\\)"]]},
  TeX: { 
    extensions: ["mediawiki-texvc.js", "sinuitx.js"],
    Macros: {
      Cb: "C_\\beta",
      eq: "\\!=\\!",
      Gauss: "\\mathcal{N}",
      H: "\\mathbf{H}",
      Hij : "\\H_{ij}",
      I: "\\mathbf{I}",
      Lijk: "\\mathbf{\\Lambda}_{ij,k}",
      Lk: "\\mathbf{\\Lambda}_k",
      LL: "L\\!L(\\mathbf{v}, \\mathbf{w})",
      LLreg: "L\\!L_\\mathrm{reg}",
      muijk: "\\mathbf{\\mu}_{ij,k}",
      muk: "\\mathbf{\\mu}_k",
      neff: "N_\\mathrm{eff}",
      r: "\\mathbf{r}",
      rij: "r_{ij}",
      c: "\\mathbf{c}",
      cij: "c_{ij}",
      seq: "\\mathbf{x}",
      Qij: "\\mathbf{Q}_{ij}",
      q: "\\mathbf{q}",
      qij: "\\mathbf{q'}_{ij}",
      Sn: "\\mathcal{S}_n",
      v: "\\mathbf{v}",
      vi: "\\mathcal{v}_{i}",
      vj: "\\mathcal{v}_{j}",
      via: "\\mathcal{v}_{ia}",
      vja: "\\mathcal{v}_{ja}",
      w: "\\mathbf{w}",
      wij: "\\mathbf{w}_{ij}",
      wijab: "\\mathcal{w}_{ijab}",
      wijcd: "\\mathcal{w}_{ijcd}",
      wklcd: "\\mathcal{w}_{klcd}",
      X: "\\mathbf{X}",
      angstrom: "\\AA \\; \\;"
      }
  }
});
</script>


 
<!--
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  extensions: ["tex2jax.js", "[siunitx]/siunitx.js"],
  TeX: { TagSide: "left" }
});
MathJax.Ajax.config.path['siunitx']  = '../latex/MathJax-siunitx-master/';
</script>
//-->



<link rel="stylesheet" href="css/style.css" type="text/css" />
<link rel="stylesheet" href="css/toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PhD Thesis Susann Vorberg</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Summary</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="1" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>1</b> Background</a><ul>
<li class="chapter" data-level="1.1" data-path="general-intro.html"><a href="general-intro.html"><i class="fa fa-check"></i><b>1.1</b> Biological Background</a></li>
<li class="chapter" data-level="1.2" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html"><i class="fa fa-check"></i><b>1.2</b> Introduction to Contact Prediction</a><ul>
<li class="chapter" data-level="1.2.1" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html#local-methods"><i class="fa fa-check"></i><b>1.2.1</b> Local Statistical Models</a></li>
<li class="chapter" data-level="1.2.2" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html#global-methods"><i class="fa fa-check"></i><b>1.2.2</b> Global Statistical Models</a></li>
<li class="chapter" data-level="1.2.3" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html#meta-predictors"><i class="fa fa-check"></i><b>1.2.3</b> Machine Learning Methods and Meta-Predictors</a></li>
<li class="chapter" data-level="1.2.4" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html#maxent"><i class="fa fa-check"></i><b>1.2.4</b> Modelling Protein Families with Potts Model</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="application-contact-prediction.html"><a href="application-contact-prediction.html"><i class="fa fa-check"></i><b>1.3</b> Applications</a></li>
<li class="chapter" data-level="1.4" data-path="intro-cp-evaluation.html"><a href="intro-cp-evaluation.html"><i class="fa fa-check"></i><b>1.4</b> Evaluating Contact Prediction Methods</a><ul>
<li class="chapter" data-level="1.4.1" data-path="intro-cp-evaluation.html"><a href="intro-cp-evaluation.html#seq-sep"><i class="fa fa-check"></i><b>1.4.1</b> Sequence Separation</a></li>
<li class="chapter" data-level="1.4.2" data-path="intro-cp-evaluation.html"><a href="intro-cp-evaluation.html#interpretation-of-evaluation-results"><i class="fa fa-check"></i><b>1.4.2</b> Interpretation of Evaluation Results</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="challenges.html"><a href="challenges.html"><i class="fa fa-check"></i><b>1.5</b> Challenges for Coevolutionary Inference</a><ul>
<li class="chapter" data-level="1.5.1" data-path="challenges.html"><a href="challenges.html#phylogenetic-noise"><i class="fa fa-check"></i><b>1.5.1</b> Phylogenetic Effects as a Source of Noise</a></li>
<li class="chapter" data-level="1.5.2" data-path="challenges.html"><a href="challenges.html#entropic-effects-as-a-source-of-noise"><i class="fa fa-check"></i><b>1.5.2</b> Entropic Effects as a Source of Noise</a></li>
<li class="chapter" data-level="1.5.3" data-path="challenges.html"><a href="challenges.html#finite-sampling-effects"><i class="fa fa-check"></i><b>1.5.3</b> Finite Sampling Effects</a></li>
<li class="chapter" data-level="1.5.4" data-path="challenges.html"><a href="challenges.html#multiple-sequence-alignments"><i class="fa fa-check"></i><b>1.5.4</b> Multiple Sequence Alignments</a></li>
<li class="chapter" data-level="1.5.5" data-path="challenges.html"><a href="challenges.html#alternative-sources-of-coevolution"><i class="fa fa-check"></i><b>1.5.5</b> Alternative Sources of Coevolution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="interpreting-coupling-matrices.html"><a href="interpreting-coupling-matrices.html"><i class="fa fa-check"></i><b>2</b> Interpretation of Coupling Matrices</a><ul>
<li class="chapter" data-level="2.1" data-path="correlation-between-couplings-and-class.html"><a href="correlation-between-couplings-and-class.html"><i class="fa fa-check"></i><b>2.1</b> Single Coupling Values Carry Evidence of Contacts</a></li>
<li class="chapter" data-level="2.2" data-path="coupling-profiles.html"><a href="coupling-profiles.html"><i class="fa fa-check"></i><b>2.2</b> Coupling Profiles Vary with Distance</a></li>
<li class="chapter" data-level="2.3" data-path="physico-chemical-fingerprints-in-coupling-matrices.html"><a href="physico-chemical-fingerprints-in-coupling-matrices.html"><i class="fa fa-check"></i><b>2.3</b> Physico-Chemical Fingerprints in Coupling Matrices</a></li>
<li class="chapter" data-level="2.4" data-path="higher-order-coupling-profiles.html"><a href="higher-order-coupling-profiles.html"><i class="fa fa-check"></i><b>2.4</b> Higher Order Dependencies Between Couplings</a></li>
<li class="chapter" data-level="2.5" data-path="discussion.html"><a href="discussion.html"><i class="fa fa-check"></i><b>2.5</b> Discussion</a></li>
<li class="chapter" data-level="2.6" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>2.6</b> Methods</a><ul>
<li class="chapter" data-level="2.6.1" data-path="methods.html"><a href="methods.html#dataset"><i class="fa fa-check"></i><b>2.6.1</b> Dataset</a></li>
<li class="chapter" data-level="2.6.2" data-path="methods.html"><a href="methods.html#diff-ccmpred-ccmpredpy"><i class="fa fa-check"></i><b>2.6.2</b> Computing Pseudo-Likelihood Couplings</a></li>
<li class="chapter" data-level="2.6.3" data-path="methods.html"><a href="methods.html#seq-reweighting"><i class="fa fa-check"></i><b>2.6.3</b> Sequence Reweighting</a></li>
<li class="chapter" data-level="2.6.4" data-path="methods.html"><a href="methods.html#amino-acid-frequencies"><i class="fa fa-check"></i><b>2.6.4</b> Computing Amino Acid Frequencies</a></li>
<li class="chapter" data-level="2.6.5" data-path="methods.html"><a href="methods.html#methods-regularization"><i class="fa fa-check"></i><b>2.6.5</b> Regularization</a></li>
<li class="chapter" data-level="2.6.6" data-path="methods.html"><a href="methods.html#method-coupling-correlation"><i class="fa fa-check"></i><b>2.6.6</b> Correlation of Couplings with Contact Class</a></li>
<li class="chapter" data-level="2.6.7" data-path="methods.html"><a href="methods.html#method-coupling-profile"><i class="fa fa-check"></i><b>2.6.7</b> Coupling Distribution Plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="optimizing-full-likelihood.html"><a href="optimizing-full-likelihood.html"><i class="fa fa-check"></i><b>3</b> Optimizing the Full Likelihood</a><ul>
<li class="chapter" data-level="3.1" data-path="full-likelihood-gradient.html"><a href="full-likelihood-gradient.html"><i class="fa fa-check"></i><b>3.1</b> Approximating the Gradient of the Full Likelihood with Contrastive Divergence</a></li>
<li class="chapter" data-level="3.2" data-path="full-likelihood-optimization.html"><a href="full-likelihood-optimization.html"><i class="fa fa-check"></i><b>3.2</b> Optimizing the Full Likelihood</a><ul>
<li class="chapter" data-level="3.2.1" data-path="full-likelihood-optimization.html"><a href="full-likelihood-optimization.html#convergence-criteria-sgd"><i class="fa fa-check"></i><b>3.2.1</b> Convergence Criterion for Stochastic Gradient Descent</a></li>
<li class="chapter" data-level="3.2.2" data-path="full-likelihood-optimization.html"><a href="full-likelihood-optimization.html#sgd-hyperparameter-tuning"><i class="fa fa-check"></i><b>3.2.2</b> Tuning Hyperparameters of Stochastic Gradient Descent Optimizer</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html"><i class="fa fa-check"></i><b>3.3</b> Tuning the Gibbs Sampling Scheme for Contrastive Divergence</a><ul>
<li class="chapter" data-level="3.3.1" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html#regularization-for-cd-with-sgd"><i class="fa fa-check"></i><b>3.3.1</b> Tuning Regularization Coefficients for Contrastive Divergence</a></li>
<li class="chapter" data-level="3.3.2" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html#cd-sampling-size"><i class="fa fa-check"></i><b>3.3.2</b> Varying the Sample Size</a></li>
<li class="chapter" data-level="3.3.3" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html#cd-gibbs-steps"><i class="fa fa-check"></i><b>3.3.3</b> Varying the number of Gibbs Steps</a></li>
<li class="chapter" data-level="3.3.4" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html#cd-gibbs-steps"><i class="fa fa-check"></i><b>3.3.4</b> Persistent Contrastive Divergence</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="adam-results.html"><a href="adam-results.html"><i class="fa fa-check"></i><b>3.4</b> Using ADAM to optimize Contrastive Divergence</a></li>
<li class="chapter" data-level="3.5" data-path="comparing-pll-cd.html"><a href="comparing-pll-cd.html"><i class="fa fa-check"></i><b>3.5</b> Comparing CD couplings to pLL couplings</a><ul>
<li class="chapter" data-level="3.5.1" data-path="comparing-pll-cd.html"><a href="comparing-pll-cd.html#protein-1c75a00"><i class="fa fa-check"></i><b>3.5.1</b> Protein 1c75A00</a></li>
<li class="chapter" data-level="3.5.2" data-path="comparing-pll-cd.html"><a href="comparing-pll-cd.html#protein-1ss3a00-and-1c55a00"><i class="fa fa-check"></i><b>3.5.2</b> Protein 1ss3A00 and 1c55A00</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="discussion-1.html"><a href="discussion-1.html"><i class="fa fa-check"></i><b>3.6</b> Discussion</a></li>
<li class="chapter" data-level="3.7" data-path="methods-1.html"><a href="methods-1.html"><i class="fa fa-check"></i><b>3.7</b> Methods</a><ul>
<li class="chapter" data-level="3.7.1" data-path="methods-1.html"><a href="methods-1.html#potts-full-likelihood"><i class="fa fa-check"></i><b>3.7.1</b> The Potts Model</a></li>
<li class="chapter" data-level="3.7.2" data-path="methods-1.html"><a href="methods-1.html#gap-treatment"><i class="fa fa-check"></i><b>3.7.2</b> Treating Gaps as Missing Information</a></li>
<li class="chapter" data-level="3.7.3" data-path="methods-1.html"><a href="methods-1.html#the-regularized-full-log-likelihood-and-its-gradient-with-gap-treatment"><i class="fa fa-check"></i><b>3.7.3</b> The Regularized Full Log Likelihood and its Gradient With Gap Treatment</a></li>
<li class="chapter" data-level="3.7.4" data-path="methods-1.html"><a href="methods-1.html#prior-v"><i class="fa fa-check"></i><b>3.7.4</b> The prior on single potentials</a></li>
<li class="chapter" data-level="3.7.5" data-path="methods-1.html"><a href="methods-1.html#methods-sgd"><i class="fa fa-check"></i><b>3.7.5</b> Stochastic Gradien Descent</a></li>
<li class="chapter" data-level="3.7.6" data-path="methods-1.html"><a href="methods-1.html#methods-cd-sampling"><i class="fa fa-check"></i><b>3.7.6</b> Computing the Gradient with Contrastive Divergence</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="contact-prior.html"><a href="contact-prior.html"><i class="fa fa-check"></i><b>4</b> Random Forest Contact Prior</a><ul>
<li class="chapter" data-level="4.1" data-path="random-forest-classifiers.html"><a href="random-forest-classifiers.html"><i class="fa fa-check"></i><b>4.1</b> Random Forest Classifiers</a></li>
<li class="chapter" data-level="4.2" data-path="hyperparameter-optimization-for-random-forest.html"><a href="hyperparameter-optimization-for-random-forest.html"><i class="fa fa-check"></i><b>4.2</b> Hyperparameter Optimization for Random Forest</a></li>
<li class="chapter" data-level="4.3" data-path="evaluating-random-forest-model-as-contact-predictor.html"><a href="evaluating-random-forest-model-as-contact-predictor.html"><i class="fa fa-check"></i><b>4.3</b> Evaluating Random Forest Model as Contact Predictor</a></li>
<li class="chapter" data-level="4.4" data-path="using-contact-scores-as-additional-features.html"><a href="using-contact-scores-as-additional-features.html"><i class="fa fa-check"></i><b>4.4</b> Using Contact Scores as Additional Features</a></li>
<li class="chapter" data-level="4.5" data-path="methods-2.html"><a href="methods-2.html"><i class="fa fa-check"></i><b>4.5</b> Methods</a><ul>
<li class="chapter" data-level="4.5.1" data-path="methods-2.html"><a href="methods-2.html#seq-features"><i class="fa fa-check"></i><b>4.5.1</b> Features used to train Random Forest Model</a></li>
<li class="chapter" data-level="4.5.2" data-path="methods-2.html"><a href="methods-2.html#simple-contact-prior-with-respect-to-protein-length"><i class="fa fa-check"></i><b>4.5.2</b> Simple Contact Prior with Respect to Protein Length</a></li>
<li class="chapter" data-level="4.5.3" data-path="methods-2.html"><a href="methods-2.html#rf-training"><i class="fa fa-check"></i><b>4.5.3</b> Cross-validation for Random Forest Training</a></li>
<li class="chapter" data-level="4.5.4" data-path="methods-2.html"><a href="methods-2.html#rf-feature-selection"><i class="fa fa-check"></i><b>4.5.4</b> Feature Selection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="bayesian-approach.html"><a href="bayesian-approach.html"><i class="fa fa-check"></i><b>5</b> A Bayesian Statistical Model for Residue-Residue Contact Prediction</a><ul>
<li class="chapter" data-level="5.1" data-path="overview-posterior-distances.html"><a href="overview-posterior-distances.html"><i class="fa fa-check"></i><b>5.1</b> Computing the Posterior Probabiilty of a Contact</a></li>
<li class="chapter" data-level="5.2" data-path="coupling-prior.html"><a href="coupling-prior.html"><i class="fa fa-check"></i><b>5.2</b> Modelling the Prior Over Couplings Depending on Contact States</a></li>
<li class="chapter" data-level="5.3" data-path="bayesian-model-training-hyperparameters.html"><a href="bayesian-model-training-hyperparameters.html"><i class="fa fa-check"></i><b>5.3</b> Training the Hyperparameters in the Likelihood Function of Contact States</a></li>
<li class="chapter" data-level="5.4" data-path="training-hyperparameters-for-a-gaussian-mixture-with-three-components.html"><a href="training-hyperparameters-for-a-gaussian-mixture-with-three-components.html"><i class="fa fa-check"></i><b>5.4</b> Training Hyperparameters for a Gaussian Mixture with Three Components</a></li>
<li class="chapter" data-level="5.5" data-path="training-hyperparameters-for-a-gaussian-mixture-with-five-components.html"><a href="training-hyperparameters-for-a-gaussian-mixture-with-five-components.html"><i class="fa fa-check"></i><b>5.5</b> Training Hyperparameters for a Gaussian Mixture with Five Components</a></li>
<li class="chapter" data-level="5.6" data-path="training-hyperparameters-for-a-gaussian-mixture-with-ten-components.html"><a href="training-hyperparameters-for-a-gaussian-mixture-with-ten-components.html"><i class="fa fa-check"></i><b>5.6</b> Training Hyperparameters for a Gaussian Mixture with Ten Components</a></li>
<li class="chapter" data-level="5.7" data-path="posterior-of-rij.html"><a href="posterior-of-rij.html"><i class="fa fa-check"></i><b>5.7</b> Computing The Posterior Probability of Contacts</a></li>
<li class="chapter" data-level="5.8" data-path="analysing-contact-maps.html"><a href="analysing-contact-maps.html"><i class="fa fa-check"></i><b>5.8</b> Analysing Contact Maps</a></li>
<li class="chapter" data-level="5.9" data-path="discussion-2.html"><a href="discussion-2.html"><i class="fa fa-check"></i><b>5.9</b> Discussion</a></li>
<li class="chapter" data-level="5.10" data-path="methods-3.html"><a href="methods-3.html"><i class="fa fa-check"></i><b>5.10</b> Methods</a><ul>
<li class="chapter" data-level="5.10.1" data-path="methods-3.html"><a href="methods-3.html#methods-coupling-prior"><i class="fa fa-check"></i><b>5.10.1</b> Modelling the Prior Over Couplings Depending on Contact States</a></li>
<li class="chapter" data-level="5.10.2" data-path="methods-3.html"><a href="methods-3.html#laplace-approx"><i class="fa fa-check"></i><b>5.10.2</b> Gaussian Approximation to the Posterior of Couplings</a></li>
<li class="chapter" data-level="5.10.3" data-path="methods-3.html"><a href="methods-3.html#likelihood-fct-distances"><i class="fa fa-check"></i><b>5.10.3</b> Integrating out the Hidden Variables to Obtain the Likelihood Function of the Contact States</a></li>
<li class="chapter" data-level="5.10.4" data-path="methods-3.html"><a href="methods-3.html#Hessian-offdiagonal"><i class="fa fa-check"></i><b>5.10.4</b> The Hessian off-diagonal Elements Carry a Negligible Signal</a></li>
<li class="chapter" data-level="5.10.5" data-path="methods-3.html"><a href="methods-3.html#neg-Hessian-computation"><i class="fa fa-check"></i><b>5.10.5</b> Efficiently Computing the negative Hessian of the regularized log-likelihood</a></li>
<li class="chapter" data-level="5.10.6" data-path="methods-3.html"><a href="methods-3.html#inv-lambda-ij-k"><i class="fa fa-check"></i><b>5.10.6</b> Efficiently Computing the Inverse of Matrix <span class="math inline">\(\Lijk\)</span></a></li>
<li class="chapter" data-level="5.10.7" data-path="methods-3.html"><a href="methods-3.html#gradient-muk"><i class="fa fa-check"></i><b>5.10.7</b> The gradient of the log likelihood with respect to <span class="math inline">\(\muk\)</span></a></li>
<li class="chapter" data-level="5.10.8" data-path="methods-3.html"><a href="methods-3.html#gradient-lambdak"><i class="fa fa-check"></i><b>5.10.8</b> The gradient of the log likelihood with respect to <span class="math inline">\(\Lk\)</span></a></li>
<li class="chapter" data-level="5.10.9" data-path="methods-3.html"><a href="methods-3.html#the-gradient-of-the-log-likelihood-with-respect-to-gamma_k"><i class="fa fa-check"></i><b>5.10.9</b> The gradient of the log likelihood with respect to <span class="math inline">\(\gamma_k\)</span></a></li>
<li class="chapter" data-level="5.10.10" data-path="methods-3.html"><a href="methods-3.html#bayesian-model-distances"><i class="fa fa-check"></i><b>5.10.10</b> Extending the Bayesian Statistical Model for the Prediction of Protein Residue-Residue Distances</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="practical-methods.html"><a href="practical-methods.html"><i class="fa fa-check"></i><b>5.11</b> Practical Methods</a><ul>
<li class="chapter" data-level="5.11.1" data-path="practical-methods.html"><a href="practical-methods.html#training-hyperparameters-bayesian-model"><i class="fa fa-check"></i><b>5.11.1</b> Training the Hyperparameters in the Likelihood Function of Contact States</a></li>
<li class="chapter" data-level="5.11.2" data-path="practical-methods.html"><a href="practical-methods.html#dataset-training-bayesian-model"><i class="fa fa-check"></i><b>5.11.2</b> Dataset Specifications</a></li>
<li class="chapter" data-level="5.11.3" data-path="practical-methods.html"><a href="practical-methods.html#model-specifications-training-bayesian-model"><i class="fa fa-check"></i><b>5.11.3</b> Model Specifications</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion-and-outlook.html"><a href="conclusion-and-outlook.html"><i class="fa fa-check"></i><b>6</b> Conclusion and Outlook</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="abbrev.html"><a href="abbrev.html"><i class="fa fa-check"></i><b>A</b> Abbreviations</a></li>
<li class="chapter" data-level="B" data-path="amino-acids.html"><a href="amino-acids.html"><i class="fa fa-check"></i><b>B</b> Amino Acid Alphabet</a></li>
<li class="chapter" data-level="C" data-path="dataset-properties.html"><a href="dataset-properties.html"><i class="fa fa-check"></i><b>C</b> Dataset Properties</a></li>
<li class="chapter" data-level="D" data-path="interpretation-of-coupling-matrices.html"><a href="interpretation-of-coupling-matrices.html"><i class="fa fa-check"></i><b>D</b> Interpretation of Coupling Matrices</a></li>
<li class="chapter" data-level="E" data-path="optimizing-full-likelihood-with-gradient-descent.html"><a href="optimizing-full-likelihood-with-gradient-descent.html"><i class="fa fa-check"></i><b>E</b> Optimizing Full Likelihood with Gradient Descent</a></li>
<li class="chapter" data-level="F" data-path="training-of-the-random-forest-contact-prior.html"><a href="training-of-the-random-forest-contact-prior.html"><i class="fa fa-check"></i><b>F</b> Training of the Random Forest Contact Prior</a></li>
<li class="chapter" data-level="G" data-path="bayesian-statistical-model-for-contact-prediction.html"><a href="bayesian-statistical-model-for-contact-prediction.html"><i class="fa fa-check"></i><b>G</b> Bayesian statistical model for contact prediction</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PhD thesis: residue-residue contact prediction</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="optimizing-full-likelihood-with-gradient-descent" class="section level1">
<h1><span class="header-section-number">E</span> Optimizing Full Likelihood with Gradient Descent</h1>

<div class="figure" style="text-align: center"><span id="fig:performance-cd-alphaopt-0"></span>
<iframe src="img/full_likelihood/appendix/precision_vs_rank_learningrates_comparealpha0.html" width="85%" height="500px">
</iframe>
<p class="caption">
Figure E.1: Mean precision for top ranked contact predictions over 300 proteins. Contact scores are computed as the <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the couplings <span class="math inline">\(\wij\)</span>. <strong>pseudo-likelihood</strong>: couplings computed with pseudo-likelihood. <strong>CD alpha0 = 5e-4</strong>: couplings computed with <a href="abbrev.html#abbrev">CD</a> using stochastic gradient descent with initial learning rate, <span class="math inline">\(\alpha_0 \eq 5e-4\)</span>. <strong>CD alpha0 = 1e-3</strong>: couplings computed with <a href="abbrev.html#abbrev">CD</a> using stochastic gradient descent with initial learning rate, <span class="math inline">\(\alpha_0 \eq 1e-3\)</span>. <strong>CD alpha0 = 5e-2Neff^-0.5</strong>: couplings computed with <a href="abbrev.html#abbrev">CD</a> using stochastic gradient descent with initial learning rate defined as a function of <a href="abbrev.html#abbrev">Neff</a>, <span class="math inline">\(\alpha_0 \eq \frac{5e-2}{\sqrt{N_{\text{eff}}}}\)</span>.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:learning-rate-schedules"></span>
<iframe src="img/full_likelihood/appendix/learning_rate_schedules_alpha0_1e-4_notitle.html" width="100%" height="500px">
</iframe>
<p class="caption">
Figure E.2: Value of learning rate against the number of iterations for different learning rate schedules. Red legend group represents the <strong>exponential</strong> learning rate schedule <span class="math inline">\(\alpha_{t+1} = \alpha_0 \cdot\exp(- \gamma t)\)</span>. Blue legend group represents the <strong>linear</strong> learning rate schedule <span class="math inline">\(\alpha = \alpha_0 / (1 + \gamma \cdot t)\)</span>. Green legend group represents the <strong>sigmoidal</strong> learning rate schedule <span class="math inline">\(\alpha_{t+1} = \alpha_{t} / (1 + \gamma \cdot t)\)</span>. Purple legend group represents the <strong>square root</strong> learning rate schedule <span class="math inline">\(\alpha = \alpha_0 / \sqrt{1 + \gamma \cdot t}\)</span>. The initial learning rate <span class="math inline">\(\alpha_0\)</span> is set to 1e-4, the iteration number is given by <span class="math inline">\(t\)</span> and <span class="math inline">\(\gamma\)</span> is the decay rate and its value is given in brackets in the legend.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:performance-cd-linschedule"></span>
<iframe src="img/full_likelihood/appendix/precision_vs_rank_alpha0_0_lin_decayrate.html" width="85%" height="500px">
</iframe>
<p class="caption">
Figure E.3: Mean precision for top ranked contact predictions over 300 proteins. Contact scores are computed as the <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the couplings <span class="math inline">\(\wij\)</span>. pseudo-likelihood: contact scores computed from pseudo-likelihood. The other methods derive contact scores from couplings computed from <a href="abbrev.html#abbrev">CD</a> using stochastic gradient descent with an initial learning rate defined with respect to <a href="abbrev.html#abbrev">Neff</a> and a <em>linear</em> learning rate annealing schedule <span class="math inline">\(\alpha = \frac{\alpha_0}{1 + \gamma t}\)</span> with decay rate <span class="math inline">\(\gamma\)</span> as specified in the legend.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:performance-cd-sigschedule"></span>
<iframe src="img/full_likelihood/appendix/precision_vs_rank_alpha0_0_sig_decayrate.html" width="85%" height="500px">
</iframe>
<p class="caption">
Figure E.4: Mean precision for top ranked contact predictions over 300 proteins. Contact scores are computed as the <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the couplings <span class="math inline">\(\wij\)</span>. pseudo-likelihood: contact scores computed from pseudo-likelihood. The other methods derive contact scores from couplings computed from <a href="abbrev.html#abbrev">CD</a> using stochastic gradient descent with an initial learning rate defined with respect to <a href="abbrev.html#abbrev">Neff</a> and a <em>sigmoidal</em> learning rate annealing schedule <span class="math inline">\(\alpha_{t+1} = \frac{\alpha_{t}}{1 + \gamma t}\)</span> with t being the iteration number and decay rate <span class="math inline">\(\gamma\)</span> as specified in the legend.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:performance-cd-sqrtschedule"></span>
<iframe src="img/full_likelihood/appendix/precision_vs_rank_alpha0_0_sqrt_decayrate.html" width="85%" height="500px">
</iframe>
<p class="caption">
Figure E.5: Mean precision for top ranked contact predictions over 300 proteins. Contact scores are computed as the <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the couplings <span class="math inline">\(\wij\)</span>. pseudo-likelihood: contact scores computed from pseudo-likelihood. The other methods derive contact scores from couplings computed from <a href="abbrev.html#abbrev">CD</a> using stochastic gradient descent with an initial learning rate defined with respect to <a href="abbrev.html#abbrev">Neff</a> and a <em>square root</em> learning rate annealing schedule <span class="math inline">\(\alpha = \frac{\alpha_0}{\sqrt{1 + \gamma t}}\)</span> with t being the iteration number and decay rate <span class="math inline">\(\gamma\)</span> as specified in the legend.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:performance-cd-expschedule"></span>
<iframe src="img/full_likelihood/appendix/precision_vs_rank_alpha0_0_exp_decayrate.html" width="85%" height="500px">
</iframe>
<p class="caption">
Figure E.6: Mean precision for top ranked contact predictions over 300 proteins. Contact scores are computed as the <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the couplings <span class="math inline">\(\wij\)</span>. pseudo-likelihood: contact scores computed from pseudo-likelihood. The other methods derive contact scores from couplings computed from <a href="abbrev.html#abbrev">CD</a> using stochastic gradient descent with an initial learning rate defined with respect to <a href="abbrev.html#abbrev">Neff</a> and a <em>exponential</em> learning rate annealing schedule <span class="math inline">\(\alpha = \alpha_0 \cdot\exp(- \gamma t)\)</span> with t being the iteration number and decay rate <span class="math inline">\(\gamma\)</span> as specified in the legend.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:cd-numit-lin-learning-rate-schedule"></span>
<iframe src="img/full_likelihood/appendix/distribution_numiterations_against_linear_learningrate_schedule.html" width="90%" height="500px">
</iframe>
<p class="caption">
Figure E.7: Distribution of the number of iterations until convergence for stochastic gradient descent optimizations of the full likelihood using different decay rates with a <strong>linear</strong> learning rate schedule <span class="math inline">\(\alpha = \alpha_0 / (1 + \gamma t)\)</span> with <span class="math inline">\(t\)</span> being the iteration number and the decay rate <span class="math inline">\(\gamma\)</span> is specified in the legend. Initial learning rate <span class="math inline">\(\alpha_0\)</span> defined with respect to <a href="abbrev.html#abbrev">Neff</a> and maximum number of iterations is set to 5000.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:cd-numit-sig-learning-rate-schedule"></span>
<iframe src="img/full_likelihood/appendix/distribution_numiterations_against_sigmoidal_learningrate_schedule.html" width="90%" height="500px">
</iframe>
<p class="caption">
Figure E.8: Distribution of the number of iterations until convergence for stochastic gradient descent optimizations of the full likelihood using different decay rates with a <strong>sigmoidal</strong> learning rate schedule <span class="math inline">\(\alpha_{t+1} = \alpha_{t} / (1 + \gamma t)\)</span> with <span class="math inline">\(t\)</span> being the iteration number and the decay rate <span class="math inline">\(\gamma\)</span> is specified in the legend. Initial learning rate <span class="math inline">\(\alpha_0\)</span> defined with respect to <a href="abbrev.html#abbrev">Neff</a> and maximum number of iterations is set to 5000.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:cd-numit-sqrt-learning-rate-schedule"></span>
<iframe src="img/full_likelihood/appendix/distribution_numiterations_against_sqrt_learningrate_schedule.html" width="90%" height="500px">
</iframe>
<p class="caption">
Figure E.9: Distribution of the number of iterations until convergence for stochastic gradient descent optimizations of the full likelihood using different decay rates with a <strong>square root</strong> learning rate schedule <span class="math inline">\(\alpha = \alpha_0 / \sqrt{1 + \gamma t}\)</span> with <span class="math inline">\(t\)</span> being the iteration number and the decay rate <span class="math inline">\(\gamma\)</span> is specified in the legend. Initial learning rate <span class="math inline">\(\alpha_0\)</span> defined with respect to <a href="abbrev.html#abbrev">Neff</a> and maximum number of iterations is set to 5000.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:cd-numit-exp-learning-rate-schedule"></span>
<iframe src="img/full_likelihood/appendix/distribution_numiterations_against_exponential_learningrate_schedule.html" width="90%" height="500px">
</iframe>
<p class="caption">
Figure E.10: Distribution of the number of iterations until convergence for stochastic gradient descent optimizations of the full likelihood using different decay rates with an <strong>exponential</strong> learning rate schedule <span class="math inline">\(\alpha = \alpha_0 \cdot\exp(- \gamma t)\)</span> with <span class="math inline">\(t\)</span> being the iteration number and the decay rate <span class="math inline">\(\gamma\)</span> is specified in the legend. Initial learning rate <span class="math inline">\(\alpha_0\)</span> defined with respect to <a href="abbrev.html#abbrev">Neff</a> and maximum number of iterations is set to 5000.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:numit-convergence-convergence-prev"></span>
<iframe src="img/full_likelihood/appendix/distribution_numiterations_against_convergence_prev.html" width="85%" height="500px">
</iframe>
<p class="caption">
Figure E.11: Distribution of the number of iterations until convergence for gradient descent optimizations of the full likelihood. The relative change of the L2 norm over coupling parameters, <span class="math inline">\(||\w||_2\)</span>, is evaluated over a defined number of previous iterations and is specified in the legend. Convergence is assumed when the relative change falls below a small value <span class="math inline">\(\epsilon \eq 1e-8\)</span>. The optimal hyperparameters settings for <a href="abbrev.html#abbrev">SGD</a> as described in section <a href="full-likelihood-optimization.html#sgd-hyperparameter-tuning">3.2.2</a> have been used.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:full-likelihood-opt-fixv"></span>
<iframe src="img/full_likelihood/appendix/precision_vs_rank_fixed_v.html" width="90%" height="500px">
</iframe>
<p class="caption">
Figure E.12: Mean precision for top ranked contact predictions over 300 proteins. Contact scores are computed as the <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the couplings <span class="math inline">\(\wij\)</span>. <a href="abbrev.html#abbrev">SGD</a> settings for <a href="abbrev.html#abbrev">CD</a> optimization are as follows: sigmoidal learning rate schedule with decay rate <span class="math inline">\(\gamma = \mathrm{5e-6}\)</span> and initial learning rate <span class="math inline">\(\alpha_0 \eq \mathrm{5e-2}/N_{\textrm{eff}}\)</span>. <strong>pseudo-likelihood</strong>: contact scores computed from pseudo-likelihood. <strong>CD fixed v at v* </strong>: contact scores computed from <a href="abbrev.html#abbrev">CD</a> with <a href="abbrev.html#abbrev">SGD</a> and single potentials <span class="math inline">\(\v\)</span> are not optimized but fixed at <span class="math inline">\(\v^*\)</span> as given in eq. <a href="methods-1.html#eq:prior-v">(3.17)</a>. <strong>CD lambda_v = 10</strong>: contact scores computed from <a href="abbrev.html#abbrev">CD</a> with <a href="abbrev.html#abbrev">SGD</a> and single potentials <span class="math inline">\(\v\)</span> are subject to optimization using L2-reglarization with <span class="math inline">\(\lambda_v \eq 10\)</span>.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:cd-samplesize-protein1ahoa00"></span>
<img src="img/full_likelihood/appendix/1ahoa00_parameter_gradient_norm_for_samplesizes.png" alt="Monitoring parameter norm and gradient norm for protein 1ahoA00 during SGD optimization of CD using different sample sizes. Protein 1ahoA00 has length L=64 and 378 sequences in the alignment (Neff=229). The number of sequences, that is used for Gibbs sampling to approximate the gradient, is given in the legend with 1L = 64 sequences, 5L = 320 sequences, 10L = min(10L, N) = 378 sequences, 0.2Neff = 46 sequences, 0.3Neff = 69 sequences, 0.4Neff = 92 sequences. Left L2-norm of the gradients for coupling parameters, \(||\nabla_{\w} L\!L(\v^*, \w)||_2\) (without contribution of regularizer). Right L2-norm of the coupling parameters \(||\w||_2\)." width="100%" />
<p class="caption">
Figure E.13: Monitoring parameter norm and gradient norm for protein 1ahoA00 during <a href="abbrev.html#abbrev">SGD</a> optimization of <a href="abbrev.html#abbrev">CD</a> using different sample sizes. Protein 1ahoA00 has length L=64 and 378 sequences in the alignment (<a href="abbrev.html#abbrev">Neff</a>=229). The number of sequences, that is used for Gibbs sampling to approximate the gradient, is given in the legend with 1L = 64 sequences, 5L = 320 sequences, 10L = min(10L, N) = 378 sequences, 0.2Neff = 46 sequences, 0.3Neff = 69 sequences, 0.4Neff = 92 sequences. <strong>Left</strong> L2-norm of the gradients for coupling parameters, <span class="math inline">\(||\nabla_{\w} L\!L(\v^*, \w)||_2\)</span> (without contribution of regularizer). <strong>Right</strong> L2-norm of the coupling parameters <span class="math inline">\(||\w||_2\)</span>.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:cd-precision-gibbssteps-neff"></span>
<iframe src="img/full_likelihood/appendix/precision_vs_rank_facetted_by_neff_gibbssteps.html" width="100%" height="600px">
</iframe>
<p class="caption">
Figure E.14: Mean precision for top ranked contact predictions over 300 proteins splitted into four equally sized subsets with respect to <a href="abbrev.html#abbrev">Neff</a>. Contact scores are computed as the <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the couplings <span class="math inline">\(\wij\)</span>. Subsets are defined according to quantiles of <a href="abbrev.html#abbrev">Neff</a> values. Upper left: Subset of proteins with <a href="abbrev.html#abbrev">Neff</a> &lt; Q1. Upper right: Subset of proteins with Q1 &lt;= <a href="abbrev.html#abbrev">Neff</a> &lt; Q2. Lower left: Subset of proteins with Q2 &lt;= <a href="abbrev.html#abbrev">Neff</a> &lt; Q3. Lower right: Subset of proteins with Q3 &lt;= <a href="abbrev.html#abbrev">Neff</a> &lt; Q4. <strong>pseudo-likelihood</strong>: contact scores computed from pseudo-likelihood. <strong>CD #Gibbs steps = X</strong>: contact scores computed from <a href="abbrev.html#abbrev">CD</a> optimized with <a href="abbrev.html#abbrev">SGD</a> and evolving each Markov chain using the number of Gibbs steps specified in the legend.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:cd-gibbssteps-single-proteins-gradient"></span>
<img src="img/full_likelihood/gibbs_sampling/gradient_norm_1ahoa00.png" alt="Monitoring L2 norm of the gradient, \(||\nabla_{\w} L\!L(\v^*, \w)||_2\), for protein 1aho_A_00 and 1c75_A_00 during SGD optimization using different number of Gibbs steps and initial learning rates, \(\alpha_0\). Number of Gibbs steps is given in the legend, as well as particular choices for the initial learning rate, when not using the default \(\alpha_0 = \frac{5e-2}{\sqrt{N_{\text{eff}}}}\). Left Protein 1aho_A_00 has length L=64 and 378 sequences in the alignment (Neff=229) Right Protein 1c75_A_00 has length L=71 and 28078 sequences in the alignment (Neff=16808)." width="48%" /><img src="img/full_likelihood/gibbs_sampling/gradient_norm_1c75a00.png" alt="Monitoring L2 norm of the gradient, \(||\nabla_{\w} L\!L(\v^*, \w)||_2\), for protein 1aho_A_00 and 1c75_A_00 during SGD optimization using different number of Gibbs steps and initial learning rates, \(\alpha_0\). Number of Gibbs steps is given in the legend, as well as particular choices for the initial learning rate, when not using the default \(\alpha_0 = \frac{5e-2}{\sqrt{N_{\text{eff}}}}\). Left Protein 1aho_A_00 has length L=64 and 378 sequences in the alignment (Neff=229) Right Protein 1c75_A_00 has length L=71 and 28078 sequences in the alignment (Neff=16808)." width="48%" />
<p class="caption">
Figure E.15: Monitoring L2 norm of the gradient, <span class="math inline">\(||\nabla_{\w} L\!L(\v^*, \w)||_2\)</span>, for protein 1aho_A_00 and 1c75_A_00 during <a href="abbrev.html#abbrev">SGD</a> optimization using different number of Gibbs steps and initial learning rates, <span class="math inline">\(\alpha_0\)</span>. Number of Gibbs steps is given in the legend, as well as particular choices for the initial learning rate, when not using the default <span class="math inline">\(\alpha_0 = \frac{5e-2}{\sqrt{N_{\text{eff}}}}\)</span>. <strong>Left</strong> Protein 1aho_A_00 has length L=64 and 378 sequences in the alignment (<a href="abbrev.html#abbrev">Neff</a>=229) <strong>Right</strong> Protein 1c75_A_00 has length L=71 and 28078 sequences in the alignment (<a href="abbrev.html#abbrev">Neff</a>=16808).
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:adam-learning-rate"></span>
<img src="img/full_likelihood/adam/parameter_norms_1c75A00_1mkcA00_learningrates.png" alt="L2-norm of the coupling parameters, \(||\w||_2\), during CD optimization with ADAM with different fixed learning rates (no decay). The learning rate \(\alpha_0\) is specified in the legend. Left Protein 1c75A00 has length L=71 and 28078 sequences in the alignment (Neff=16808) Right Protein 1mkcA00 has length L=43 and 142 sequences in the alignment (Neff=96)." width="100%" />
<p class="caption">
Figure E.16: L2-norm of the coupling parameters, <span class="math inline">\(||\w||_2\)</span>, during <a href="abbrev.html#abbrev">CD</a> optimization with <em>ADAM</em> with different fixed learning rates (no decay). The learning rate <span class="math inline">\(\alpha_0\)</span> is specified in the legend. <strong>Left</strong> Protein 1c75A00 has length L=71 and 28078 sequences in the alignment (<a href="abbrev.html#abbrev">Neff</a>=16808) <strong>Right</strong> Protein 1mkcA00 has length L=43 and 142 sequences in the alignment (<a href="abbrev.html#abbrev">Neff</a>=96).
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:adam-learning-rate-annealing"></span>
<img src="img/full_likelihood/adam/1c75A00_decaying_learningrates_parameternorm.png" alt="L2-norm of the coupling parameters, \(||\w||_2\), during CD optimization with ADAM and different learning rate annealing schedules. The learning rate \(\alpha\) is specified with respect to Neff as \(\alpha = 2\mathrm{e}{-3}\log(\text{N}_{\text{eff}})\). The learning rate annealing schedule is specified in the legend. Left Convergence plot for protein 1mkc_A_00 having protein length L=43 and 142 sequences in the alignment (Neff=96). Left Protein 1c75A00 has length L=71 and 28078 sequences in the alignment (Neff=16808) Right Protein 1mkcA00 has length L=43 and 142 sequences in the alignment (Neff=96)." width="48%" /><img src="img/full_likelihood/adam/1mkcA00_decaying_learningrates_parameternorm.png" alt="L2-norm of the coupling parameters, \(||\w||_2\), during CD optimization with ADAM and different learning rate annealing schedules. The learning rate \(\alpha\) is specified with respect to Neff as \(\alpha = 2\mathrm{e}{-3}\log(\text{N}_{\text{eff}})\). The learning rate annealing schedule is specified in the legend. Left Convergence plot for protein 1mkc_A_00 having protein length L=43 and 142 sequences in the alignment (Neff=96). Left Protein 1c75A00 has length L=71 and 28078 sequences in the alignment (Neff=16808) Right Protein 1mkcA00 has length L=43 and 142 sequences in the alignment (Neff=96)." width="48%" />
<p class="caption">
Figure E.17: L2-norm of the coupling parameters, <span class="math inline">\(||\w||_2\)</span>, during <a href="abbrev.html#abbrev">CD</a> optimization with <em>ADAM</em> and different learning rate annealing schedules. The learning rate <span class="math inline">\(\alpha\)</span> is specified with respect to <a href="abbrev.html#abbrev">Neff</a> as <span class="math inline">\(\alpha = 2\mathrm{e}{-3}\log(\text{N}_{\text{eff}})\)</span>. The learning rate annealing schedule is specified in the legend. <strong>Left</strong> Convergence plot for protein 1mkc_A_00 having protein length L=43 and 142 sequences in the alignment (<a href="abbrev.html#abbrev">Neff</a>=96). <strong>Left</strong> Protein 1c75A00 has length L=71 and 28078 sequences in the alignment (<a href="abbrev.html#abbrev">Neff</a>=16808) <strong>Right</strong> Protein 1mkcA00 has length L=43 and 142 sequences in the alignment (<a href="abbrev.html#abbrev">Neff</a>=96).
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:precision-pll-cd-1c75a00"></span>
<iframe src="img/full_likelihood/comparing_couplings/1c75A00/1c75A00_precision_vs_rank_8seqsep_8contacthr.html" width="90%" height="600px">
</iframe>
<p class="caption">
Figure E.18: Precision of top ranked contact predictions for protein 1c75A00. Contact scores are computed as the <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the couplings <span class="math inline">\(\wij\)</span>. <strong>pseudo-likelihood</strong>: contact scores computed from pseudo-likelihood. <strong>contrastive divergence</strong>: contact scores computed from <a href="abbrev.html#abbrev">CD</a> optimized with <a href="abbrev.html#abbrev">SGD</a>.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:precision-cd-final-neff"></span>
<iframe src="img/full_likelihood/final/meanprecision_by_neff.html" width="90%" height="500px">
</iframe>
<p class="caption">
Figure E.19: Rolling mean over the mean precision of the L/10 to L top ranked predictions per protein for testset with 2300 proteins. Contact scores computed as <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm over pseudo-likelihood and contrastive divergence couplings. The rolling mean has been computed for the central protein within a window of 20 proteins. Window is shrunk for the proteins at the borders of Neff distribution.
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:pll-predictions-small-neff-proteins"></span>
<img src="img/full_likelihood/comparing_couplings/1ss3A00/contact_map_pseudo-likelihood_apc_1ss3A00.png" alt="Contact maps for protein 1ss3A00 and 1c55A00 computed as APC corrected Frobenius norm of the pseudo-likelihood couplings. Contacts are defined according to a \(8 \angstrom \Cb\) distance cutoff. Left: predicted contact map and native distance map for protein 1ss3A00 (protein length=50, N=42, Neff=36). Right predicted contact map and native distance map for protein 1c55A00 (protein length = 40, N=115, Neff = 78)." width="49%" /><img src="img/full_likelihood/comparing_couplings/1c55A00/contact_map_pseudo-likelihood_apc_1c55A00.png" alt="Contact maps for protein 1ss3A00 and 1c55A00 computed as APC corrected Frobenius norm of the pseudo-likelihood couplings. Contacts are defined according to a \(8 \angstrom \Cb\) distance cutoff. Left: predicted contact map and native distance map for protein 1ss3A00 (protein length=50, N=42, Neff=36). Right predicted contact map and native distance map for protein 1c55A00 (protein length = 40, N=115, Neff = 78)." width="49%" />
<p class="caption">
Figure E.20: Contact maps for protein 1ss3A00 and 1c55A00 computed as <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the pseudo-likelihood couplings. Contacts are defined according to a <span class="math inline">\(8 \angstrom \Cb\)</span> distance cutoff. <strong>Left</strong>: predicted contact map and native distance map for protein 1ss3A00 (protein length=50, N=42, <a href="abbrev.html#abbrev">Neff</a>=36). <strong>Right</strong> predicted contact map and native distance map for protein 1c55A00 (protein length = 40, N=115, Neff = 78).
</p>
</div>

<div class="figure" style="text-align: center"><span id="fig:contactmaps-1c55a00-different-samplesize"></span>
<img src="img/full_likelihood/comparing_couplings/1c55A00/1c55A00_seqsep8_contacthr8_apcTrue_samplesize03neff.png" alt="Contact maps for protein 1c55A00 (protein length = 40, N=115, Neff = 88) computed as APC corrected Frobenius norm of the contrastive-divergence couplings computed with different sample size choices. Contacts are defined according to a \(8 \angstrom \Cb\) distance cutoff. Top Left: sample size=0.3neff \(\approx\) 23 sequences. Top Right sample size=0.5neff \(\approx\) 39 sequences. Bottom Left: sample size=0.8neff \(\approx\) 62 sequences. Bottom Right sample size=max(10L,N)=&gt;115 sequences." width="49%" /><img src="img/full_likelihood/comparing_couplings/1c55A00/1c55A00_seqsep8_contacthr8_apcTrue_samplesize05neff.png" alt="Contact maps for protein 1c55A00 (protein length = 40, N=115, Neff = 88) computed as APC corrected Frobenius norm of the contrastive-divergence couplings computed with different sample size choices. Contacts are defined according to a \(8 \angstrom \Cb\) distance cutoff. Top Left: sample size=0.3neff \(\approx\) 23 sequences. Top Right sample size=0.5neff \(\approx\) 39 sequences. Bottom Left: sample size=0.8neff \(\approx\) 62 sequences. Bottom Right sample size=max(10L,N)=&gt;115 sequences." width="49%" /><img src="img/full_likelihood/comparing_couplings/1c55A00/1c55A00_seqsep8_contacthr8_apcTrue_samplesize08neff.png" alt="Contact maps for protein 1c55A00 (protein length = 40, N=115, Neff = 88) computed as APC corrected Frobenius norm of the contrastive-divergence couplings computed with different sample size choices. Contacts are defined according to a \(8 \angstrom \Cb\) distance cutoff. Top Left: sample size=0.3neff \(\approx\) 23 sequences. Top Right sample size=0.5neff \(\approx\) 39 sequences. Bottom Left: sample size=0.8neff \(\approx\) 62 sequences. Bottom Right sample size=max(10L,N)=&gt;115 sequences." width="49%" /><img src="img/full_likelihood/comparing_couplings/1c55A00/1c55A00_seqsep8_contacthr8_apcTrue_samplesize10L.png" alt="Contact maps for protein 1c55A00 (protein length = 40, N=115, Neff = 88) computed as APC corrected Frobenius norm of the contrastive-divergence couplings computed with different sample size choices. Contacts are defined according to a \(8 \angstrom \Cb\) distance cutoff. Top Left: sample size=0.3neff \(\approx\) 23 sequences. Top Right sample size=0.5neff \(\approx\) 39 sequences. Bottom Left: sample size=0.8neff \(\approx\) 62 sequences. Bottom Right sample size=max(10L,N)=&gt;115 sequences." width="49%" />
<p class="caption">
Figure E.21: Contact maps for protein 1c55A00 (protein length = 40, N=115, Neff = 88) computed as <a href="abbrev.html#abbrev">APC</a> corrected Frobenius norm of the contrastive-divergence couplings computed with different sample size choices. Contacts are defined according to a <span class="math inline">\(8 \angstrom \Cb\)</span> distance cutoff. <strong>Top Left</strong>: sample size=0.3neff <span class="math inline">\(\approx\)</span> 23 sequences. <strong>Top Right</strong> sample size=0.5neff <span class="math inline">\(\approx\)</span> 39 sequences. <strong>Bottom Left</strong>: sample size=0.8neff <span class="math inline">\(\approx\)</span> 62 sequences. <strong>Bottom Right</strong> sample size=max(10L,N)=&gt;115 sequences.
</p>
</div>
<!--
## Statistics for Comparing Couplings computed with Pseudo-likelihood and Contrastive Divergence 

protein|L|Neff|spearman rho|spearman pvalue|kolmogorov-smirnov|kolmogorov-smirnov pvalue
------|----|-------|-------|-------|---------|---------
1amuA01|170|23508|0.883827|0.000000|0.186982|0.000000
1a0pA01|98|21369|0.966502|0.000000|0.027141|0.059272
1a0pA02|182|19856|0.928410|0.000000|0.066602|0.000000
1d06A00|130|19073|0.977675|0.000000|0.036255|0.000031
1ae9A00|179|18927|0.956188|0.000000|0.052539|0.000000
1bazA00|53|17286|0.982963|0.000000|0.031930|0.477401
1c75A00|71|16808|0.947536|0.000000|0.047082|0.007797
1bboA02|31|15883|0.971095|0.000000|0.062366|0.316705
1ecsA00|126|15824|0.968401|0.000000|0.035810|0.000079
1e3oC01|71|15252|0.971069|0.000000|0.047485|0.007088
1cc5A00|83|14881|0.963036|0.000000|0.040846|0.006616
1bquA01|104|14778|0.960453|0.000000|0.035288|0.002460
1dw9A01|88|14559|0.977450|0.000000|0.040491|0.003631
1bl0A02|65|14511|0.911355|0.000000|0.080288|0.000003
1a8lA01|117|13387|0.971515|0.000000|0.039935|0.000038
1djxB01|52|13052|0.965467|0.000000|0.044495|0.141169
1al3A01|101|12992|0.976814|0.000000|0.026931|0.050425
1ek6A01|128|12707|0.989052|0.000000|0.033711|0.000188
1atgA02|109|12596|0.982236|0.000000|0.040095|0.000149
1daqA00|71|12537|0.977804|0.000000|0.040241|0.034764
: Proteins with the 20th largest [Neff](#abbrev) values from the dataset of 300 proteins used to evaluate [CD](#abbrev). Statistics have been computed for the contact scores computed from pseudo-likelihood couplings and [CD](#abbrev) couplings. Contact scores are computed as the [APC](#abbrev) corrected Frobenius norm of the couplings $\wij$. [CD](#abbrev) coupling parameters have been optimized with [SGD](#abbrev) using the following hyperparameter settings: simgoidal learning rate schedule with initial learning rate $\alpha_0 = \frac{5e-2}{\sqrt{N_{\text{eff}}}}$ and decay rate $\gamma = 5e-6$. Regularization coefficient $\lambda_w = 0.1L$. In order to approximate the gradient with [CD](#abbrev), 0.3Neff sequences are randomly selected and evolved with a Gibbs chain for 1 step.

protein|L|Neff|spearman rho|spearman pvalue|kolmogorov-smirnov|kolmogorov-smirnov pvalue
------|----|-------|-------|-------|---------|---------
1ehsA00|48|3|0.933381|0.000000|0.100177|0.000022
1e0fI01|33|8|0.972043|0.000000|0.043561|0.688999
1d02A00|202|11|0.979033|0.000000|0.058667|0.000000
1cfrA00|285|15|0.970660|0.000000|0.059501|0.000000
1dmlA00|291|19|0.939673|0.000000|0.068752|0.000000
16vpA00|321|20|0.900008|0.000000|0.110903|0.000000
1c01A00|76|20|0.887544|0.000000|0.077895|0.000000
1cq3A00|233|20|0.861049|0.000000|0.047765|0.000000
1dc1A02|88|20|0.871925|0.000000|0.103187|0.000000
1cmbA00|104|20|0.712424|0.000000|0.127147|0.000000
1d4vA02|31|24|0.840701|0.000000|0.079570|0.100219
1bvp101|120|25|0.884899|0.000000|0.088936|0.000000
1dc1A01|235|25|0.805065|0.000000|0.176978|0.000000
1ahlA00|49|25|0.794887|0.000000|0.147959|0.000000
1bvp103|96|28|0.905518|0.000000|0.093640|0.000000
1e5rA02|87|28|0.964278|0.000000|0.068431|0.000000
1bcpB01|86|33|0.842349|0.000000|0.068947|0.000000
1cl8A00|276|34|0.666682|0.000000|0.146271|0.000000
1c1kA01|102|34|0.787192|0.000000|0.067171|0.000000
1bdsA00|43|35|0.744768|0.000000|0.173865|0.000000
: Proteins with the 20th smallest [Neff](#abbrev) values from the dataset of 300 proteins used to evaluate [CD](#abbrev). Statistics have been computed for the contact scores computed from pseudo-likelihood couplings and [CD](#abbrev) couplings. Contact scores are computed as the [APC](#abbrev) corrected Frobenius norm of the couplings $\wij$. [CD](#abbrev) coupling parameters have been optimized with [SGD](#abbrev) using the following hyperparameter settings: simgoidal learning rate schedule with initial learning rate $\alpha_0 = \frac{5e-2}{\sqrt{N_{\text{eff}}}}$ and decay rate $\gamma = 5e-6$. Regularization coefficient $\lambda_w = 0.1L$. In order to approximate the gradient with [CD](#abbrev), 0.3Neff sequences are randomly selected and evolved with a Gibbs chain for 1 step.

-->
</div>
<div id="disqus_thread"></div>
<script>
/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://contactpredictionthesis.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
            </section>

          </div>
        </div>
      </div>
<a href="interpretation-of-coupling-matrices.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="training-of-the-random-forest-contact-prior.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/susannvorberg/phd_thesis/edit/master/10-appendix.Rmd",
"text": "Edit"
},
"download": ["PhD_thesis_Susann_Vorberg.pdf"],
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
