<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>PhD thesis: residue-residue contact prediction</title>
  <meta name="description" content="This is my PhD thesis on residue-residue contact prediction.">
  <meta name="generator" content="bookdown 0.3.9 and GitBook 2.6.7">

  <meta property="og:title" content="PhD thesis: residue-residue contact prediction" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is my PhD thesis on residue-residue contact prediction." />
  <meta name="github-repo" content="susannvorberg/phd_thesis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="PhD thesis: residue-residue contact prediction" />
  
  <meta name="twitter:description" content="This is my PhD thesis on residue-residue contact prediction." />
  

<meta name="author" content="Susann Vorberg">


<meta name="date" content="2017-10-26">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="bayesian-statistical-model-for-contact-prediction.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />








<script type="text/x-mathjax-config">
 MathJax.Hub.Config({
  tex2jax: {inlineMath: [["$","$"],["\\(","\\)"]]},
  TeX: { 
    extensions: ["mediawiki-texvc.js", "sinuitx.js"],
    Macros: {
      Cb: "C_\\beta",
      eq: "\\!=\\!",
      Gauss: "\\mathcal{N}",
      H: "\\mathbf{H}",
      Hij : "\\H_{ij}",
      I: "\\mathbf{I}",
      Lijk: "\\mathbf{\\Lambda}_{ij,k}",
      Lk: "\\mathbf{\\Lambda}_k",
      LL: "L\\!L(\\mathbf{v}, \\mathbf{w})",
      LLreg: "L\\!L_\\mathrm{reg}",
      muijk: "\\mathbf{\\mu}_{ij,k}",
      muk: "\\mathbf{\\mu}_k",
      neff: "N_\\mathrm{eff}",
      r: "\\mathbf{r}",
      rij: "r_{ij}",
      c: "\\mathbf{c}",
      cij: "c_{ij}",
      seq: "\\mathbf{x}",
      Qij: "\\mathbf{Q}_{ij}",
      q: "\\mathbf{q}",
      qij: "\\mathbf{q'}_{ij}",
      Sn: "\\mathcal{S}_n",
      v: "\\mathbf{v}",
      vi: "\\mathcal{v}_{i}",
      vj: "\\mathcal{v}_{j}",
      via: "\\mathcal{v}_{ia}",
      vja: "\\mathcal{v}_{ja}",
      w: "\\mathbf{w}",
      wij: "\\mathbf{w}_{ij}",
      wijab: "\\mathcal{w}_{ijab}",
      wijcd: "\\mathcal{w}_{ijcd}",
      wklcd: "\\mathcal{w}_{klcd}",
      X: "\\mathbf{X}",
      angstrom: "\\AA \\; \\;"
      }
  }
});
</script>


 
<!--
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  extensions: ["tex2jax.js", "[siunitx]/siunitx.js"],
  TeX: { TagSide: "left" }
});
MathJax.Ajax.config.path['siunitx']  = '../latex/MathJax-siunitx-master/';
</script>
//-->



<link rel="stylesheet" href="css/style.css" type="text/css" />
<link rel="stylesheet" href="css/toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PhD Thesis Susann Vorberg</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Summary</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="1" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>1</b> Background</a><ul>
<li class="chapter" data-level="1.1" data-path="general-intro.html"><a href="general-intro.html"><i class="fa fa-check"></i><b>1.1</b> Biological Background</a></li>
<li class="chapter" data-level="1.2" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html"><i class="fa fa-check"></i><b>1.2</b> Introduction to Contact Prediction</a><ul>
<li class="chapter" data-level="1.2.1" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html#local-methods"><i class="fa fa-check"></i><b>1.2.1</b> Local Statistical Models</a></li>
<li class="chapter" data-level="1.2.2" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html#global-methods"><i class="fa fa-check"></i><b>1.2.2</b> Global Statistical Models</a></li>
<li class="chapter" data-level="1.2.3" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html#meta-predictors"><i class="fa fa-check"></i><b>1.2.3</b> Machine Learning Methods and Meta-Predictors</a></li>
<li class="chapter" data-level="1.2.4" data-path="introduction-to-contact-prediction.html"><a href="introduction-to-contact-prediction.html#maxent"><i class="fa fa-check"></i><b>1.2.4</b> Modelling Protein Families with Potts Model</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="application-contact-prediction.html"><a href="application-contact-prediction.html"><i class="fa fa-check"></i><b>1.3</b> Applications</a></li>
<li class="chapter" data-level="1.4" data-path="intro-cp-evaluation.html"><a href="intro-cp-evaluation.html"><i class="fa fa-check"></i><b>1.4</b> Evaluating Contact Prediction Methods</a><ul>
<li class="chapter" data-level="1.4.1" data-path="intro-cp-evaluation.html"><a href="intro-cp-evaluation.html#seq-sep"><i class="fa fa-check"></i><b>1.4.1</b> Sequence Separation</a></li>
<li class="chapter" data-level="1.4.2" data-path="intro-cp-evaluation.html"><a href="intro-cp-evaluation.html#interpretation-of-evaluation-results"><i class="fa fa-check"></i><b>1.4.2</b> Interpretation of Evaluation Results</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="challenges.html"><a href="challenges.html"><i class="fa fa-check"></i><b>1.5</b> Challenges for Coevolutionary Inference</a><ul>
<li class="chapter" data-level="1.5.1" data-path="challenges.html"><a href="challenges.html#phylogenetic-noise"><i class="fa fa-check"></i><b>1.5.1</b> Phylogenetic Effects as a Source of Noise</a></li>
<li class="chapter" data-level="1.5.2" data-path="challenges.html"><a href="challenges.html#entropic-effects-as-a-source-of-noise"><i class="fa fa-check"></i><b>1.5.2</b> Entropic Effects as a Source of Noise</a></li>
<li class="chapter" data-level="1.5.3" data-path="challenges.html"><a href="challenges.html#finite-sampling-effects"><i class="fa fa-check"></i><b>1.5.3</b> Finite Sampling Effects</a></li>
<li class="chapter" data-level="1.5.4" data-path="challenges.html"><a href="challenges.html#multiple-sequence-alignments"><i class="fa fa-check"></i><b>1.5.4</b> Multiple Sequence Alignments</a></li>
<li class="chapter" data-level="1.5.5" data-path="challenges.html"><a href="challenges.html#alternative-sources-of-coevolution"><i class="fa fa-check"></i><b>1.5.5</b> Alternative Sources of Coevolution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="interpreting-coupling-matrices.html"><a href="interpreting-coupling-matrices.html"><i class="fa fa-check"></i><b>2</b> Interpretation of Coupling Matrices</a><ul>
<li class="chapter" data-level="2.1" data-path="correlation-between-couplings-and-class.html"><a href="correlation-between-couplings-and-class.html"><i class="fa fa-check"></i><b>2.1</b> Single Coupling Values Carry Evidence of Contacts</a></li>
<li class="chapter" data-level="2.2" data-path="coupling-profiles.html"><a href="coupling-profiles.html"><i class="fa fa-check"></i><b>2.2</b> Coupling Profiles Vary with Distance</a></li>
<li class="chapter" data-level="2.3" data-path="physico-chemical-fingerprints-in-coupling-matrices.html"><a href="physico-chemical-fingerprints-in-coupling-matrices.html"><i class="fa fa-check"></i><b>2.3</b> Physico-Chemical Fingerprints in Coupling Matrices</a></li>
<li class="chapter" data-level="2.4" data-path="higher-order-coupling-profiles.html"><a href="higher-order-coupling-profiles.html"><i class="fa fa-check"></i><b>2.4</b> Higher Order Dependencies Between Couplings</a></li>
<li class="chapter" data-level="2.5" data-path="discussion.html"><a href="discussion.html"><i class="fa fa-check"></i><b>2.5</b> Discussion</a></li>
<li class="chapter" data-level="2.6" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>2.6</b> Methods</a><ul>
<li class="chapter" data-level="2.6.1" data-path="methods.html"><a href="methods.html#dataset"><i class="fa fa-check"></i><b>2.6.1</b> Dataset</a></li>
<li class="chapter" data-level="2.6.2" data-path="methods.html"><a href="methods.html#diff-ccmpred-ccmpredpy"><i class="fa fa-check"></i><b>2.6.2</b> Computing Pseudo-Likelihood Couplings</a></li>
<li class="chapter" data-level="2.6.3" data-path="methods.html"><a href="methods.html#seq-reweighting"><i class="fa fa-check"></i><b>2.6.3</b> Sequence Reweighting</a></li>
<li class="chapter" data-level="2.6.4" data-path="methods.html"><a href="methods.html#amino-acid-frequencies"><i class="fa fa-check"></i><b>2.6.4</b> Computing Amino Acid Frequencies</a></li>
<li class="chapter" data-level="2.6.5" data-path="methods.html"><a href="methods.html#methods-regularization"><i class="fa fa-check"></i><b>2.6.5</b> Regularization</a></li>
<li class="chapter" data-level="2.6.6" data-path="methods.html"><a href="methods.html#method-coupling-correlation"><i class="fa fa-check"></i><b>2.6.6</b> Correlation of Couplings with Contact Class</a></li>
<li class="chapter" data-level="2.6.7" data-path="methods.html"><a href="methods.html#method-coupling-profile"><i class="fa fa-check"></i><b>2.6.7</b> Coupling Distribution Plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="optimizing-full-likelihood.html"><a href="optimizing-full-likelihood.html"><i class="fa fa-check"></i><b>3</b> Optimizing the Full Likelihood</a><ul>
<li class="chapter" data-level="3.1" data-path="full-likelihood-gradient.html"><a href="full-likelihood-gradient.html"><i class="fa fa-check"></i><b>3.1</b> Approximating the Gradient of the Full Likelihood with Contrastive Divergence</a></li>
<li class="chapter" data-level="3.2" data-path="full-likelihood-optimization.html"><a href="full-likelihood-optimization.html"><i class="fa fa-check"></i><b>3.2</b> Optimizing the Full Likelihood</a><ul>
<li class="chapter" data-level="3.2.1" data-path="full-likelihood-optimization.html"><a href="full-likelihood-optimization.html#convergence-criteria-sgd"><i class="fa fa-check"></i><b>3.2.1</b> Convergence Criterion for Stochastic Gradient Descent</a></li>
<li class="chapter" data-level="3.2.2" data-path="full-likelihood-optimization.html"><a href="full-likelihood-optimization.html#sgd-hyperparameter-tuning"><i class="fa fa-check"></i><b>3.2.2</b> Tuning Hyperparameters of Stochastic Gradient Descent Optimizer</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html"><i class="fa fa-check"></i><b>3.3</b> Tuning the Gibbs Sampling Scheme for Contrastive Divergence</a><ul>
<li class="chapter" data-level="3.3.1" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html#regularization-for-cd-with-sgd"><i class="fa fa-check"></i><b>3.3.1</b> Tuning Regularization Coefficients for Contrastive Divergence</a></li>
<li class="chapter" data-level="3.3.2" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html#cd-sampling-size"><i class="fa fa-check"></i><b>3.3.2</b> Varying the Sample Size</a></li>
<li class="chapter" data-level="3.3.3" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html#cd-gibbs-steps"><i class="fa fa-check"></i><b>3.3.3</b> Varying the number of Gibbs Steps</a></li>
<li class="chapter" data-level="3.3.4" data-path="cd-sampling-optimization.html"><a href="cd-sampling-optimization.html#cd-gibbs-steps"><i class="fa fa-check"></i><b>3.3.4</b> Persistent Contrastive Divergence</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="adam-results.html"><a href="adam-results.html"><i class="fa fa-check"></i><b>3.4</b> Using ADAM to optimize Contrastive Divergence</a></li>
<li class="chapter" data-level="3.5" data-path="comparing-pll-cd.html"><a href="comparing-pll-cd.html"><i class="fa fa-check"></i><b>3.5</b> Comparing CD couplings to pLL couplings</a><ul>
<li class="chapter" data-level="3.5.1" data-path="comparing-pll-cd.html"><a href="comparing-pll-cd.html#protein-1c75a00"><i class="fa fa-check"></i><b>3.5.1</b> Protein 1c75A00</a></li>
<li class="chapter" data-level="3.5.2" data-path="comparing-pll-cd.html"><a href="comparing-pll-cd.html#protein-1ss3a00-and-1c55a00"><i class="fa fa-check"></i><b>3.5.2</b> Protein 1ss3A00 and 1c55A00</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="discussion-1.html"><a href="discussion-1.html"><i class="fa fa-check"></i><b>3.6</b> Discussion</a></li>
<li class="chapter" data-level="3.7" data-path="methods-1.html"><a href="methods-1.html"><i class="fa fa-check"></i><b>3.7</b> Methods</a><ul>
<li class="chapter" data-level="3.7.1" data-path="methods-1.html"><a href="methods-1.html#potts-full-likelihood"><i class="fa fa-check"></i><b>3.7.1</b> The Potts Model</a></li>
<li class="chapter" data-level="3.7.2" data-path="methods-1.html"><a href="methods-1.html#gap-treatment"><i class="fa fa-check"></i><b>3.7.2</b> Treating Gaps as Missing Information</a></li>
<li class="chapter" data-level="3.7.3" data-path="methods-1.html"><a href="methods-1.html#the-regularized-full-log-likelihood-and-its-gradient-with-gap-treatment"><i class="fa fa-check"></i><b>3.7.3</b> The Regularized Full Log Likelihood and its Gradient With Gap Treatment</a></li>
<li class="chapter" data-level="3.7.4" data-path="methods-1.html"><a href="methods-1.html#prior-v"><i class="fa fa-check"></i><b>3.7.4</b> The prior on single potentials</a></li>
<li class="chapter" data-level="3.7.5" data-path="methods-1.html"><a href="methods-1.html#methods-sgd"><i class="fa fa-check"></i><b>3.7.5</b> Stochastic Gradien Descent</a></li>
<li class="chapter" data-level="3.7.6" data-path="methods-1.html"><a href="methods-1.html#methods-cd-sampling"><i class="fa fa-check"></i><b>3.7.6</b> Computing the Gradient with Contrastive Divergence</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="contact-prior.html"><a href="contact-prior.html"><i class="fa fa-check"></i><b>4</b> Random Forest Contact Prior</a><ul>
<li class="chapter" data-level="4.1" data-path="random-forest-classifiers.html"><a href="random-forest-classifiers.html"><i class="fa fa-check"></i><b>4.1</b> Random Forest Classifiers</a></li>
<li class="chapter" data-level="4.2" data-path="hyperparameter-optimization-for-random-forest.html"><a href="hyperparameter-optimization-for-random-forest.html"><i class="fa fa-check"></i><b>4.2</b> Hyperparameter Optimization for Random Forest</a></li>
<li class="chapter" data-level="4.3" data-path="evaluating-random-forest-model-as-contact-predictor.html"><a href="evaluating-random-forest-model-as-contact-predictor.html"><i class="fa fa-check"></i><b>4.3</b> Evaluating Random Forest Model as Contact Predictor</a></li>
<li class="chapter" data-level="4.4" data-path="using-contact-scores-as-additional-features.html"><a href="using-contact-scores-as-additional-features.html"><i class="fa fa-check"></i><b>4.4</b> Using Contact Scores as Additional Features</a></li>
<li class="chapter" data-level="4.5" data-path="methods-2.html"><a href="methods-2.html"><i class="fa fa-check"></i><b>4.5</b> Methods</a><ul>
<li class="chapter" data-level="4.5.1" data-path="methods-2.html"><a href="methods-2.html#seq-features"><i class="fa fa-check"></i><b>4.5.1</b> Features used to train Random Forest Model</a></li>
<li class="chapter" data-level="4.5.2" data-path="methods-2.html"><a href="methods-2.html#simple-contact-prior-with-respect-to-protein-length"><i class="fa fa-check"></i><b>4.5.2</b> Simple Contact Prior with Respect to Protein Length</a></li>
<li class="chapter" data-level="4.5.3" data-path="methods-2.html"><a href="methods-2.html#rf-training"><i class="fa fa-check"></i><b>4.5.3</b> Cross-validation for Random Forest Training</a></li>
<li class="chapter" data-level="4.5.4" data-path="methods-2.html"><a href="methods-2.html#rf-feature-selection"><i class="fa fa-check"></i><b>4.5.4</b> Feature Selection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="bayesian-approach.html"><a href="bayesian-approach.html"><i class="fa fa-check"></i><b>5</b> A Bayesian Statistical Model for Residue-Residue Contact Prediction</a><ul>
<li class="chapter" data-level="5.1" data-path="overview-posterior-distances.html"><a href="overview-posterior-distances.html"><i class="fa fa-check"></i><b>5.1</b> Computing the Posterior Probabiilty of a Contact</a></li>
<li class="chapter" data-level="5.2" data-path="coupling-prior.html"><a href="coupling-prior.html"><i class="fa fa-check"></i><b>5.2</b> Modelling the Prior Over Couplings Depending on Contact States</a></li>
<li class="chapter" data-level="5.3" data-path="bayesian-model-training-hyperparameters.html"><a href="bayesian-model-training-hyperparameters.html"><i class="fa fa-check"></i><b>5.3</b> Training the Hyperparameters in the Likelihood Function of Contact States</a></li>
<li class="chapter" data-level="5.4" data-path="training-hyperparameters-for-a-gaussian-mixture-with-three-components.html"><a href="training-hyperparameters-for-a-gaussian-mixture-with-three-components.html"><i class="fa fa-check"></i><b>5.4</b> Training Hyperparameters for a Gaussian Mixture with Three Components</a></li>
<li class="chapter" data-level="5.5" data-path="training-hyperparameters-for-a-gaussian-mixture-with-five-components.html"><a href="training-hyperparameters-for-a-gaussian-mixture-with-five-components.html"><i class="fa fa-check"></i><b>5.5</b> Training Hyperparameters for a Gaussian Mixture with Five Components</a></li>
<li class="chapter" data-level="5.6" data-path="training-hyperparameters-for-a-gaussian-mixture-with-ten-components.html"><a href="training-hyperparameters-for-a-gaussian-mixture-with-ten-components.html"><i class="fa fa-check"></i><b>5.6</b> Training Hyperparameters for a Gaussian Mixture with Ten Components</a></li>
<li class="chapter" data-level="5.7" data-path="posterior-of-rij.html"><a href="posterior-of-rij.html"><i class="fa fa-check"></i><b>5.7</b> Computing The Posterior Probability of Contacts</a></li>
<li class="chapter" data-level="5.8" data-path="analysing-contact-maps.html"><a href="analysing-contact-maps.html"><i class="fa fa-check"></i><b>5.8</b> Analysing Contact Maps</a></li>
<li class="chapter" data-level="5.9" data-path="discussion-2.html"><a href="discussion-2.html"><i class="fa fa-check"></i><b>5.9</b> Discussion</a></li>
<li class="chapter" data-level="5.10" data-path="methods-3.html"><a href="methods-3.html"><i class="fa fa-check"></i><b>5.10</b> Methods</a><ul>
<li class="chapter" data-level="5.10.1" data-path="methods-3.html"><a href="methods-3.html#methods-coupling-prior"><i class="fa fa-check"></i><b>5.10.1</b> Modelling the Prior Over Couplings Depending on Contact States</a></li>
<li class="chapter" data-level="5.10.2" data-path="methods-3.html"><a href="methods-3.html#laplace-approx"><i class="fa fa-check"></i><b>5.10.2</b> Gaussian Approximation to the Posterior of Couplings</a></li>
<li class="chapter" data-level="5.10.3" data-path="methods-3.html"><a href="methods-3.html#likelihood-fct-distances"><i class="fa fa-check"></i><b>5.10.3</b> Integrating out the Hidden Variables to Obtain the Likelihood Function of the Contact States</a></li>
<li class="chapter" data-level="5.10.4" data-path="methods-3.html"><a href="methods-3.html#Hessian-offdiagonal"><i class="fa fa-check"></i><b>5.10.4</b> The Hessian off-diagonal Elements Carry a Negligible Signal</a></li>
<li class="chapter" data-level="5.10.5" data-path="methods-3.html"><a href="methods-3.html#neg-Hessian-computation"><i class="fa fa-check"></i><b>5.10.5</b> Efficiently Computing the negative Hessian of the regularized log-likelihood</a></li>
<li class="chapter" data-level="5.10.6" data-path="methods-3.html"><a href="methods-3.html#inv-lambda-ij-k"><i class="fa fa-check"></i><b>5.10.6</b> Efficiently Computing the Inverse of Matrix <span class="math inline">\(\Lijk\)</span></a></li>
<li class="chapter" data-level="5.10.7" data-path="methods-3.html"><a href="methods-3.html#gradient-muk"><i class="fa fa-check"></i><b>5.10.7</b> The gradient of the log likelihood with respect to <span class="math inline">\(\muk\)</span></a></li>
<li class="chapter" data-level="5.10.8" data-path="methods-3.html"><a href="methods-3.html#gradient-lambdak"><i class="fa fa-check"></i><b>5.10.8</b> The gradient of the log likelihood with respect to <span class="math inline">\(\Lk\)</span></a></li>
<li class="chapter" data-level="5.10.9" data-path="methods-3.html"><a href="methods-3.html#the-gradient-of-the-log-likelihood-with-respect-to-gamma_k"><i class="fa fa-check"></i><b>5.10.9</b> The gradient of the log likelihood with respect to <span class="math inline">\(\gamma_k\)</span></a></li>
<li class="chapter" data-level="5.10.10" data-path="methods-3.html"><a href="methods-3.html#bayesian-model-distances"><i class="fa fa-check"></i><b>5.10.10</b> Extending the Bayesian Statistical Model for the Prediction of Protein Residue-Residue Distances</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="practical-methods.html"><a href="practical-methods.html"><i class="fa fa-check"></i><b>5.11</b> Practical Methods</a><ul>
<li class="chapter" data-level="5.11.1" data-path="practical-methods.html"><a href="practical-methods.html#training-hyperparameters-bayesian-model"><i class="fa fa-check"></i><b>5.11.1</b> Training the Hyperparameters in the Likelihood Function of Contact States</a></li>
<li class="chapter" data-level="5.11.2" data-path="practical-methods.html"><a href="practical-methods.html#dataset-training-bayesian-model"><i class="fa fa-check"></i><b>5.11.2</b> Dataset Specifications</a></li>
<li class="chapter" data-level="5.11.3" data-path="practical-methods.html"><a href="practical-methods.html#model-specifications-training-bayesian-model"><i class="fa fa-check"></i><b>5.11.3</b> Model Specifications</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion-and-outlook.html"><a href="conclusion-and-outlook.html"><i class="fa fa-check"></i><b>6</b> Conclusion and Outlook</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="abbrev.html"><a href="abbrev.html"><i class="fa fa-check"></i><b>A</b> Abbreviations</a></li>
<li class="chapter" data-level="B" data-path="amino-acids.html"><a href="amino-acids.html"><i class="fa fa-check"></i><b>B</b> Amino Acid Alphabet</a></li>
<li class="chapter" data-level="C" data-path="dataset-properties.html"><a href="dataset-properties.html"><i class="fa fa-check"></i><b>C</b> Dataset Properties</a></li>
<li class="chapter" data-level="D" data-path="interpretation-of-coupling-matrices.html"><a href="interpretation-of-coupling-matrices.html"><i class="fa fa-check"></i><b>D</b> Interpretation of Coupling Matrices</a></li>
<li class="chapter" data-level="E" data-path="optimizing-full-likelihood-with-gradient-descent.html"><a href="optimizing-full-likelihood-with-gradient-descent.html"><i class="fa fa-check"></i><b>E</b> Optimizing Full Likelihood with Gradient Descent</a></li>
<li class="chapter" data-level="F" data-path="training-of-the-random-forest-contact-prior.html"><a href="training-of-the-random-forest-contact-prior.html"><i class="fa fa-check"></i><b>F</b> Training of the Random Forest Contact Prior</a></li>
<li class="chapter" data-level="G" data-path="bayesian-statistical-model-for-contact-prediction.html"><a href="bayesian-statistical-model-for-contact-prediction.html"><i class="fa fa-check"></i><b>G</b> Bayesian statistical model for contact prediction</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PhD thesis: residue-residue contact prediction</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references">
<div>
<p>1. Anfinsen, C.B. (1973). Principles that Govern the Folding of Protein Chains. Science (80-. ). <em>181</em>, 223–230. Available at: <a href="http://www.sciencemag.org/content/181/4096/223" class="uri">http://www.sciencemag.org/content/181/4096/223</a>.</p>
</div>
<div>
<p>2. Wright, P.E., and Dyson, H. (1999). Intrinsically unstructured proteins: re-assessing the protein structure-function paradigm. J. Mol. Biol. <em>293</em>, 321–331. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/10550212 http://linkinghub.elsevier.com/retrieve/pii/S0022283699931108" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/10550212 http://linkinghub.elsevier.com/retrieve/pii/S0022283699931108</a>.</p>
</div>
<div>
<p>3. Fraser, P.E. (2014). Prions and prion-like proteins. J. Biol. Chem. <em>289</em>, 19839–40. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24860092 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC4106303" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/24860092 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC4106303</a>.</p>
</div>
<div>
<p>4. Samish, I., Bourne, P.E., and Najmanovich, R.J. (2015). Achievements and challenges in structural bioinformatics and computational biophysics. Bioinformatics <em>31</em>, 146–150. Available at: <a href="https://oup.silverchair-cdn.com/oup/backfile/Content{\_}public/Journal/bioinformatics/31/1/10.1093{\_}bioinformatics{\_}btu769/2/btu769.pdf?Expires=1500051331{\&amp;}Signature=WYTe4F7OxZkkcnJmsjt-r8gIyHaHX5ANqUoy6w0PUuete2b{~}5ZU{~}{~}D6KOB1vq5A8MgCnrq3pDHUn0OSgz0QFmtU2RYf907-" class="uri">https://oup.silverchair-cdn.com/oup/backfile/Content{\_}public/Journal/bioinformatics/31/1/10.1093{\_}bioinformatics{\_}btu769/2/btu769.pdf?Expires=1500051331{\&amp;}Signature=WYTe4F7OxZkkcnJmsjt-r8gIyHaHX5ANqUoy6w0PUuete2b{~}5ZU{~}{~}D6KOB1vq5A8MgCnrq3pDHUn0OSgz0QFmtU2RYf907-</a>.</p>
</div>
<div>
<p>5. Schwede, T. (2013). Protein modeling: what happened to the “protein structure gap”? Structure <em>21</em>, 1531–40. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24010712 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3816506" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/24010712 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3816506</a>.</p>
</div>
<div>
<p>6. Levinthal, C. (1969). How to Fold Graciously. 22–24. Available at: <a href="http://www.citeulike.org/user/FBerkemeier/article/380320" class="uri">http://www.citeulike.org/user/FBerkemeier/article/380320</a>.</p>
</div>
<div>
<p>7. Lesk, A.M., and Chothia, C. (1980). How different amino acid sequences determine similar protein structures: The structure and evolutionary dynamics of the globins. J. Mol. Biol. <em>136</em>, 225–270. Available at: <a href="http://linkinghub.elsevier.com/retrieve/pii/0022283680903733" class="uri">http://linkinghub.elsevier.com/retrieve/pii/0022283680903733</a>.</p>
</div>
<div>
<p>8. Sander, C., and Schneider, R. (1991). Database of homology-derived protein structures and the structural meaning of sequence alignment. Proteins <em>9</em>, 56–68. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/2017436" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/2017436</a>.</p>
</div>
<div>
<p>9. Chothia, C., and Lesk, A.M. (1986). The relation between the divergence of sequence and structure in proteins. EMBO J. <em>5</em>, 823–6. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/3709526 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC1166865" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/3709526 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC1166865</a>.</p>
</div>
<div>
<p>10. Martí-Renom, M.A., Stuart, A.C., Fiser, A., Sánchez, R., Melo, F., and Šali, A. (2000). Comparative Protein Structure Modeling of Genes and Genomes. Annu. Rev. Biophys. Biomol. Struct. <em>29</em>, 291–325. Available at: <a href="http://www.annualreviews.org/doi/10.1146/annurev.biophys.29.1.291" class="uri">http://www.annualreviews.org/doi/10.1146/annurev.biophys.29.1.291</a>.</p>
</div>
<div>
<p>11. Dorn, M., Silva, M.B. e, Buriol, L.S., and Lamb, L.C. (2014). Three-dimensional protein structure prediction: Methods and computational strategies. Comput. Biol. Chem. <em>53</em>, 251–276. Available at: <a href="http://linkinghub.elsevier.com/retrieve/pii/S1476927114001248" class="uri">http://linkinghub.elsevier.com/retrieve/pii/S1476927114001248</a>.</p>
</div>
<div>
<p>12. Berman, H.M. (2000). The Protein Data Bank. Nucleic Acids Res. <em>28</em>, 235–242. Available at: <a href="http://nar.oxfordjournals.org/content/28/1/235.abstract" class="uri">http://nar.oxfordjournals.org/content/28/1/235.abstract</a>.</p>
</div>
<div>
<p>13. The UniProt Consortium (2017). UniProt: the universal protein knowledgebase. Nucleic Acids Res. <em>45</em>, D158–D169. Available at: <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkw1099" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkw1099</a>.</p>
</div>
<div>
<p>14. Carpenter, E.P., Beis, K., Cameron, A.D., and Iwata, S. (2008). Overcoming the challenges of membrane protein crystallography. Curr. Opin. Struct. Biol. <em>18</em>, 581–6. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/18674618 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2580798" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/18674618 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2580798</a>.</p>
</div>
<div>
<p>15. Moraes, I., Evans, G., Sanchez-Weatherby, J., Newstead, S., and Stewart, P.D.S. (2014). Membrane protein structure determination - the next generation. Biochim. Biophys. Acta <em>1838</em>, 78–87. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23860256 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3898769" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/23860256 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3898769</a>.</p>
</div>
<div>
<p>16. Jacobson, M.P., Friesner, R.A., Xiang, Z., and Honig, B. (2002). On the Role of the Crystal Environment in Determining Protein Side-chain Conformations. J. Mol. Biol. <em>320</em>, 597–608. Available at: <a href="http://www.sciencedirect.com/science/article/pii/S0022283602004709?via{\%}3Dihub" class="uri">http://www.sciencedirect.com/science/article/pii/S0022283602004709?via{\%}3Dihub</a>.</p>
</div>
<div>
<p>17. Bieri, M., Kwan, A.H., Mobli, M., King, G.F., Mackay, J.P., and Gooley, P.R. (2011). Macromolecular NMR spectroscopy for the non-spectroscopist: beyond macromolecular solution structure determination. FEBS J. <em>278</em>, 704–715. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21214861 http://doi.wiley.com/10.1111/j.1742-4658.2011.08005.x" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/21214861 http://doi.wiley.com/10.1111/j.1742-4658.2011.08005.x</a>.</p>
</div>
<div>
<p>18. Billeter, M., Wagner, G., and Wüthrich, K. (2008). Solution NMR structure determination of proteins revisited. J. Biomol. NMR <em>42</em>, 155–8. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/18827972 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2748914" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/18827972 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2748914</a>.</p>
</div>
<div>
<p>19. Egelman, E.H. (2016). The Current Revolution in Cryo-EM. Biophysj <em>110</em>, 1008–1012. Available at: <a href="http://www.cell.com/biophysj/pdf/S0006-3495(16)00142-9.pdf" class="uri">http://www.cell.com/biophysj/pdf/S0006-3495(16)00142-9.pdf</a>.</p>
</div>
<div>
<p>20. Fernandez-Leiro, R., and Scheres, S.H.W. (2016). Unravelling biological macromolecules with cryo-electron microscopy. Nature <em>537</em>, 339–46. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/27629640 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5074357" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/27629640 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5074357</a>.</p>
</div>
<div>
<p>21. Reuter, J.A., Spacek, D.V., and Snyder, M.P. (2015). High-throughput sequencing technologies. Mol. Cell <em>58</em>, 586–97. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26000844 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC4494749" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/26000844 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC4494749</a>.</p>
</div>
<div>
<p>22. Goodwin, S., McPherson, J.D., and McCombie, W.R. (2016). Coming of age: ten years of next-generation sequencing technologies. Nat. Rev. Genet. <em>17</em>, 333–351. Available at: <a href="http://www.nature.com/doifinder/10.1038/nrg.2016.49" class="uri">http://www.nature.com/doifinder/10.1038/nrg.2016.49</a>.</p>
</div>
<div>
<p>23. NovaSeq System Specifications | The next era of sequencing starts now Available at: <a href="https://www.illumina.com/systems/sequencing-platforms/novaseq/specifications.html" class="uri">https://www.illumina.com/systems/sequencing-platforms/novaseq/specifications.html</a> [Accessed October 18, 2017].</p>
</div>
<div>
<p>24. Tringe, S.G., and Rubin, E.M. (2005). Metagenomics: DNA sequencing of environmental samples. Nat. Rev. Genet. <em>6</em>, 805–814. Available at: <a href="http://www.nature.com/doifinder/10.1038/nrg1709" class="uri">http://www.nature.com/doifinder/10.1038/nrg1709</a>.</p>
</div>
<div>
<p>25. Hugenholtz, P., and Tyson, G.W. (2008). Microbiology: Metagenomics. Nature <em>455</em>, 481–483. Available at: <a href="http://www.nature.com/doifinder/10.1038/455481a" class="uri">http://www.nature.com/doifinder/10.1038/455481a</a>.</p>
</div>
<div>
<p>26. Wooley, J.C., Godzik, A., and Friedberg, I. (2010). A primer on metagenomics. PLoS Comput. Biol. <em>6</em>, e1000667. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20195499 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2829047" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/20195499 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2829047</a>.</p>
</div>
<div>
<p>27. Rinke, C., Schwientek, P., Sczyrba, A., Ivanova, N.N., Anderson, I.J., Cheng, J.-F., Darling, A., Malfatti, S., Swan, B.K., and Gies, E.A. <em>et al.</em> (2013). Insights into the phylogeny and coding potential of microbial dark matter. Nature <em>499</em>, 431–437. Available at: <a href="http://www.nature.com/doifinder/10.1038/nature12352" class="uri">http://www.nature.com/doifinder/10.1038/nature12352</a>.</p>
</div>
<div>
<p>28. Mukherjee, S., Seshadri, R., Varghese, N.J., Eloe-Fadrosh, E.A., Meier-Kolthoff, J.P., Göker, M., Coates, R.C., Hadjithomas, M., Pavlopoulos, G.A., and Paez-Espino, D. <em>et al.</em> (2017). 1,003 reference genomes of bacterial and archaeal isolates expand coverage of the tree of life. Nat. Biotechnol. <em>35</em>, 676–683. Available at: <a href="http://www.nature.com/doifinder/10.1038/nbt.3886" class="uri">http://www.nature.com/doifinder/10.1038/nbt.3886</a>.</p>
</div>
<div>
<p>29. Forster, S.C. (2017). Illuminating microbial diversity. Nat. Rev. Microbiol. <em>15</em>, 578–578. Available at: <a href="http://www.nature.com/doifinder/10.1038/nrmicro.2017.106" class="uri">http://www.nature.com/doifinder/10.1038/nrmicro.2017.106</a>.</p>
</div>
<div>
<p>30. Zerihun, M.B., and Schug, A. (2017). Biomolecular coevolution and its applications: Going from structure prediction toward signaling, epistasis, and function. Biochem. Soc. Trans., BST20170063. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/29054926" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/29054926</a>.</p>
</div>
<div>
<p>31. Clabbers, M.T.B., Genderen, E. van, Wan, W., Wiegers, E.L., Gruene, T., and Abrahams, J.P. (2017). Protein structure determination by electron diffraction using a single three-dimensional nanocrystal. Acta Crystallogr. Sect. D, Struct. Biol. <em>73</em>, 738–748. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28876237 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5586247" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/28876237 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5586247</a>.</p>
</div>
<div>
<p>32. Dukka, B.K. (2016). Recent advances in sequence-based protein structure prediction. Brief. Bioinform. <em>31</em>, 1–12. Available at: <a href="https://academic.oup.com/bib/article-lookup/doi/10.1093/bib/bbw070" class="uri">https://academic.oup.com/bib/article-lookup/doi/10.1093/bib/bbw070</a>.</p>
</div>
<div>
<p>33. Li, W., Zhang, Y., and Skolnick, J. (2004). Application of sparse NMR restraints to large-scale protein structure prediction. Biophys. J. <em>87</em>, 1241–8. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15298926 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC1304462" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/15298926 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC1304462</a>.</p>
</div>
<div>
<p>34. Walzthoeni, T., Leitner, A., Stengel, F., and Aebersold, R. (2013). Mass spectrometry supported determination of protein complex structure. Curr. Opin. Struct. Biol. <em>23</em>, 252–260. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23522702 http://linkinghub.elsevier.com/retrieve/pii/S0959440X13000377" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/23522702 http://linkinghub.elsevier.com/retrieve/pii/S0959440X13000377</a>.</p>
</div>
<div>
<p>35. Rappsilber, J. (2011). The beginning of a beautiful friendship: cross-linking/mass spectrometry and modelling of proteins and multi-protein complexes. J. Struct. Biol. <em>173</em>, 530–40. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21029779 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3043253" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/21029779 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3043253</a>.</p>
</div>
<div>
<p>36. Ornes, S. (2016). Let the structural symphony begin. Nature <em>536</em>, 361–363. Available at: <a href="http://www.nature.com/doifinder/10.1038/536361a" class="uri">http://www.nature.com/doifinder/10.1038/536361a</a>.</p>
</div>
<div>
<p>37. Ward, A.B., Sali, A., and Wilson, I.A. (2013). Biochemistry. Integrative structural biology. Science <em>339</em>, 913–5. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23430643 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3633482" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/23430643 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3633482</a>.</p>
</div>
<div>
<p>38. Tang, Y., Huang, Y.J., Hopf, T.A., Sander, C., Marks, D.S., and Montelione, G.T. (2015). Protein structure determination by combining sparse NMR data with evolutionary couplings. Nat. Methods <em>advance on</em>. Available at: <a href="http://dx.doi.org/10.1038/nmeth.3455" class="uri">http://dx.doi.org/10.1038/nmeth.3455</a>.</p>
</div>
<div>
<p>39. Marks, D.S., Colwell, L.J., Sheridan, R., Hopf, T.A., Pagnani, A., Zecchina, R., and Sander, C. (2011). Protein 3D structure computed from evolutionary sequence variation. PLoS One <em>6</em>, e28766. Available at: <a href="http://dx.plos.org/10.1371/journal.pone.0028766" class="uri">http://dx.plos.org/10.1371/journal.pone.0028766</a>.</p>
</div>
<div>
<p>40. Vendruscolo, M., Kussell, E., and Domany, E. (1997). Recovery of protein structure from contact maps. Fold. Des. <em>2</em>, 295–306. Available at: <a href="http://dx.doi.org/10.1016/S1359-0278(97)00041-2" class="uri">http://dx.doi.org/10.1016/S1359-0278(97)00041-2</a>.</p>
</div>
<div>
<p>41. Kim, D.E., Dimaio, F., Yu-Ruei Wang, R., Song, Y., and Baker, D. (2014). One contact for every twelve residues allows robust and accurate topology-level protein structure modeling. Proteins <em>82 Suppl 2</em>, 208–18. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23900763" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/23900763</a>.</p>
</div>
<div>
<p>42. Duarte, J.M., Sathyapriya, R., Stehr, H., Filippis, I., and Lappe, M. (2010). Optimal contact definition for reconstruction of contact maps. BMC Bioinformatics <em>11</em>, 283. Available at: <a href="http://www.biomedcentral.com/1471-2105/11/283" class="uri">http://www.biomedcentral.com/1471-2105/11/283</a>.</p>
</div>
<div>
<p>43. Sadowski, M.I. (2013). Prediction of protein domain boundaries from inverse covariances. Proteins <em>81</em>, 253–60. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22987736 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3563215" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/22987736 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3563215</a>.</p>
</div>
<div>
<p>44. Parisi, G., Zea, D.J., Monzon, A.M., and Marino-Buslje, C. (2015). Conformational diversity and the emergence of sequence signatures during evolution. Curr. Opin. Struct. Biol. <em>32C</em>, 58–65. Available at: <a href="http://www.sciencedirect.com/science/article/pii/S0959440X15000147" class="uri">http://www.sciencedirect.com/science/article/pii/S0959440X15000147</a>.</p>
</div>
<div>
<p>45. Hopf, T.A., Ingraham, J.B., Poelwijk, F.J., Schärfe, C.P.I., Springer, M., Sander, C., and Marks, D.S. (2017). Mutation effects predicted from sequence co-variation. Nat. Biotechnol. <em>35</em>, 128–135. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28092658 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5383098 http://www.nature.com/doifinder/10.1038/nbt.3769" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/28092658 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5383098 http://www.nature.com/doifinder/10.1038/nbt.3769</a>.</p>
</div>
<div>
<p>46. Göbel, U., Sander, C., Schneider, R., and Valencia, A. (1994). Correlated mutations and residue contacts in proteins. Proteins <em>18</em>, 309–17. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/8208723" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/8208723</a>.</p>
</div>
<div>
<p>47. Godzik, A., and Sander, C. (1989). Conservation of residue interactions in a family of Ca-binding proteins. &quot;Protein Eng. Des. Sel. <em>2</em>, 589–596. Available at: <a href="https://academic.oup.com/peds/article-lookup/doi/10.1093/protein/2.8.589" class="uri">https://academic.oup.com/peds/article-lookup/doi/10.1093/protein/2.8.589</a>.</p>
</div>
<div>
<p>48. Neher, E. (1994). How frequent are correlated changes in families of protein sequences? Proc. Natl. Acad. Sci. U. S. A. <em>91</em>, 98–102. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=42893{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=42893{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>49. Taylor, W.R., and Hatrick, K. (1994). Compensating changes in protein multiple sequence alignments. &quot;Protein Eng. Des. Sel. <em>7</em>, 341–348. Available at: <a href="http://peds.oxfordjournals.org/content/7/3/341.abstract" class="uri">http://peds.oxfordjournals.org/content/7/3/341.abstract</a>.</p>
</div>
<div>
<p>50. Oliveira, L., Paiva, A.C.M., and Vriend, G. (2002). Correlated mutation analyses on very large sequence families. Chembiochem <em>3</em>, 1010–7. Available at: <a href="http://onlinelibrary.wiley.com/doi/10.1002/1439-7633(20021004)3:10{\%}3C1010::AID-CBIC1010{\%}3E3.0.CO;2-T/full" class="uri">http://onlinelibrary.wiley.com/doi/10.1002/1439-7633(20021004)3:10{\%}3C1010::AID-CBIC1010{\%}3E3.0.CO;2-T/full</a>.</p>
</div>
<div>
<p>51. Shindyalov, I., Kolchanov, N., and Sander, C. (1994). Can three-dimensional contacts in protein structures be predicted by analysis of correlated mutations? &quot;Protein Eng. Des. Sel. <em>7</em>, 349–358. Available at: <a href="http://peds.oxfordjournals.org/content/7/3/349" class="uri">http://peds.oxfordjournals.org/content/7/3/349</a>.</p>
</div>
<div>
<p>52. Clarke, N.D. (1995). Covariation of residues in the homeodomain sequence family. Protein Sci. <em>4</em>, 2269–78. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2143025{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2143025{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>53. Korber, B. (1993). Covariation of Mutations in the V3 Loop of Human Immunodeficiency Virus Type 1 Envelope Protein: An Information Theoretic Analysis. Proc. Natl. Acad. Sci. <em>90</em>, 7176–7180. Available at: <a href="http://www.pnas.org/content/90/15/7176.abstract?ijkey=6e1c9cbdef66bd0beefedc88ea07591b837f2213{\&amp;}keytype2=tf{\_}ipsecsha" class="uri">http://www.pnas.org/content/90/15/7176.abstract?ijkey=6e1c9cbdef66bd0beefedc88ea07591b837f2213{\&amp;}keytype2=tf{\_}ipsecsha</a>.</p>
</div>
<div>
<p>54. Martin, L.C., Gloor, G.B., Dunn, S.D., and Wahl, L.M. (2005). Using information theory to search for co-evolving residues in proteins. Bioinformatics <em>21</em>, 4116–24. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/21/22/4116.full" class="uri">http://bioinformatics.oxfordjournals.org/content/21/22/4116.full</a>.</p>
</div>
<div>
<p>55. Atchley, W.R., Wollenberg, K.R., Fitch, W.M., Terhalle, W., and Dress, A.W. (2000). Correlations Among Amino Acid Sites in bHLH Protein Domains: An Information Theoretic Analysis. Mol. Biol. Evol. <em>17</em>, 164–178. Available at: <a href="http://mbe.oxfordjournals.org/content/17/1/164.abstract?ijkey=2a1f0a044a8fd2213955e4f2c17fa84682dd2571{\&amp;}keytype2=tf{\_}ipsecsha" class="uri">http://mbe.oxfordjournals.org/content/17/1/164.abstract?ijkey=2a1f0a044a8fd2213955e4f2c17fa84682dd2571{\&amp;}keytype2=tf{\_}ipsecsha</a>.</p>
</div>
<div>
<p>56. Fodor, A.A., and Aldrich, R.W. (2004). Influence of conservation on calculations of amino acid covariance in multiple sequence alignments. Proteins <em>56</em>, 211–21. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15211506" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/15211506</a>.</p>
</div>
<div>
<p>57. Tillier, E.R., and Lui, T.W. (2003). Using multiple interdependency to separate functional from phylogenetic correlations in protein alignments. Bioinformatics <em>19</em>, 750–755. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/19/6/750.abstract?ijkey=d19c5a2492bfa46d1a02d661c29a610113da8db3{\&amp;}keytype2=tf{\_}ipsecsha" class="uri">http://bioinformatics.oxfordjournals.org/content/19/6/750.abstract?ijkey=d19c5a2492bfa46d1a02d661c29a610113da8db3{\&amp;}keytype2=tf{\_}ipsecsha</a>.</p>
</div>
<div>
<p>58. Gouveia-Oliveira, R., and Pedersen, A.G. (2007). Finding coevolving amino acid residues using row and column weighting of mutual information and multi-dimensional amino acid representation. Algorithms Mol. Biol. <em>2</em>, 12. Available at: <a href="http://www.almob.org/content/2/1/12" class="uri">http://www.almob.org/content/2/1/12</a>.</p>
</div>
<div>
<p>59. Dunn, S.D., Wahl, L.M., and Gloor, G.B. (2008). Mutual information without the influence of phylogeny or entropy dramatically improves residue contact prediction. Bioinformatics <em>24</em>, 333–40. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/24/3/333" class="uri">http://bioinformatics.oxfordjournals.org/content/24/3/333</a>.</p>
</div>
<div>
<p>60. Kass, I., and Horovitz, A. (2002). Mapping pathways of allosteric communication in GroEL by analysis of correlated mutations. Proteins <em>48</em>, 611–7. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/12211028" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/12211028</a>.</p>
</div>
<div>
<p>61. Noivirt, O., Eisenstein, M., and Horovitz, A. (2005). Detection and reduction of evolutionary noise in correlated mutation analysis. Protein Eng. Des. Sel. <em>18</em>, 247–53. Available at: <a href="http://peds.oxfordjournals.org/content/18/5/247.full" class="uri">http://peds.oxfordjournals.org/content/18/5/247.full</a>.</p>
</div>
<div>
<p>62. Lapedes, A., Giraud, B., Liu, L., and Stormo, G. (1999). Correlated mutations in models of protein sequences: phylogenetic and structural effects. <em>33</em>, 236–256. Available at: <a href="http://www.citeulike.org/user/qluo/article/5092214" class="uri">http://www.citeulike.org/user/qluo/article/5092214</a>.</p>
</div>
<div>
<p>63. Burger, L., and Nimwegen, E. van (2010). Disentangling direct from indirect co-evolution of residues in protein alignments. PLoS Comput. Biol. <em>6</em>, e1000633. Available at: <a href="http://dx.plos.org/10.1371/journal.pcbi.1000633" class="uri">http://dx.plos.org/10.1371/journal.pcbi.1000633</a>.</p>
</div>
<div>
<p>64. Weigt, M., White, R.A., Szurmant, H., Hoch, J.A., and Hwa, T. (2009). Identification of direct residue contacts in protein-protein interaction by message passing. Proc. Natl. Acad. Sci. U. S. A. <em>106</em>, 67–72. Available at: <a href="http://www.pnas.org/content/106/1/67.abstract" class="uri">http://www.pnas.org/content/106/1/67.abstract</a>.</p>
</div>
<div>
<p>65. Juan, D. de, Pazos, F., and Valencia, A. (2013). Emerging methods in protein co-evolution. Nat. Rev. Genet. <em>14</em>, 249–61. Available at: <a href="http://www.readcube.com/articles/10.1038/nrg3414?locale=en" class="uri">http://www.readcube.com/articles/10.1038/nrg3414?locale=en</a>.</p>
</div>
<div>
<p>66. Jones, D.T., Buchan, D.W.A., Cozzetto, D., and Pontil, M. (2012). PSICOV: precise structural contact prediction using sparse inverse covariance estimation on large multiple sequence alignments. Bioinformatics <em>28</em>, 184–90. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/28/2/184.full" class="uri">http://bioinformatics.oxfordjournals.org/content/28/2/184.full</a>.</p>
</div>
<div>
<p>67. Burger, L., and Nimwegen, E. van (2008). Accurate prediction of protein-protein interactions from sequence alignments using a Bayesian method. Mol. Syst. Biol. <em>4</em>, 165. Available at: <a href="http://msb.embopress.org/content/4/1/165.abstract" class="uri">http://msb.embopress.org/content/4/1/165.abstract</a>.</p>
</div>
<div>
<p>68. Cheng, J., and Baldi, P. (2007). Improved residue contact prediction using support vector machines and a large feature set. BMC Bioinformatics <em>8</em>. Available at: <a href="http://dx.doi.org/10.1186/1471-2105-8-113" class="uri">http://dx.doi.org/10.1186/1471-2105-8-113</a>.</p>
</div>
<div>
<p>69. Wu, S., and Zhang, Y. (2008). A comprehensive assessment of sequence-based and template-based methods for protein contact prediction. Bioinformatics <em>24</em>, 924–31. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2648832{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2648832{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>70. Li, Y., Fang, Y., and Fang, J. (2011). Predicting residue-residue contacts using random forest models. Bioinformatics <em>27</em>. Available at: <a href="http://dx.doi.org/10.1093/bioinformatics/btr579" class="uri">http://dx.doi.org/10.1093/bioinformatics/btr579</a>.</p>
</div>
<div>
<p>71. Wang, X.-F., Chen, Z., Wang, C., Yan, R.-X., Zhang, Z., and Song, J. (2011). Predicting residue-residue contacts and helix-helix interactions in transmembrane proteins using an integrative feature-based random forest approach. PLoS One <em>6</em>, e26767. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3203928{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3203928{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>72. Wang, Z., and Xu, J. (2013). Predicting protein contact map using evolutionary and physical constraints by integer programming. Bioinformatics <em>29</em>, i266–73. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3694661{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3694661{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>73. Fariselli, P., Olmea, O., Valencia, A., and Casadio, R. (2001). Prediction of contact maps with neural networks and correlated mutations. Protein Eng. Des. Sel. <em>14</em>, 835–843. Available at: <a href="http://peds.oxfordjournals.org/content/14/11/835.long" class="uri">http://peds.oxfordjournals.org/content/14/11/835.long</a>.</p>
</div>
<div>
<p>74. Shackelford, G., and Karplus, K. (2007). Contact prediction using mutual information and neural nets. Proteins <em>69 Suppl 8</em>, 159–64. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17932918" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/17932918</a>.</p>
</div>
<div>
<p>75. Hamilton, N., Burrage, K., Ragan, M.A., and Huber, T. (2004). Protein contact prediction using patterns of correlation. Proteins Struct. Funct. Bioinforma. <em>56</em>, 679–684. Available at: <a href="http://onlinelibrary.wiley.com/doi/10.1002/prot.20160/abstract" class="uri">http://onlinelibrary.wiley.com/doi/10.1002/prot.20160/abstract</a>.</p>
</div>
<div>
<p>76. Xue, B., Faraggi, E., and Zhou, Y. (2009). Predicting residue-residue contact maps by a two-layer, integrated neural-network method. Proteins <em>76</em>, 176–83. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2716487{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2716487{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>77. Tegge, A.N., Wang, Z., Eickholt, J., and Cheng, J. (2009). NNcon: improved protein contact map prediction using 2D-recursive neural networks. Nucleic Acids Res. <em>37</em>, W515–8. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2703959{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2703959{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>78. Eickholt, J., and Cheng, J. (2012). Predicting protein residue–residue contacts using deep networks and boosting. <em>28</em>, 3066–3072.</p>
</div>
<div>
<p>79. Di Lena, P., Nagata, K., and Baldi, P. (2012). Deep architectures for protein contact map prediction. Bioinformatics <em>28</em>, 2449–57. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/28/19/2449.full{\#}sec-14" class="uri">http://bioinformatics.oxfordjournals.org/content/28/19/2449.full{\#}sec-14</a>.</p>
</div>
<div>
<p>80. Chen, P., and Li, J. (2010). Prediction of protein long-range contacts using an ensemble of genetic algorithm classifiers with sequence profile centers. BMC Struct. Biol. <em>10 Suppl 1</em>, S2. Available at: <a href="http://bmcstructbiol.biomedcentral.com/articles/10.1186/1472-6807-10-S1-S2" class="uri">http://bmcstructbiol.biomedcentral.com/articles/10.1186/1472-6807-10-S1-S2</a>.</p>
</div>
<div>
<p>81. Jones, D.T., Singh, T., Kosciolek, T., and Tetchner, S. (2015). MetaPSICOV: combining coevolution methods for accurate prediction of contacts and long range hydrogen bonding in proteins. Bioinformatics <em>31</em>, 999–1006. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/31/7/999.short" class="uri">http://bioinformatics.oxfordjournals.org/content/31/7/999.short</a>.</p>
</div>
<div>
<p>82. Skwark, M.J., Abdel-Rehim, A., and Elofsson, A. (2013). PconsC: combination of direct information methods and alignments improves contact prediction. Bioinformatics <em>29</em>, 1815–6. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/29/14/1815.long" class="uri">http://bioinformatics.oxfordjournals.org/content/29/14/1815.long</a>.</p>
</div>
<div>
<p>83. Skwark, M.J., Michel, M., Menendez Hurtado, D., Ekeberg, M., and Elofsson, A. (2016). Accurate contact predictions for thousands of protein families using PconsC3. bioRxiv.</p>
</div>
<div>
<p>84. Schneider, M., and Brock, O. (2014). Combining Physicochemical and Evolutionary Information for Protein Contact Prediction. PLoS One <em>9</em>, e108438. Available at: <a href="http://www.plosone.org/article/info{\%}3Adoi{\%}2F10.1371{\%}2Fjournal.pone.0108438{\#}abstract0" class="uri">http://www.plosone.org/article/info{\%}3Adoi{\%}2F10.1371{\%}2Fjournal.pone.0108438{\#}abstract0</a>.</p>
</div>
<div>
<p>85. Jones, D.T., Singh, T., Kosciolek, T., and Tetchner, S. (2015). MetaPSICOV: combining coevolution methods for accurate prediction of contacts and long range hydrogen bonding in proteins. Bioinformatics <em>31</em>, 999–1006. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/31/7/999.short" class="uri">http://bioinformatics.oxfordjournals.org/content/31/7/999.short</a>.</p>
</div>
<div>
<p>86. Wang, S., Sun, S., Li, Z., Zhang, R., and Xu, J. (2016). Accurate De Novo Prediction of Protein Contact Map by Ultra-Deep Learning Model. PLoS Comput. Biol. <em>13</em>, e1005324. Available at: <a href="http://arxiv.org/abs/1609.00680 http://www.ncbi.nlm.nih.gov/pubmed/28056090 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5249242" class="uri">http://arxiv.org/abs/1609.00680 http://www.ncbi.nlm.nih.gov/pubmed/28056090 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5249242</a>.</p>
</div>
<div>
<p>87. Stahl, K., Schneider, M., and Brock, O. (2017). EPSILON-CP: using deep learning to combine information from multiple sources for protein contact prediction. BMC Bioinformatics <em>18</em>, 303. Available at: <a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1713-x" class="uri">http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1713-x</a>.</p>
</div>
<div>
<p>88. He, B., Mortuza, S.M., Wang, Y., Shen, H.-B., and Zhang, Y. (2017). NeBcon: Protein contact map prediction using neural network training coupled with naïve Bayes classifiers. Bioinformatics. Available at: <a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btx164" class="uri">https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btx164</a>.</p>
</div>
<div>
<p>89. Andreani, J., and Söding, J. (2015). Bbcontacts: Prediction of <span class="math inline">\(\beta\)</span>-strand pairing from direct coupling patterns. Bioinformatics <em>31</em>, 1729–37. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/31/11/1729" class="uri">http://bioinformatics.oxfordjournals.org/content/31/11/1729</a>.</p>
</div>
<div>
<p>90. Skwark, M.J., Raimondi, D., Michel, M., and Elofsson, A. (2014). Improved contact predictions using the recognition of protein like contact patterns. PLoS Comput. Biol. <em>10</em>, e1003889. Available at: <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003889" class="uri">http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003889</a>.</p>
</div>
<div>
<p>91. Monastyrskyy, B., D’Andrea, D., Fidelis, K., Tramontano, A., and Kryshtafovych, A. (2015). New encouraging developments in contact prediction: Assessment of the CASP11 results. Proteins. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26474083" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/26474083</a>.</p>
</div>
<div>
<p>92. Jaynes, E.T. (1957). Information Theory and Statistical Mechanics I. Phys. Rev. <em>106</em>, 620–630. Available at: <a href="https://link.aps.org/doi/10.1103/PhysRev.106.620" class="uri">https://link.aps.org/doi/10.1103/PhysRev.106.620</a>.</p>
</div>
<div>
<p>93. Jaynes, E.T. (1957). Information Theory and Statistical Mechanics. II. Phys. Rev. <em>108</em>, 171–190. Available at: <a href="https://link.aps.org/doi/10.1103/PhysRev.108.171" class="uri">https://link.aps.org/doi/10.1103/PhysRev.108.171</a>.</p>
</div>
<div>
<p>94. Wainwright, M.J., and Jordan, M.I. (2007). Graphical Models, Exponential Families, and Variational Inference. Found. Trends Mach. Learn. <em>1</em>, 1–305. Available at: <a href="http://www.nowpublishers.com/article/Details/MAL-001" class="uri">http://www.nowpublishers.com/article/Details/MAL-001</a>.</p>
</div>
<div>
<p>95. Murphy, K.P. (2012). Machine Learning: A Probabilistic Perspective (MIT Press).</p>
</div>
<div>
<p>96. Morcos, F., Pagnani, A., Lunt, B., Bertolino, A., Marks, D.S., Sander, C., Zecchina, R., Onuchic, J.N., Hwa, T., and Weigt, M. (2011). Direct-coupling analysis of residue coevolution captures native contacts across many protein families. Proc. Natl. Acad. Sci. U. S. A. <em>108</em>, E1293–301. Available at: <a href="http://www.pnas.org/content/108/49/E1293.full" class="uri">http://www.pnas.org/content/108/49/E1293.full</a>.</p>
</div>
<div>
<p>97. Cocco, S., Feinauer, C., Figliuzzi, M., Monasson, R., and Weigt, M. (2017). Inverse Statistical Physics of Protein Sequences: A Key Issues Review. arXiv. Available at: <a href="https://arxiv.org/pdf/1703.01222.pdf" class="uri">https://arxiv.org/pdf/1703.01222.pdf</a>.</p>
</div>
<div>
<p>98. Koller, D., and Friedman, N.I.R. (2009). Probabilistic graphical models: Principles and Techniques (MIT Press).</p>
</div>
<div>
<p>99. Ekeberg, M., Lövkvist, C., Lan, Y., Weigt, M., and Aurell, E. (2013). Improved contact prediction in proteins: Using pseudolikelihoods to infer Potts models. Phys. Rev. E <em>87</em>, 012707. Available at: <a href="http://link.aps.org/doi/10.1103/PhysRevE.87.012707" class="uri">http://link.aps.org/doi/10.1103/PhysRevE.87.012707</a>.</p>
</div>
<div>
<p>100. Stein, R.R., Marks, D.S., and Sander, C. (2015). Inferring Pairwise Interactions from Biological Data Using Maximum-Entropy Probability Models. PLOS Comput. Biol. <em>11</em>, e1004182. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=4520494{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=4520494{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>101. Seemayer, S., Gruber, M., and Söding, J. (2014). CCMpred-fast and precise prediction of protein residue-residue contacts from correlated mutations. Bioinformatics, btu500. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/early/2014/08/12/bioinformatics.btu500" class="uri">http://bioinformatics.oxfordjournals.org/content/early/2014/08/12/bioinformatics.btu500</a>.</p>
</div>
<div>
<p>102. Ekeberg, M., Hartonen, T., and Aurell, E. (2014). Fast pseudolikelihood maximization for direct-coupling analysis of protein structure from many homologous amino-acid sequences. J. Comput. Phys. <em>276</em>, 341–356. Available at: <a href="http://www.sciencedirect.com/science/article/pii/S0021999114005178" class="uri">http://www.sciencedirect.com/science/article/pii/S0021999114005178</a>.</p>
</div>
<div>
<p>103. Kamisetty, H., Ovchinnikov, S., and Baker, D. (2013). Assessing the utility of coevolution-based residue-residue contact predictions in a sequence- and structure-rich era. Proc. Natl. Acad. Sci. U. S. A. <em>110</em>, 15674–9. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3785744{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3785744{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>104. Lapedes, A., Giraud, B., and Jarzynski, C. (2012). Using Sequence Alignments to Predict Protein Structure and Stability With High Accuracy. Available at: <a href="http://arxiv.org/abs/1207.2484" class="uri">http://arxiv.org/abs/1207.2484</a>.</p>
</div>
<div>
<p>105. Balakrishnan, S., Kamisetty, H., Carbonell, J.G., Lee, S.-I., and Langmead, C.J. (2011). Learning generative models for protein fold families. Proteins <em>79</em>, 1061–78. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21268112" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/21268112</a>.</p>
</div>
<div>
<p>106. Friedman, J., Hastie, T., and Tibshirani, R. (2008). Sparse inverse covariance estimation with the graphical lasso. Biostatistics <em>9</em>, 432–41. Available at: <a href="http://biostatistics.oxfordjournals.org/content/9/3/432.abstract" class="uri">http://biostatistics.oxfordjournals.org/content/9/3/432.abstract</a>.</p>
</div>
<div>
<p>107. Banerjee, O., El Ghaoui, L., and D’Aspremont, A. (2008). Model Selection Through Sparse Maximum Likelihood Estimation for Multivariate Gaussian or Binary Data. J. Mach. Learn. Res. <em>9</em>, 485–516. Available at: <a href="http://dl.acm.org/citation.cfm?id=1390681.1390696" class="uri">http://dl.acm.org/citation.cfm?id=1390681.1390696</a>.</p>
</div>
<div>
<p>108. Baldassi, C., Zamparo, M., Feinauer, C., Procaccini, A., Zecchina, R., Weigt, M., and Pagnani, A. (2014). Fast and accurate multivariate gaussian modeling of protein families: predicting residue contacts and protein-interaction partners. PLoS One <em>9</em>, e92721. Available at: <a href="http://dx.plos.org/10.1371/journal.pone.0092721" class="uri">http://dx.plos.org/10.1371/journal.pone.0092721</a>.</p>
</div>
<div>
<p>109. Besag, J. (1975). Statistical Analysis of Non-Lattice Data. Source Stat. <em>24</em>, 179–195. Available at: <a href="http://www.jstor.org http://www.jstor.org/stable/2987782" class="uri">http://www.jstor.org http://www.jstor.org/stable/2987782</a>.</p>
</div>
<div>
<p>110. Gidas, B. (1988). Consistency of maximum likelihood and pseudo-likelihood estimators for Gibbs Distributions. Stoch. Differ. Syst. Stoch. Control Theory Appl. Available at: <a href="http://www.researchgate.net/publication/244456377{\_}Consistency{\_}of{\_}maximum{\_}likelihood{\_}and{\_}pseudo-likelihood{\_}estimators{\_}for{\_}Gibbs{\_}Distributions" class="uri">http://www.researchgate.net/publication/244456377{\_}Consistency{\_}of{\_}maximum{\_}likelihood{\_}and{\_}pseudo-likelihood{\_}estimators{\_}for{\_}Gibbs{\_}Distributions</a>.</p>
</div>
<div>
<p>111. Feinauer, C., Skwark, M.J., Pagnani, A., and Aurell, E. (2014). Improving contact prediction along three dimensions. 19. Available at: <a href="http://arxiv.org/abs/1403.0379" class="uri">http://arxiv.org/abs/1403.0379</a>.</p>
</div>
<div>
<p>112. Zhang, H., Huang, Q., Bei, Z., Wei, Y., and Floudas, C.A. (2016). COMSAT: Residue contact prediction of transmembrane proteins based on support vector machines and mixed integer linear programming. Proteins Struct. Funct. Bioinforma., n/a–n/a. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26756402" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/26756402</a>.</p>
</div>
<div>
<p>113. Yu, X., Wu, X., Bermejo, G.A., Brooks, B.R., and Taraska, J.W. (2013). Accurate high-throughput structure mapping and prediction with transition metal ion FRET. Structure <em>21</em>, 9–19. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23273426 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3700372" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/23273426 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3700372</a>.</p>
</div>
<div>
<p>114. Kalinin, S., Peulen, T., Sindbert, S., Rothwell, P.J., Berger, S., Restle, T., Goody, R.S., Gohlke, H., and Seidel, C.A.M. (2012). A toolkit and benchmark study for FRET-restrained high-precision structural modeling. Nat. Methods <em>9</em>, 1218–1225. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23142871 http://www.nature.com/doifinder/10.1038/nmeth.2222" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/23142871 http://www.nature.com/doifinder/10.1038/nmeth.2222</a>.</p>
</div>
<div>
<p>115. Bowers, P.M., Strauss, C.E., and Baker, D. (2000). De novo protein structure determination using sparse NMR data. J. Biomol. NMR <em>18</em>, 311–8. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11200525" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/11200525</a>.</p>
</div>
<div>
<p>116. Kolinski, A., and Skolnick, J. (1998). Assembly of protein structure from sparse experimental data: An efficient Monte Carlo model. Proteins Struct. Funct. Genet. <em>32</em>, 475–494. Available at: <a href="http://doi.wiley.com/10.1002/{\%}28SICI{\%}291097-0134{\%}2819980901{\%}2932{\%}3A4{\%}3C475{\%}3A{\%}3AAID-PROT6{\%}3E3.0.CO{\%}3B2-F" class="uri">http://doi.wiley.com/10.1002/{\%}28SICI{\%}291097-0134{\%}2819980901{\%}2932{\%}3A4{\%}3C475{\%}3A{\%}3AAID-PROT6{\%}3E3.0.CO{\%}3B2-F</a>.</p>
</div>
<div>
<p>117. Aszódi, A., Taylor, W.R., and Gradwell, M.J. (1995). Global Fold Determination from a Small Number of Distance Restraints. J. Mol. Biol. <em>251</em>, 308–326. Available at: <a href="http://www.sciencedirect.com/science/article/pii/S0022283685704366?via{\%}3Dihub" class="uri">http://www.sciencedirect.com/science/article/pii/S0022283685704366?via{\%}3Dihub</a>.</p>
</div>
<div>
<p>118. Wu, S., Szilagyi, A., and Zhang, Y. (2011). Improving protein structure prediction using multiple sequence-based contact predictions. Structure <em>19</em>, 1182–91. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3154634{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3154634{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>119. Tress, M.L., and Valencia, A. (2010). Predicted residue-residue contacts can help the scoring of 3D models. Proteins Struct. Funct. Bioinforma. <em>78</em>, NA–NA. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20408174 http://doi.wiley.com/10.1002/prot.22714" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/20408174 http://doi.wiley.com/10.1002/prot.22714</a>.</p>
</div>
<div>
<p>120. Hopf, T.A., Colwell, L.J., Sheridan, R., Rost, B., Sander, C., and Marks, D.S. (2012). Three-dimensional structures of membrane proteins from genomic sequencing. Cell <em>149</em>, 1607–21. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3641781{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3641781{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>121. Ovchinnikov, S., Kamisetty, H., and Baker, D. (2014). Robust and accurate prediction of residue-residue interactions across protein interfaces using evolutionary information. Elife <em>3</em>, e02030. Available at: <a href="http://elifesciences.org/content/3/e02030.abstract" class="uri">http://elifesciences.org/content/3/e02030.abstract</a>.</p>
</div>
<div>
<p>122. Hopf, T.A., Schärfe, C.P.I., Rodrigues, J.P.G.L.M., Green, A.G., Sander, C., Bonvin, A.M.J.J., and Marks, D.S. (2014). Sequence co-evolution gives 3D contacts and structures of protein complexes. Available at: <a href="http://arxiv.org/abs/1405.0929" class="uri">http://arxiv.org/abs/1405.0929</a>.</p>
</div>
<div>
<p>123. Hayat, S., Sander, C., Marks, D.S., and Elofsson, A. (2015). All-atom 3D structure prediction of transmembrane <span class="math inline">\(\beta\)</span>-barrel proteins from sequences. Proc. Natl. Acad. Sci. U. S. A. <em>112</em>, 5413–8. Available at: <a href="http://www.pnas.org/content/112/17/5413.short" class="uri">http://www.pnas.org/content/112/17/5413.short</a>.</p>
</div>
<div>
<p>124. Hopf, T.A., Morinaga, S., Ihara, S., Touhara, K., Marks, D.S., and Benton, R. (2015). Amino acid coevolution reveals three-dimensional structure and functional domains of insect odorant receptors. Nat. Commun. <em>6</em>, 6077. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=4364406{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=4364406{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>125. Raval, A., Piana, S., Eastwood, M.P., and Shaw, D.E. (2015). Assessment of the utility of contact-based restraints in accelerating the prediction of protein structure using molecular dynamics simulations. Protein Sci. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26266489" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/26266489</a>.</p>
</div>
<div>
<p>126. Wang, Y., and Barth, P. (2015). Evolutionary-guided de novo structure prediction of self-associated transmembrane helical proteins with near-atomic accuracy. Nat. Commun. <em>6</em>, 7196. Available at: <a href="http://www.nature.com/ncomms/2015/150521/ncomms8196/abs/ncomms8196.html" class="uri">http://www.nature.com/ncomms/2015/150521/ncomms8196/abs/ncomms8196.html</a>.</p>
</div>
<div>
<p>127. Ovchinnikov, S., Kinch, L., Park, H., Liao, Y., Pei, J., Kim, D.E., Kamisetty, H., Grishin, N.V., and Baker, D. (2015). Large scale determination of previously unsolved protein structures using evolutionary information. Elife <em>4</em>, e09248. Available at: <a href="http://elifesciences.org/content/early/2015/09/03/eLife.09248.abstract" class="uri">http://elifesciences.org/content/early/2015/09/03/eLife.09248.abstract</a>.</p>
</div>
<div>
<p>128. Ovchinnikov, S., Park, H., Varghese, N., Huang, P.-S., Pavlopoulos, G.A., Kim, D.E., Kamisetty, H., Kyrpides, N.C., and Baker, D. (2017). Protein structure determination using metagenome sequence data. Science <em>355</em>, 294–298. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28104891 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5493203" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/28104891 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5493203</a>.</p>
</div>
<div>
<p>129. Bhattacharya, D., Cao, R., and Cheng, J. (2016). UniCon3D: de novo protein structure prediction using united-residue conformational search via stepwise, probabilistic sampling. Bioinformatics, btw316. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/27259540" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/27259540</a>.</p>
</div>
<div>
<p>130. Braun, T., Koehler Leman, J., and Lange, O.F. (2015). Combining Evolutionary Information and an Iterative Sampling Strategy for Accurate Protein Structure Prediction. PLoS Comput. Biol. <em>11</em>, e1004661. Available at: <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004661" class="uri">http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004661</a>.</p>
</div>
<div>
<p>131. Mabrouk, M., Putz, I., Werner, T., Schneider, M., Neeb, M., Bartels, P., and Brock, O. (2015). RBO Aleph: leveraging novel information sources for protein structure prediction. Nucleic Acids Res. <em>43</em>, W343–8. Available at: <a href="http://nar.oxfordjournals.org/content/43/W1/W343" class="uri">http://nar.oxfordjournals.org/content/43/W1/W343</a>.</p>
</div>
<div>
<p>132. Pietal, M.J., Bujnicki, J.M., and Kozlowski, L.P. (2015). GDFuzz3D: a method for protein 3D structure reconstruction from contact maps, based on a non-Euclidean distance function. Bioinformatics, btv390. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/early/2015/07/23/bioinformatics.btv390" class="uri">http://bioinformatics.oxfordjournals.org/content/early/2015/07/23/bioinformatics.btv390</a>.</p>
</div>
<div>
<p>133. Michel, M., Hayat, S., Skwark, M.J., Sander, C., Marks, D.S., and Elofsson, A. (2014). PconsFold: improved contact predictions improve protein models. Bioinformatics <em>30</em>, i482–i488. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/30/17/i482.long" class="uri">http://bioinformatics.oxfordjournals.org/content/30/17/i482.long</a>.</p>
</div>
<div>
<p>134. Konopka, B.M., Ciombor, M., Kurczynska, M., and Kotulska, M. (2014). Automated Procedure for Contact-Map-Based Protein Structure Reconstruction. J. Membr. Biol. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24682239" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/24682239</a>.</p>
</div>
<div>
<p>135. Kosciolek, T., and Jones, D.T. (2014). De novo structure prediction of globular proteins aided by sequence variation-derived contacts. PLoS One <em>9</em>, e92197. Available at: <a href="http://dx.plos.org/10.1371/journal.pone.0092197" class="uri">http://dx.plos.org/10.1371/journal.pone.0092197</a>.</p>
</div>
<div>
<p>136. Nugent, T., and Jones, D.T. (2012). Accurate de novo structure prediction of large transmembrane protein domains using fragment-assembly and correlated mutation analysis. Proc. Natl. Acad. Sci. U. S. A. <em>109</em>, E1540–7. Available at: <a href="http://www.pnas.org/content/early/2012/05/21/1120036109" class="uri">http://www.pnas.org/content/early/2012/05/21/1120036109</a>.</p>
</div>
<div>
<p>137. Sathyapriya, R., Duarte, J.M., Stehr, H., Filippis, I., and Lappe, M. (2009). Defining an essence of structure determining residue contacts in proteins. PLoS Comput. Biol. <em>5</em>, e1000584. Available at: <a href="http://dx.plos.org/10.1371/journal.pcbi.1000584" class="uri">http://dx.plos.org/10.1371/journal.pcbi.1000584</a>.</p>
</div>
<div>
<p>138. Chen, Y., Ding, F., and Dokholyan, N.V. (2007). Fidelity of the Protein Structure Reconstruction from Inter-Residue Proximity Constraints. J. Phys. Chem. B <em>111</em>, 7432–7438. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17542631 http://pubs.acs.org/doi/abs/10.1021/jp068963t" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/17542631 http://pubs.acs.org/doi/abs/10.1021/jp068963t</a>.</p>
</div>
<div>
<p>139. Vassura, M., Margara, L., Di Lena, P., Medri, F., Fariselli, P., and Casadio, R. (2007). Reconstruction of 3D structures from protein contact maps. IEEE/ACM Trans. Comput. Biol. Bioinform. <em>5</em>, 357–67. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/18670040" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/18670040</a>.</p>
</div>
<div>
<p>140. Adhikari, B., Bhattacharya, D., Cao, R., and Cheng, J. (2017). Assessing Predicted Contacts for Building Protein Three-Dimensional Models. Methods Mol. Biol. <em>1484</em>, 115–126. Available at: <a href="http://link.springer.com/10.1007/978-1-4939-6406-2{\_}9" class="uri">http://link.springer.com/10.1007/978-1-4939-6406-2{\_}9</a>.</p>
</div>
<div>
<p>141. Di Lena, P., Vassura, M., Margara, L., Fariselli, P., and Casadio, R. (2009). On the Reconstruction of Three-dimensional Protein Structures from Contact Maps. Algorithms <em>2</em>, 76–92. Available at: <a href="http://www.mdpi.com/1999-4893/2/1/76" class="uri">http://www.mdpi.com/1999-4893/2/1/76</a>.</p>
</div>
<div>
<p>142. Zhang, Y., Kolinski, A., and Skolnick, J. (2003). TOUCHSTONE II: a new approach to ab initio protein structure prediction. Biophys. J. <em>85</em>, 1145–64. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=1303233{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=1303233{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>143. Wang, S., Li, W., Zhang, R., Liu, S., and Xu, J. (2016). CoinFold: a web server for protein contact prediction and contact-assisted protein folding. Nucleic Acids Res., gkw307. Available at: <a href="http://nar.oxfordjournals.org/content/early/2016/04/25/nar.gkw307.long" class="uri">http://nar.oxfordjournals.org/content/early/2016/04/25/nar.gkw307.long</a>.</p>
</div>
<div>
<p>144. Adhikari, B., Bhattacharya, D., Cao, R., and Cheng, J. (2015). CONFOLD: Residue-residue contact-guided ab initio protein folding. Proteins <em>83</em>, 1436–49. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25974172" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/25974172</a>.</p>
</div>
<div>
<p>145. Oliveira, S.H.P. de, Shi, J., and Deane, C.M. (2016). Comparing co-evolution methods and their application to template-free protein structure prediction. Bioinformatics, btw618. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/27682254" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/27682254</a>.</p>
</div>
<div>
<p>146. Rodriguez-Rivas, J., Marsili, S., Juan, D., and Valencia, A. (2016). Conservation of coevolving protein interfaces bridges prokaryote-eukaryote homologies in the twilight zone. Proc. Natl. Acad. Sci. U. S. A. <em>113</em>, 15018–15023. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/27965389 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5206536" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/27965389 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5206536</a>.</p>
</div>
<div>
<p>147. Feinauer, C., Szurmant, H., Weigt, M., and Pagnani, A. (2016). Inter-Protein Sequence Co-Evolution Predicts Known Physical Interactions in Bacterial Ribosomes and the Trp Operon. PLoS One <em>11</em>, e0149166. Available at: <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0149166" class="uri">http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0149166</a>.</p>
</div>
<div>
<p>148. Gueudré, T., Baldassi, C., Zamparo, M., Weigt, M., and Pagnani, A. (2016). Simultaneous identification of specifically interacting paralogs and inter-protein contacts by Direct-Coupling Analysis. 19. Available at: <a href="http://arxiv.org/abs/1605.03745" class="uri">http://arxiv.org/abs/1605.03745</a>.</p>
</div>
<div>
<p>149. Bitbol, A.-F., Dwyer, R.S., Colwell, L.J., and Wingreen, N.S. (2016). Inferring interaction partners from protein sequences. Proc. Natl. Acad. Sci. <em>113</em>, 12180–12185. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/27663738 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5087060 http://www.pnas.org/lookup/doi/10.1073/pnas.1606762113" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/27663738 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5087060 http://www.pnas.org/lookup/doi/10.1073/pnas.1606762113</a>.</p>
</div>
<div>
<p>150. Uguzzoni, G., John Lovis, S., Oteri, F., Schug, A., Szurmant, H., and Weigt, M. (2017). Large-scale identification of coevolution signals across homo-oligomeric protein interfaces by direct coupling analysis. Proc. Natl. Acad. Sci. <em>114</em>, E2662–E2671. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28289198 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5380090 http://www.pnas.org/lookup/doi/10.1073/pnas.1615068114" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/28289198 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC5380090 http://www.pnas.org/lookup/doi/10.1073/pnas.1615068114</a>.</p>
</div>
<div>
<p>151. Dos Santos, R.N., Morcos, F., Jana, B., Andricopulo, A.D., and Onuchic, J.N. (2015). Dimeric interactions and complex formation using direct coevolutionary couplings. Sci. Rep. <em>5</em>, 13652. Available at: <a href="http://www.nature.com/srep/2015/150904/srep13652/full/srep13652.html" class="uri">http://www.nature.com/srep/2015/150904/srep13652/full/srep13652.html</a>.</p>
</div>
<div>
<p>152. Sfriso, P., Duran-Frigola, M., Mosca, R., Emperador, A., Aloy, P., and Orozco, M. (2016). Residues Coevolution Guides the Systematic Identification of Alternative Functional Conformations in Proteins. Structure <em>24</em>, 116–126. Available at: <a href="http://linkinghub.elsevier.com/retrieve/pii/S0969212615004657" class="uri">http://linkinghub.elsevier.com/retrieve/pii/S0969212615004657</a>.</p>
</div>
<div>
<p>153. Sutto, L., Marsili, S., Valencia, A., and Gervasio, F.L. (2015). From residue coevolution to protein conformational ensembles and functional dynamics. Proc. Natl. Acad. Sci. U. S. A., 1508584112. Available at: <a href="http://www.pnas.org/content/early/2015/10/20/1508584112.abstract" class="uri">http://www.pnas.org/content/early/2015/10/20/1508584112.abstract</a>.</p>
</div>
<div>
<p>154. Jana, B., Morcos, F., and Onuchic, J.N. (2014). From structure to function: the convergence of structure based models and co-evolutionary information. Phys. Chem. Chem. Phys. <em>16</em>, 6496. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24603809 http://xlink.rsc.org/?DOI=c3cp55275f" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/24603809 http://xlink.rsc.org/?DOI=c3cp55275f</a>.</p>
</div>
<div>
<p>155. Morcos, F., Jana, B., Hwa, T., and Onuchic, J.N. (2013). Coevolutionary signals across protein lineages help capture multiple protein conformations. Proc. Natl. Acad. Sci. U. S. A. <em>110</em>, 20533–8. Available at: <a href="http://www.pnas.org/content/110/51/20533.short" class="uri">http://www.pnas.org/content/110/51/20533.short</a>.</p>
</div>
<div>
<p>156. Jeon, J., Nam, H.-J., Choi, Y.S., Yang, J.-S., Hwang, J., and Kim, S. (2011). Molecular evolution of protein conformational changes revealed by a network of evolutionarily coupled residues. Mol. Biol. Evol. <em>28</em>, 2675–85. Available at: <a href="http://mbe.oxfordjournals.org/content/28/9/2675.full" class="uri">http://mbe.oxfordjournals.org/content/28/9/2675.full</a>.</p>
</div>
<div>
<p>157. Nawy, T. (2016). Structural biology: RNA structure from sequence. Nat. Methods <em>13</em>, 465–465. Available at: <a href="http://dx.doi.org/10.1038/nmeth.3892" class="uri">http://dx.doi.org/10.1038/nmeth.3892</a>.</p>
</div>
<div>
<p>158. Weinreb, C., Gross, T., Sander, C., and Marks, D.S. (2015). 3D RNA from evolutionary couplings. Available at: <a href="http://arxiv.org/abs/1510.01420" class="uri">http://arxiv.org/abs/1510.01420</a>.</p>
</div>
<div>
<p>159. De Leonardis, E., Lutz, B., Ratz, S., Cocco, S., Monasson, R., Schug, A., and Weigt, M. (2015). Direct-Coupling Analysis of nucleotide coevolution facilitates RNA secondary and tertiary structure prediction. Nucleic Acids Res., gkv932. Available at: <a href="http://nar.oxfordjournals.org/content/early/2015/09/29/nar.gkv932.full" class="uri">http://nar.oxfordjournals.org/content/early/2015/09/29/nar.gkv932.full</a>.</p>
</div>
<div>
<p>160. Suvarna Vani, K., and Praveen Kumar, K. (2018). Feature Extraction of Protein Contact Maps from Protein 3D-Coordinates. In Inf. commun. technol. adv. intell. syst. comput. (Springer, Singapore), pp. 311–320. Available at: <a href="http://link.springer.com/10.1007/978-981-10-5508-9{\_}30" class="uri">http://link.springer.com/10.1007/978-981-10-5508-9{\_}30</a>.</p>
</div>
<div>
<p>161. Woźniak, P., Kotulska, M., and Vriend, G. (2017). Correlated mutations distinguish misfolded and properly folded proteins. Bioinformatics <em>33</em>, 1497–1504.</p>
</div>
<div>
<p>162. Cao, R., Adhikari, B., Bhattacharya, D., Sun, M., Hou, J., and Cheng, J. (2016). QAcon: single model quality assessment using protein structural and contact information with machine learning techniques. Bioinformatics <em>14</em>, btw694. Available at: <a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btw694" class="uri">https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btw694</a>.</p>
</div>
<div>
<p>163. Terashi, G., Nakamura, Y., Shimoyama, H., and Takeda-Shitaka, M. (2014). Quality Assessment Methods for 3D Protein Structure Models Based on a Residue–Residue Distance Matrix Prediction. Chem. Pharm. Bull. <em>62</em>, 744–753. Available at: <a href="https://www.jstage.jst.go.jp/article/cpb/62/8/62{\_}c13-00973/{\_}article" class="uri">https://www.jstage.jst.go.jp/article/cpb/62/8/62{\_}c13-00973/{\_}article</a>.</p>
</div>
<div>
<p>164. Skwark, M.J., Croucher, N.J., Puranen, S., Chewapreecha, C., Pesonen, M., Xu, Y.Y., Turner, P., Harris, S.R., Beres, S.B., and Musser, J.M. <em>et al.</em> (2017). Interacting networks of resistance, virulence and core machinery genes identified by genome-wide epistasis analysis. PLOS Genet. <em>13</em>, e1006508. Available at: <a href="http://dx.plos.org/10.1371/journal.pgen.1006508" class="uri">http://dx.plos.org/10.1371/journal.pgen.1006508</a>.</p>
</div>
<div>
<p>165. Gao, C.-Y., Zhou, H.-J., and Aurell, E. (2017). Correlation-Compressed Direct Coupling Analysis. arXiv. Available at: <a href="http://arxiv.org/abs/1710.04819" class="uri">http://arxiv.org/abs/1710.04819</a>.</p>
</div>
<div>
<p>166. Wu, N.C., Du, Y., Le, S., Young, A.P., Zhang, T.-H., Wang, Y., Zhou, J., Yoshizawa, J.M., Dong, L., and Li, X. <em>et al.</em> (2016). Coupling high-throughput genetics with phylogenetic information reveals an epistatic interaction on the influenza A virus M segment. BMC Genomics <em>17</em>, 46. Available at: <a href="http://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-2358-7" class="uri">http://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-2358-7</a>.</p>
</div>
<div>
<p>167. Figliuzzi, M., Jacquier, H., Schug, A., Tenaillon, O., and Weigt, M. (2015). Coevolutionary landscape inference and the context-dependence of mutations in beta-lactamase TEM-1. Mol. Biol. Evol., msv211. Available at: <a href="http://mbe.oxfordjournals.org/content/early/2015/10/06/molbev.msv211.abstract" class="uri">http://mbe.oxfordjournals.org/content/early/2015/10/06/molbev.msv211.abstract</a>.</p>
</div>
<div>
<p>168. Asti, L., Uguzzoni, G., Marcatili, P., and Pagnani, A. (2016). Maximum-Entropy Models of Sequenced Immune Repertoires Predict Antigen-Antibody Affinity. PLoS Comput. Biol. <em>12</em>, e1004870. Available at: <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004870" class="uri">http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004870</a>.</p>
</div>
<div>
<p>169. Elhanati, Y., Murugan, A., Callan, C.G., Mora, T., and Walczak, A.M. (2014). Quantifying selection in immune receptor repertoires. Proc. Natl. Acad. Sci. U. S. A. <em>111</em>, 9875–80. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24941953 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC4103359" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/24941953 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC4103359</a>.</p>
</div>
<div>
<p>170. Franceus, J., Verhaeghe, T., and Desmet, T. (2016). Correlated positions in protein evolution and engineering. J. Ind. Microbiol. Biotechnol., 1–9. Available at: <a href="http://link.springer.com/10.1007/s10295-016-1811-1" class="uri">http://link.springer.com/10.1007/s10295-016-1811-1</a>.</p>
</div>
<div>
<p>171. Tian, P., and Best, R.B. (2017). How Many Protein Sequences Fold to a Given Structure? A Coevolutionary Analysis. Biophys. J. <em>113</em>, 1719–1730. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/29045866" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/29045866</a>.</p>
</div>
<div>
<p>172. Fox, G., Sievers, F., and Higgins, D.G. (2016). Using de novo protein structure predictions to measure the quality of very large multiple sequence alignments. Bioinformatics <em>32</em>, 814–20. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26568625" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/26568625</a>.</p>
</div>
<div>
<p>173. Monastyrskyy, B., Fidelis, K., Tramontano, A., and Kryshtafovych, A. (2011). Evaluation of residue-residue contact predictions in CASP9. Proteins <em>79 Suppl 1</em>, 119–25. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21928322 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3226919" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/21928322 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC3226919</a>.</p>
</div>
<div>
<p>174. Monastyrskyy, B., D’Andrea, D., Fidelis, K., Tramontano, A., and Kryshtafovych, A. (2014). Evaluation of residue-residue contact prediction in CASP10. Proteins <em>82 Suppl 2</em>, 138–53. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3823628{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=3823628{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>175. Ashkenazy, H., Unger, R., and Kliger, Y. (2009). Optimal data collection for correlated mutation analysis. Proteins <em>74</em>, 545–55. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/18655065" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/18655065</a>.</p>
</div>
<div>
<p>176. Kosciolek, T., and Jones, D.T. (2015). Accurate contact predictions using coevolution techniques and machine learning. Proteins Struct. Funct. Bioinforma., n/a–n/a. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26205532" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/26205532</a>.</p>
</div>
<div>
<p>177. Betts, M.J., and Russell, R.B. Amino Acid Properties and Consequences of Substitutions. In Bioinforma. genet. (Chichester, UK: John Wiley &amp; Sons, Ltd), pp. 289–316. Available at: <a href="http://doi.wiley.com/10.1002/0470867302.ch14" class="uri">http://doi.wiley.com/10.1002/0470867302.ch14</a>.</p>
</div>
<div>
<p>178. Anishchenko, I., Ovchinnikov, S., Kamisetty, H., and Baker, D. (2017). Origins of coevolution between residues distant in protein 3D structures. Proc. Natl. Acad. Sci., 201702664. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28784799 http://www.pnas.org/lookup/doi/10.1073/pnas.1702664114" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/28784799 http://www.pnas.org/lookup/doi/10.1073/pnas.1702664114</a>.</p>
</div>
<div>
<p>179. Marks, D.S., Hopf, T.A., and Sander, C. (2012). Protein structure prediction from sequence variation. Nat. Biotechnol. <em>30</em>, 1072–1080. Available at: <a href="http://www.nature.com/nbt/journal/v30/n11/full/nbt.2419.html" class="uri">http://www.nature.com/nbt/journal/v30/n11/full/nbt.2419.html</a>.</p>
</div>
<div>
<p>180. Buslje, C.M., Santos, J., Delfino, J.M., and Nielsen, M. (2009). Correction for phylogeny, small number of observations and data redundancy improves the identification of coevolving amino acid pairs using mutual information. Bioinformatics <em>25</em>, 1125–31. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19276150 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2672635" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/19276150 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2672635</a>.</p>
</div>
<div>
<p>181. The UniProt Consortium (2013). Update on activities at the Universal Protein Resource (UniProt) in 2013. Nucleic Acids Res. <em>41</em>, D43–7. Available at: <a href="http://nar.oxfordjournals.org/content/41/D1/D43" class="uri">http://nar.oxfordjournals.org/content/41/D1/D43</a>.</p>
</div>
<div>
<p>182. Finn, R.D., Coggill, P., Eberhardt, R.Y., Eddy, S.R., Mistry, J., Mitchell, A.L., Potter, S.C., Punta, M., Qureshi, M., and Sangrador-Vegas, A. <em>et al.</em> (2016). The Pfam protein families database: towards a more sustainable future. Nucleic Acids Res. <em>44</em>, D279–D285. Available at: <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkv1344" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkv1344</a>.</p>
</div>
<div>
<p>183. Remmert, M., Biegert, A., Hauser, A., and Söding, J. (2012). HHblits: lightning-fast iterative protein sequence searching by HMM-HMM alignment. Nat. Methods <em>9</em>, 173–5. Available at: <a href="http://dx.doi.org/10.1038/nmeth.1818" class="uri">http://dx.doi.org/10.1038/nmeth.1818</a>.</p>
</div>
<div>
<p>184. Espada, R., Parra, R.G., Mora, T., Walczak, A.M., and Ferreiro, D. (2015). Capturing coevolutionary signals in repeat proteins. BMC Bioinformatics <em>16</em>, 207. Available at: <a href="http://arxiv.org/abs/1407.6903" class="uri">http://arxiv.org/abs/1407.6903</a>.</p>
</div>
<div>
<p>185. Toth-Petroczy, A., Palmedo, P., Ingraham, J., Hopf, T.A., Berger, B., Sander, C., Marks, D.S., Alexander, P., He, Y., and Chen, Y. <em>et al.</em> (2016). Structured States of Disordered Proteins from Genomic Sequences. Cell <em>167</em>, 158–170.e12. Available at: <a href="http://linkinghub.elsevier.com/retrieve/pii/S0092867416312430" class="uri">http://linkinghub.elsevier.com/retrieve/pii/S0092867416312430</a>.</p>
</div>
<div>
<p>186. Avila-Herrera, A., and Pollard, K.S. (2015). Coevolutionary analyses require phylogenetically deep alignments and better null models to accurately detect inter-protein contacts within and between species. BMC Bioinformatics <em>16</em>, 268. Available at: <a href="http://www.biomedcentral.com/1471-2105/16/268" class="uri">http://www.biomedcentral.com/1471-2105/16/268</a>.</p>
</div>
<div>
<p>187. Lee, B.-C., and Kim, D. (2009). A new method for revealing correlated mutations under the structural and functional constraints in proteins. Bioinformatics <em>25</em>, 2506–13. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19628501" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/19628501</a>.</p>
</div>
<div>
<p>188. Ovchinnikov, S., Kim, D.E., Wang, R.Y.-R., Liu, Y., DiMaio, F., and Baker, D. (2015). Improved de novo structure prediction in CASP11 by incorporating Co-evolution information into rosetta. Proteins. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26677056" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/26677056</a>.</p>
</div>
<div>
<p>189. Noel, J.K., Morcos, F., and Onuchic, J.N. (2016). Sequence co-evolutionary information is a natural partner to minimally-frustrated models of biomolecular dynamics. F1000Research <em>5</em>. Available at: <a href="http://f1000research.com/articles/5-106/v1" class="uri">http://f1000research.com/articles/5-106/v1</a>.</p>
</div>
<div>
<p>190. Burley, S., and Petsko, G. (1985). Aromatic-aromatic interaction: a mechanism of protein structure stabilization. Science (80-. ). <em>229</em>, 23–28. Available at: <a href="http://www.sciencemag.org/cgi/doi/10.1126/science.3892686" class="uri">http://www.sciencemag.org/cgi/doi/10.1126/science.3892686</a>.</p>
</div>
<div>
<p>191. Coucke, A., Uguzzoni, G., Oteri, F., Cocco, S., Monasson, R., and Weigt, M. (2016). Direct coevolutionary couplings reflect biophysical residue interactions in proteins. J. Chem. Phys. <em>145</em>, 174102. Available at: <a href="http://scitation.aip.org/content/aip/journal/jcp/145/17/10.1063/1.4966156" class="uri">http://scitation.aip.org/content/aip/journal/jcp/145/17/10.1063/1.4966156</a>.</p>
</div>
<div>
<p>192. Sillitoe, I., Lewis, T.E., Cuff, A., Das, S., Ashford, P., Dawson, N.L., Furnham, N., Laskowski, R.A., Lee, D., and Lees, J.G. <em>et al.</em> (2015). CATH: comprehensive structural and functional annotations for genome sequences. Nucleic Acids Res. <em>43</em>, D376–D381. Available at: <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gku947" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gku947</a>.</p>
</div>
<div>
<p>193. Hinton, G.E. (2002). Training Products of Experts by Minimizing Contrastive Divergence. Neural Comput. <em>14</em>, 1771–1800. Available at: <a href="http://www.gatsby.ucl.ac.uk/publications/tr/tr00-004.pdf" class="uri">http://www.gatsby.ucl.ac.uk/publications/tr/tr00-004.pdf</a>.</p>
</div>
<div>
<p>194. Andrieu, C., Freitas, N. de, Doucet, A., and Jordan, M.I. (2003). An Introduction to MCMC for Machine Learning. Mach. Learn. <em>50</em>, 5–43. Available at: <a href="http://link.springer.com/10.1023/A:1020281327116" class="uri">http://link.springer.com/10.1023/A:1020281327116</a>.</p>
</div>
<div>
<p>195. Fischer, A., and Igel, C. (2012). An Introduction to Restricted Boltzmann Machines. Lect. Notes Comput. Sci. Prog. Pattern Recognition, Image Anal. Comput. Vision, Appl. <em>7441</em>, 14–36. Available at: <a href="http://link.springer.com/chapter/10.1007/978-3-642-33275-3{\_}2" class="uri">http://link.springer.com/chapter/10.1007/978-3-642-33275-3{\_}2</a>.</p>
</div>
<div>
<p>196. Bengio, Y., and Delalleau, O. (2009). Justifying and Generalizing Contrastive Divergence. Neural Comput. <em>21</em>, 1601–21. Available at: <a href="http://www.iro.umontreal.ca/{~}lisa/publications2/index.php/attachments/single/105" class="uri">http://www.iro.umontreal.ca/{~}lisa/publications2/index.php/attachments/single/105</a>.</p>
</div>
<div>
<p>197. Ruder, S. (2017). An overview of gradient descent optimization algorithms. arXiv. Available at: <a href="http://arxiv.org/abs/1609.04747" class="uri">http://arxiv.org/abs/1609.04747</a>.</p>
</div>
<div>
<p>198. Bottou, L. (2012). Stochastic Gradient Descent Tricks. In Neural networks: Tricks of the trade (Springer, Berlin, Heidelberg), pp. 421–436. Available at: <a href="http://link.springer.com/10.1007/978-3-642-35289-8{\_}25" class="uri">http://link.springer.com/10.1007/978-3-642-35289-8{\_}25</a>.</p>
</div>
<div>
<p>199. Bottou, L. (2010). Large-Scale Machine Learning with Stochastic Gradient Descent. 177–186. Available at: <a href="https://link.springer.com/chapter/10.1007/978-3-7908-2604-3{\_}16" class="uri">https://link.springer.com/chapter/10.1007/978-3-7908-2604-3{\_}16</a>.</p>
</div>
<div>
<p>200. Schaul, T., Zhang, S., and Lecun, Y. (2013). No More Pesky Learning Rates. arXiv. Available at: <a href="https://arxiv.org/pdf/1206.1106.pdf" class="uri">https://arxiv.org/pdf/1206.1106.pdf</a>.</p>
</div>
<div>
<p>201. Zeiler, M.D. (2012). ADADELTA: An Adaptive Learning Rate Method. 6. Available at: <a href="http://arxiv.org/abs/1212.5701" class="uri">http://arxiv.org/abs/1212.5701</a>.</p>
</div>
<div>
<p>202. Bengio, Y. (2012). Practical Recommendations for Gradient-Based Training of Deep Architectures. In Neural networks: Tricks of the trade (Springer Berlin Heidelberg), pp. 437–478. Available at: <a href="https://arxiv.org/pdf/1206.5533v2.pdf" class="uri">https://arxiv.org/pdf/1206.5533v2.pdf</a>.</p>
</div>
<div>
<p>203. Mahsereci, M., Balles, L., Lassner, C., and Hennig, P. (2017). Early Stopping without a Validation Set. arXiv. Available at: <a href="http://arxiv.org/abs/1703.09580" class="uri">http://arxiv.org/abs/1703.09580</a>.</p>
</div>
<div>
<p>204. Carreira-Perpiñán, M. a, and Hinton, G.E. (2005). On Contrastive Divergence Learning. Artif. Intell. Stat. <em>0</em>, 17. Available at: <a href="http://learning.cs.toronto.edu/{~}hinton/absps/cdmiguel.pdf" class="uri">http://learning.cs.toronto.edu/{~}hinton/absps/cdmiguel.pdf</a>.</p>
</div>
<div>
<p>205. Ma, X., and Wang, X. (2016). Average Contrastive Divergence for Training Restricted Boltzmann Machines. Entropy <em>18</em>, 35. Available at: <a href="http://www.mdpi.com/1099-4300/18/1/35" class="uri">http://www.mdpi.com/1099-4300/18/1/35</a>.</p>
</div>
<div>
<p>206. Tieleman, T. (2008). Training Restricted Boltzmann Machines using Approximations to the Likelihood Gradient. Proc. 25th Int. Conf. Mach. Learn. <em>307</em>, 7.</p>
</div>
<div>
<p>207. Fischer, A., and Igel, C. (2010). Empirical Analysis of the Divergence of Gibbs Sampling Based Learning Algorithms for Restricted Boltzmann Machines. In Artif. neural networks – icann 2010 (Springer, Berlin, Heidelberg), pp. 208–217. Available at: <a href="http://link.springer.com/10.1007/978-3-642-15825-4{\_}26" class="uri">http://link.springer.com/10.1007/978-3-642-15825-4{\_}26</a>.</p>
</div>
<div>
<p>208. Hyvärinen, A. (2006). Consistency of pseudolikelihood estimation of fully visible Boltzmann machines. Available at: <a href="https://www.cs.helsinki.fi/u/ahyvarin/papers/NC06.pdf" class="uri">https://www.cs.helsinki.fi/u/ahyvarin/papers/NC06.pdf</a>.</p>
</div>
<div>
<p>209. Hyvarinen, A. (2007). Connections Between Score Matching, Contrastive Divergence, and Pseudolikelihood for Continuous-Valued Variables. IEEE Trans. Neural Networks <em>18</em>, 1529–1531. Available at: <a href="http://ieeexplore.ieee.org/document/4298117/" class="uri">http://ieeexplore.ieee.org/document/4298117/</a>.</p>
</div>
<div>
<p>210. Asuncion, A.U., Liu, Q., Ihler, A.T., and Smyth, P. (2010). Learning with Blocks: Composite Likelihood and Contrastive Divergence. Proc. Mach. Learn. Res. <em>9</em>, 33–40. Available at: <a href="http://www.ics.uci.edu/{~}asuncion/pubs/AISTATS{\_}10.pdf" class="uri">http://www.ics.uci.edu/{~}asuncion/pubs/AISTATS{\_}10.pdf</a>.</p>
</div>
<div>
<p>211. Swersky, K., Chen, B., Marlin, B., and Freitas, N. de (2010). A tutorial on stochastic approximation algorithms for training Restricted Boltzmann Machines and Deep Belief Nets. In 2010 inf. theory appl. work. (IEEE), pp. 1–10. Available at: <a href="http://ieeexplore.ieee.org/document/5454138/" class="uri">http://ieeexplore.ieee.org/document/5454138/</a>.</p>
</div>
<div>
<p>212. Kingma, D., and Ba, J. (2014). Adam: A Method for Stochastic Optimization. Available at: <a href="http://arxiv.org/abs/1412.6980" class="uri">http://arxiv.org/abs/1412.6980</a>.</p>
</div>
<div>
<p>213. Chollet, F. and others (2015). Keras. Available at: <a href="https://github.com/fchollet/keras" class="uri">https://github.com/fchollet/keras</a>.</p>
</div>
<div>
<p>214. Dieleman, S., Schlüter, J., Raffel, C., Olson, E., Sønderby, S.K., Nouri, D., Maturana, D., Thoma, M., Battenberg, E., and Kelly, J. <em>et al.</em> (2015). Lasagne: First release. Available at: <a href="https://zenodo.org/record/27878" class="uri">https://zenodo.org/record/27878</a>.</p>
</div>
<div>
<p>215. Ma, J., Wang, S., Wang, Z., and Xu, J. (2015). Protein contact prediction by integrating joint evolutionary coupling analysis and supervised learning. Bioinformatics, btv472. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/early/2015/09/04/bioinformatics.btv472" class="uri">http://bioinformatics.oxfordjournals.org/content/early/2015/09/04/bioinformatics.btv472</a>.</p>
</div>
<div>
<p>216. Ho, T.K. (1998). The random subspace method for constructing decision forests. IEEE Trans. Pattern Anal. Mach. Intell. <em>20</em>, 832–844. Available at: <a href="http://ieeexplore.ieee.org/document/709601/" class="uri">http://ieeexplore.ieee.org/document/709601/</a>.</p>
</div>
<div>
<p>217. Tin Kam Ho (1995). Random decision forests. In Proc. 3rd int. conf. doc. anal. recognit. (IEEE Comput. Soc. Press), pp. 278–282. Available at: <a href="http://ieeexplore.ieee.org/document/598994/" class="uri">http://ieeexplore.ieee.org/document/598994/</a>.</p>
</div>
<div>
<p>218. Breiman, L. (2001). Random Forests. Mach. Learn. <em>45</em>, 5–32. Available at: <a href="http://link.springer.com/10.1023/A:1010933404324" class="uri">http://link.springer.com/10.1023/A:1010933404324</a>.</p>
</div>
<div>
<p>219. Menze, B.H., Kelm, B.M., Masuch, R., Himmelreich, U., Bachert, P., Petrich, W., and Hamprecht, F.A. (2009). A comparison of random forest and its Gini importance with standard chemometric methods for the feature selection and classification of spectral data. BMC Bioinformatics <em>10</em>, 213. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19591666 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2724423" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/19591666 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2724423</a>.</p>
</div>
<div>
<p>220. Louppe, G. (2014). Understanding Random Forests: From Theory to Practice. Available at: <a href="http://arxiv.org/abs/1407.7502" class="uri">http://arxiv.org/abs/1407.7502</a>.</p>
</div>
<div>
<p>221. Strobl, C., Boulesteix, A.-L., Zeileis, A., and Hothorn, T. (2007). Bias in random forest variable importance measures: Illustrations, sources and a solution. BMC Bioinformatics <em>8</em>, 25. Available at: <a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-8-25" class="uri">http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-8-25</a>.</p>
</div>
<div>
<p>222. Bernard, S., Heutte, L., and Adam, S. (2009). Influence of Hyperparameters on Random Forest Accuracy. In (Springer, Berlin, Heidelberg), pp. 171–180. Available at: <a href="http://link.springer.com/10.1007/978-3-642-02326-2{\_}18" class="uri">http://link.springer.com/10.1007/978-3-642-02326-2{\_}18</a>.</p>
</div>
<div>
<p>223. Fodor, A.A., and Aldrich, R.W. (2004). Influence of conservation on calculations of amino acid covariance in multiple sequence alignments. Proteins <em>56</em>, 211–21. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15211506" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/15211506</a>.</p>
</div>
<div>
<p>224. Miyazawa, S., and Jernigan, R.L. (1999). Self-consistent estimation of inter-residue protein contact energies based on an equilibrium mixture approximation of residues. Proteins <em>34</em>, 49–68. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/10336383" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/10336383</a>.</p>
</div>
<div>
<p>225. Petersen, B., Petersen, T.N., Andersen, P., Nielsen, M., and Lundegaard, C. (2009). BMC Structural Biology A generic method for assignment of reliability scores applied to solvent accessibility predictions. BMC Struct. Biol. <em>9</em>. Available at: <a href="http://www.biomedcentral.com/1472-6807/9/51" class="uri">http://www.biomedcentral.com/1472-6807/9/51</a>.</p>
</div>
<div>
<p>226. Jones, D.T. (1999). Protein secondary structure prediction based on position-specific scoring matrices 1 1Edited by G. Von Heijne. J. Mol. Biol. <em>292</em>, 195–202. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/10493868 http://linkinghub.elsevier.com/retrieve/pii/S0022283699930917" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/10493868 http://linkinghub.elsevier.com/retrieve/pii/S0022283699930917</a>.</p>
</div>
<div>
<p>227. Robinson, A.B., and Robinson, L.R. (1991). Distribution of glutamine and asparagine residues and their near neighbors in peptides and proteins. Proc. Natl. Acad. Sci. U. S. A. <em>88</em>, 8880–4. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/1924347 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC52614" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/1924347 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC52614</a>.</p>
</div>
<div>
<p>228. Atchley, W.R., Zhao, J., Fernandes, A.D., and Drüke, T. (2005). Solving the protein sequence metric problem. Proc. Natl. Acad. Sci. U. S. A. <em>102</em>, 6395–400. Available at: <a href="http://www.pnas.org/content/102/18/6395.abstract" class="uri">http://www.pnas.org/content/102/18/6395.abstract</a>.</p>
</div>
<div>
<p>229. Grantham, R. (1974). Amino acid difference formula to help explain protein evolution. Science <em>185</em>, 862–4. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/4843792" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/4843792</a>.</p>
</div>
<div>
<p>230. Kawashima, S., Pokarowski, P., Pokarowska, M., Kolinski, A., Katayama, T., and Kanehisa, M. (2008). AAindex: amino acid index database, progress report 2008. Nucleic Acids Res. <em>36</em>, D202–5. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17998252 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2238890" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/17998252 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2238890</a>.</p>
</div>
<div>
<p>231. Zimmerman, J.M., Eliezer, N., and Simha, R. (1968). The characterization of amino acid sequences in proteins by statistical methods. J. Theor. Biol. <em>21</em>, 170–201. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/5700434" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/5700434</a>.</p>
</div>
<div>
<p>232. Wimley, W.C., and White, S.H. (1996). Experimentally determined hydrophobicity scale for proteins at membrane interfaces. Nat. Struct. Biol. <em>3</em>, 842–8. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/8836100" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/8836100</a>.</p>
</div>
<div>
<p>233. Kyte, J., and Doolittle, R.F. (1982). A simple method for displaying the hydropathic character of a protein. J. Mol. Biol. <em>157</em>, 105–132. Available at: <a href="http://www.sciencedirect.com/science/article/pii/0022283682905150" class="uri">http://www.sciencedirect.com/science/article/pii/0022283682905150</a>.</p>
</div>
<div>
<p>234. Cornette, J.L., Cease, K.B., Margalit, H., Spouge, J.L., Berzofsky, J.A., and DeLisi, C. (1987). Hydrophobicity scales and computational techniques for detecting amphipathic structures in proteins. J. Mol. Biol. <em>195</em>, 659–685. Available at: <a href="http://linkinghub.elsevier.com/retrieve/pii/0022283687901896" class="uri">http://linkinghub.elsevier.com/retrieve/pii/0022283687901896</a>.</p>
</div>
<div>
<p>235. Pontius, J., Richelle, J., and Wodak, S.J. (1996). Deviations from Standard Atomic Volumes as a Quality Measure for Protein Crystal Structures. J. Mol. Biol. <em>264</em>, 121–136. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/8950272 http://linkinghub.elsevier.com/retrieve/pii/S0022283696906282" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/8950272 http://linkinghub.elsevier.com/retrieve/pii/S0022283696906282</a>.</p>
</div>
<div>
<p>236. Zhu, H., and Braun, W. (1999). Sequence specificity, statistical potentials, and three-dimensional structure prediction with self-correcting distance geometry calculations of beta-sheet formation in proteins. Protein Sci. <em>8</em>, 326–42. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2144259{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2144259{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div>
<p>237. Pedregosa, F., Varoquaux, G., Gramfort, A., Michel, V., Thirion, B., Grisel, O., Blondel, M., Prettenhofer, P., Weiss, R., and Dubourg, V. <em>et al.</em> (2011). Scikit-learn: Machine Learning in Python. J. Mach. Learn. Res. <em>12</em>, 2825–2830. Available at: <a href="http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html" class="uri">http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html</a>.</p>
</div>
<div>
<p>238. Byrd, R.H., Lu, P., Nocedal, J., and Zhu, C. (1995). A Limited Memory Algorithm for Bound Constrained Optimization. SIAM J. Sci. Comput. <em>16</em>, 1190–1208. Available at: <a href="http://epubs.siam.org/doi/10.1137/0916069" class="uri">http://epubs.siam.org/doi/10.1137/0916069</a>.</p>
</div>
<div>
<p>239. Livingstone, C.D., and Barton, G.J. (1993). Protein sequence alignments: a strategy for the hierarchical analysis of residue conservation. Bioinformatics <em>9</em>, 745–756. Available at: <a href="http://bioinformatics.oxfordjournals.org/content/9/6/745" class="uri">http://bioinformatics.oxfordjournals.org/content/9/6/745</a>.</p>
</div>
</div>
</div>
<div id="disqus_thread"></div>
<script>
/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://contactpredictionthesis.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
            </section>

          </div>
        </div>
      </div>
<a href="bayesian-statistical-model-for-contact-prediction.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/susannvorberg/phd_thesis/edit/master/12-references.md",
"text": "Edit"
},
"download": ["PhD_thesis_Susann_Vorberg.pdf"],
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
