<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>PhD thesis: residue-residue contact prediction</title>
  <meta name="description" content="This is my PhD thesis on residue-residue contact prediction.">
  <meta name="generator" content="bookdown 0.3.9 and GitBook 2.6.7">

  <meta property="og:title" content="PhD thesis: residue-residue contact prediction" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is my PhD thesis on residue-residue contact prediction." />
  <meta name="github-repo" content="susannvorberg/phd_thesis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="PhD thesis: residue-residue contact prediction" />
  
  <meta name="twitter:description" content="This is my PhD thesis on residue-residue contact prediction." />
  

<meta name="author" content="Susann Vorberg">


<meta name="date" content="2017-07-21">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="contact-prediction.html">
<link rel="next" href="results.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />








<script type="text/x-mathjax-config">
 MathJax.Hub.Config({
  TeX: { 
    extensions: ["mediawiki-texvc.js"],
    Macros: {
      eq: "\\!=\\!",
      neff: "N_\\mathrm{eff}",
      seq: "\\mathbf{x}",
      v: "\\mathbf{v}",
      w: "\\mathbf{w}"
      }
  }
});
</script>

<!--
<script type="text/x-mathjax-config">
 MathJax.Hub.Config({
   extensions: ["[siunitx]/siunitx.js"]
 });
 MathJax.Ajax.config.path['siunitx']  = 'http://rawgit.com/burnpanck/MathJax-siunitx/master/';
 </script>


<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { TagSide: "left" }
});
</script>
//-->



<link rel="stylesheet" href="css/style.css" type="text/css" />
<link rel="stylesheet" href="css/toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PhD Thesis Susann Vorberg</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Summary</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="structure-prediction.html"><a href="structure-prediction.html"><i class="fa fa-check"></i><b>1.1</b> Structure Prediction</a><ul>
<li class="chapter" data-level="1.1.1" data-path="structure-prediction.html"><a href="structure-prediction.html#template-based-methods"><i class="fa fa-check"></i><b>1.1.1</b> Template-based methods</a></li>
<li class="chapter" data-level="1.1.2" data-path="structure-prediction.html"><a href="structure-prediction.html#template-free-structure-prediction"><i class="fa fa-check"></i><b>1.1.2</b> Template-free structure prediction</a></li>
<li class="chapter" data-level="1.1.3" data-path="structure-prediction.html"><a href="structure-prediction.html#contact-assisted-denovo-predictions"><i class="fa fa-check"></i><b>1.1.3</b> contact assisted denovo predictions</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="contact-prediction.html"><a href="contact-prediction.html"><i class="fa fa-check"></i><b>1.2</b> Contact Prediction</a><ul>
<li class="chapter" data-level="1.2.1" data-path="contact-prediction.html"><a href="contact-prediction.html#correlated-mutations"><i class="fa fa-check"></i><b>1.2.1</b> Correlated mutations</a></li>
<li class="chapter" data-level="1.2.2" data-path="contact-prediction.html"><a href="contact-prediction.html#benchmarking-methods"><i class="fa fa-check"></i><b>1.2.2</b> Benchmarking methods</a></li>
<li class="chapter" data-level="1.2.3" data-path="contact-prediction.html"><a href="contact-prediction.html#pitfalls"><i class="fa fa-check"></i><b>1.2.3</b> Pitfalls</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="state-of-the-art-cp.html"><a href="state-of-the-art-cp.html"><i class="fa fa-check"></i><b>1.3</b> State of the Art CP</a><ul>
<li class="chapter" data-level="1.3.1" data-path="state-of-the-art-cp.html"><a href="state-of-the-art-cp.html#correlation-vs-causation"><i class="fa fa-check"></i><b>1.3.1</b> Correlation vs Causation</a></li>
<li class="chapter" data-level="1.3.2" data-path="state-of-the-art-cp.html"><a href="state-of-the-art-cp.html#local-methods"><i class="fa fa-check"></i><b>1.3.2</b> Local methods</a></li>
<li class="chapter" data-level="1.3.3" data-path="state-of-the-art-cp.html"><a href="state-of-the-art-cp.html#maxent"><i class="fa fa-check"></i><b>1.3.3</b> Maximum entropy models for modelling protein families</a></li>
<li class="chapter" data-level="1.3.4" data-path="state-of-the-art-cp.html"><a href="state-of-the-art-cp.html#computing-contact-map-from-coupling-matrix"><i class="fa fa-check"></i><b>1.3.4</b> Computing contact map from coupling matrix</a></li>
<li class="chapter" data-level="1.3.5" data-path="state-of-the-art-cp.html"><a href="state-of-the-art-cp.html#meta-predictors"><i class="fa fa-check"></i><b>1.3.5</b> Meta-predictors</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>2</b> Results</a><ul>
<li class="chapter" data-level="2.1" data-path="optimizing-full-likelihood.html"><a href="optimizing-full-likelihood.html"><i class="fa fa-check"></i><b>2.1</b> Optimizing Full-Likelihood</a><ul>
<li class="chapter" data-level="2.1.1" data-path="optimizing-full-likelihood.html"><a href="optimizing-full-likelihood.html#likelihood-gradient"><i class="fa fa-check"></i><b>2.1.1</b> Likelihood Gradient</a></li>
<li class="chapter" data-level="2.1.2" data-path="optimizing-full-likelihood.html"><a href="optimizing-full-likelihood.html#contrastive-divergence"><i class="fa fa-check"></i><b>2.1.2</b> Contrastive Divergence</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods</a><ul>
<li class="chapter" data-level="3.1" data-path="dataset.html"><a href="dataset.html"><i class="fa fa-check"></i><b>3.1</b> Dataset</a></li>
<li class="chapter" data-level="3.2" data-path="contact-prediction-1.html"><a href="contact-prediction-1.html"><i class="fa fa-check"></i><b>3.2</b> Contact Prediction</a><ul>
<li class="chapter" data-level="3.2.1" data-path="contact-prediction-1.html"><a href="contact-prediction-1.html#seq-reweighting"><i class="fa fa-check"></i><b>3.2.1</b> Sequence Reweighting</a></li>
<li class="chapter" data-level="3.2.2" data-path="contact-prediction-1.html"><a href="contact-prediction-1.html#computing-amino-acid-frequencies"><i class="fa fa-check"></i><b>3.2.2</b> Computing Amino Acid Frequencies</a></li>
<li class="chapter" data-level="3.2.3" data-path="contact-prediction-1.html"><a href="contact-prediction-1.html#pseudolikelihood"><i class="fa fa-check"></i><b>3.2.3</b> Pseudolikelihood</a></li>
<li class="chapter" data-level="3.2.4" data-path="contact-prediction-1.html"><a href="contact-prediction-1.html#full-likelihood"><i class="fa fa-check"></i><b>3.2.4</b> Full-likelihood</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="abbrev.html"><a href="abbrev.html"><i class="fa fa-check"></i><b>A</b> Abbreviations</a></li>
<li class="chapter" data-level="B" data-path="dataset-properties.html"><a href="dataset-properties.html"><i class="fa fa-check"></i><b>B</b> Dataset Properties</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PhD thesis: residue-residue contact prediction</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="state-of-the-art-cp" class="section level2">
<h2><span class="header-section-number">1.3</span> State of the Art CP</h2>
<div id="correlation-vs-causation" class="section level3">
<h3><span class="header-section-number">1.3.1</span> Correlation vs Causation</h3>
<p>A recurring difficulty when dealing with interacting systems is distinguishing the direct interactions, resulting from adjacent interplay between two units, from interactions mediated via multi-step paths across other elements [Ekeberg master thesis]</p>
</div>
<div id="local-methods" class="section level3">
<h3><span class="header-section-number">1.3.2</span> Local methods</h3>
<p>MI and correlation measures suffer from transitivity of correlations</p>
</div>
<div id="maxent" class="section level3">
<h3><span class="header-section-number">1.3.3</span> Maximum entropy models for modelling protein families</h3>
<p>The principle of maximum entropy was postulated by Jaynes in 1957 <span class="citation">[<a href="#ref-Jaynes1957a">20</a>]</span> <span class="citation">[<a href="#ref-Jaynes1957b">21</a>]</span>, stating that the probability distribution which best represents observed data is the one that is in agreement with measured constraints and has the largest entropy. It is a distribution that makes minimal assumptions and is the least biased estimate of a distribution given the measured constraints.</p>
<p>Applied to the problem of modelling protein families, one would seek a probability distribution over protein sequences <span class="math inline">\(\seq = (x_i, ..., x_L)\)</span> of length <span class="math inline">\(L\)</span> reproducing the empirical single <span class="math inline">\(\mathcal{f}(x_i \eq a)\)</span> and pairwise <span class="math inline">\(\mathcal{f}(x_i \eq a, x_j \eq b)\)</span> amino acid frequencies of an alignment <span class="math inline">\(\mathbf{X}\)</span> with <span class="math inline">\(N\)</span> sequences:</p>
<span class="math display" id="eq:emp-freq">\[\begin{equation}
 p(x_i\eq a) = \mathcal{f}(x_i\eq a) = \frac{1}{N}\sum_{n=1}^N I(x_i^{(n)} \eq a) \\
 p(x_i\eq a, x_j \eq b) = \mathcal{f}(x_i\eq a, x_j\eq b) = \frac{1}{N} \sum_{n=1}^N I(x_i^{(n)} \eq a, x_j^{(n)} \eq b)
 \tag{1.1}
\end{equation}\]</span>
<p>Maximizing the entropy <span class="math inline">\(S= -\sum_{\seq} p(\seq) \log p(\seq)\)</span> of the distribution under the given constraints <a href="state-of-the-art-cp.html#eq:emp-freq">(1.1)</a> using Lagrange multipliers results in the formulation of an exponential model known as <em>Potts model</em> or its generalized form a <em>Markov Random Field</em>:</p>
<span class="math display" id="eq:max-ent-model">\[\begin{equation}
p(\seq) = \frac{1}{Z} \exp \left( \sum_{i=1}^L v_i(x_i) \sum_{\substack{i,j=1 \\ i \neq j}}^L w_{ij}(x_i, x_j) \right)
\tag{1.2}
\end{equation}\]</span>
<p><span class="math inline">\(Z\)</span> is a normalization constant also known as <em>partition function</em> ensuring the total probabilty adds up to one by summing over all possible assignments to <span class="math inline">\(\seq\)</span>.</p>
<span class="math display" id="eq:partition-fct-likelihood">\[\begin{equation}
  Z = \sum_{\seq&#39; \in [1,...,20]^L} p(\seq&#39; | \v, \w)
  \tag{1.3}
\end{equation}\]</span>
<div id="interpretation-of-model-parameters" class="section level4">
<h4><span class="header-section-number">1.3.3.1</span> Interpretation of model parameters</h4>
<p>Markov Random Field belongs to class of undirected graphical models. Therefore they can be represented as a graph with nodes corresponding to positions in the alignment and edges describing the dependency structure between nodes.</p>
<p>A protein sequence can be encoded in multiple ways, altering the resultign model.</p>
<p>Using a binary encoding for amino acids <span class="math inline">\(x_i \in [0, 1]^{21}\)</span>, a sequence <span class="math inline">\(x\)</span> can be interpreted as continous random variable. For continous, real-valued variables x, the maximum entropy model can be represented as a graphical Gaussian model. This interpretation of the model is used as the basis for PSICOV.</p>
<p>Otherwise sequences are represented as multinomial variables with 21 states (20 amino acids and one gap).</p>
<p>The single-emission potentials <span class="math inline">\(v_i(a)\)</span> correspond to the single amino acid frequencies whereas the pairwise-emission potentials <span class="math inline">\(w_{ij}(a,b)\)</span> also called couplings reflect the tendency of an amino acid a at position i to co-occur with an amino acid b at position j in the alignment.</p>
<p>Can disentangle direct and indirect correlations.</p>
<p>Infer parameters of a maximum entropy model, more specifically a Potts model (statistical physics) aka markov random fiels (computer science). Likelihood function is convex, but Maximum Likelihood inference of model is infeasible: Likelihood function needs to be reevaluated at each iteration during optimization but partition function term sums over 20^L sequences.</p>
<p>Many approximations: - mean field (Marks), Psicov - pseudo-likelihood - belief-propagation (accurate but slow)</p>
</div>
<div id="infering-model-parameters" class="section level4">
<h4><span class="header-section-number">1.3.3.2</span> Infering model parameters</h4>
<p>Typically, one would obtain parameter estimates by maximizing the log-likelihood of observed data.</p>
<p>However, given the 20 amino acids and typical proteins of length <span class="math inline">\(L=100\)</span>, the partition function <span class="math inline">\(Z\)</span> has to sum over <span class="math inline">\(20^100\)</span> sequences. Because of this exponential complexityit is computationally intractable to evaluate the likelihood-function at every iteration of an optimization procedure.</p>
<p>Several approaches have been developed to evade the infeasible computation of the partition function and obtain approximate solutions for the model parameters. Some of these approaches have successfully been adapted to the problem of modelling protein families.</p>
</div>
<div id="pseudo-likelihood" class="section level4">
<h4><span class="header-section-number">1.3.3.3</span> Pseudo-Likelihood</h4>
<p>Instead of optimizing the likelihood, Besag suggested in 1975 to rather optimize a function that he termed <em>pseudo-likelihood</em> that replaces the joint probability with the product over conditionals <span class="citation">[<a href="#ref-Besag1975">22</a>]</span>:</p>
<span class="math display">\[\begin{equation}
  p(\seq | \v,\w) = \prod_{i=1}^L p(x_i | \seq_{/xi}\v,\w)  = \prod_{i=1}^L \left( \frac{1}{Z_i} \exp[v_i(x_i)] \prod_{\substack{j=1 \\ j \neq i}}^L \exp[w_{ij}(x_i, x_j)]   \right)
\end{equation}\]</span>
<p>Here, the normalization term <span class="math inline">\(Z_i\)</span> sums only over all assignments to one position <span class="math inline">\(i\)</span> in sequence:</p>
<span class="math display">\[\begin{equation}
  Z_i = \sum_{a=1}^{20} \exp[v_i(a)] \prod_{\substack{j=1 \\ j \neq i}}^L \exp[w_{ij}(a, x_j)] 
\end{equation}\]</span>
<p>Besag showed that <em>pseudo-likelihood</em> is an consistent estimator for the likelihood in the limit of large data.</p>
<p>the proba- bility of them-th observation, xm, is approximated by the product of the conditional probabilities<span class="citation">[<a href="#ref-Stein2015a">23</a>]</span></p>
<p>Recall from above that our pseudo-likelihood uses the full representation and fixes the gauge by the regularization terms Rl2. Our procedure is therefore first to infer the interaction parameters using using the pseudo-likelihood and the regularization, and then change to the zero-sum gauge <span class="citation">[<a href="#ref-Ekeberg2013">24</a>]</span></p>
<p>On one hand, this improvement might not be surpris-ing: it is known that, for very large data sets, pseudo- likelihood maximization becomes asymptotically equiva- lent to full maximum-likelihood inference, whereas mean- field inference remains intrinsically approximate, and this may result in an improved PLM performance. On the other hand, the above advantage holds if and only if the following two conditions are fulfilled: data a drawn independently from a probability distribution, and this probability distribution is the the Boltzmann distribution of a Potts model. None of these two con- ditions actually hold for real protein sequences <span class="citation">[<a href="#ref-Ekeberg2013">24</a>]</span></p>
<p>Disregarding the improvements, we find that overall the predicted contact pairs for plmDCA and mfDCA are highly overlapping, illustrating the robustness of DCA results with respect to the algorithmic implementation<span class="citation">[<a href="#ref-Ekeberg2013">24</a>]</span></p>
<p>The pseudo-likelihood inherits the concavity<span class="citation">[<a href="#ref-Stein2015a">23</a>]</span></p>
</div>
</div>
<div id="computing-contact-map-from-coupling-matrix" class="section level3">
<h3><span class="header-section-number">1.3.4</span> Computing contact map from coupling matrix</h3>
<ul>
<li>direct information</li>
<li>frobenius norm</li>
</ul>
<p>average product correction (also for MI)</p>
<p>(benchmark plot for localmethods + ccmpred)</p>
</div>
<div id="meta-predictors" class="section level3">
<h3><span class="header-section-number">1.3.5</span> Meta-predictors</h3>
<ul>
<li>combining different approaches</li>
<li>jones et al: overlap between methods but also many unique predictions</li>
<li>machine learning methods incorporate sequence-derived features:</li>
<li>secondary structure predictions</li>
<li>solvent accessibilty</li>
<li>contact potentials</li>
<li>msa properties</li>
<li>pssms</li>
<li>physico-chemcial properties of amino acids</li>
</ul>
<p>However, Meta-predictors will improve if basic methods improve. Ultra-deep learning paper identifies coevolution features as crucial feature.</p>

</div>
</div>
<!-- </div> -->
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-Jaynes1957a">
<p>20. Jaynes, E.T. (1957). Information Theory and Statistical Mechanics I. Phys. Rev. <em>106</em>, 620–630. Available at: <a href="https://link.aps.org/doi/10.1103/PhysRev.106.620" class="uri">https://link.aps.org/doi/10.1103/PhysRev.106.620</a>.</p>
</div>
<div id="ref-Jaynes1957b">
<p>21. Jaynes, E.T. (1957). Information Theory and Statistical Mechanics. II. Phys. Rev. <em>108</em>, 171–190. Available at: <a href="https://link.aps.org/doi/10.1103/PhysRev.108.171" class="uri">https://link.aps.org/doi/10.1103/PhysRev.108.171</a>.</p>
</div>
<div id="ref-Besag1975">
<p>22. Besag, J. (1975). Statistical Analysis of Non-Lattice Data. Source Stat. <em>24</em>, 179–195. Available at: <a href="http://www.jstor.org http://www.jstor.org/stable/2987782" class="uri">http://www.jstor.org http://www.jstor.org/stable/2987782</a>.</p>
</div>
<div id="ref-Stein2015a">
<p>23. Stein, R.R., Marks, D.S., and Sander, C. (2015). Inferring Pairwise Interactions from Biological Data Using Maximum-Entropy Probability Models. PLOS Comput. Biol. <em>11</em>, e1004182. Available at: <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=4520494{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=4520494{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
<div id="ref-Ekeberg2013">
<p>24. Ekeberg, M., Lövkvist, C., Lan, Y., Weigt, M., and Aurell, E. (2013). Improved contact prediction in proteins: Using pseudolikelihoods to infer Potts models. Phys. Rev. E <em>87</em>, 012707. Available at: <a href="http://link.aps.org/doi/10.1103/PhysRevE.87.012707" class="uri">http://link.aps.org/doi/10.1103/PhysRevE.87.012707</a>.</p>
</div>
</div>
<div id="disqus_thread"></div>
<script>
/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://contactpredictionthesis.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
            </section>

          </div>
        </div>
      </div>
<a href="contact-prediction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="results.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/susannvorberg/phd_thesis/edit/master/03-intro.Rmd",
"text": "Edit"
},
"download": ["PhD thesis Susann Vorberg.pdf", "PhD thesis Susann Vorberg.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
